{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential Expression with limma-voom pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressMessages({library(SummarizedExperiment)\n",
    "                  library(data.table)\n",
    "                  library(tidyverse)\n",
    "                  library(memoise)\n",
    "                  library(limma)\n",
    "                  library(edgeR)\n",
    "                  library(sva)})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function from jaffelab github\n",
    "merge_rse_metrics <- function(rse) {\n",
    "    stopifnot(is(rse, 'RangedSummarizedExperiment'))\n",
    "\n",
    "    rse$overallMapRate = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$overallMapRate, rse$numReads)\n",
    "    rse$mitoRate = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$mitoRate, rse$numMapped)\n",
    "    rse$rRNA_rate = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$rRNA_rate, rse$numMapped)\n",
    "    rse$totalAssignedGene = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$totalAssignedGene, rse$numMapped)\n",
    "\n",
    "    rse$numMapped = sapply(rse$numMapped, sum)\n",
    "    rse$numReads = sapply(rse$numReads, sum)\n",
    "    rse$numUnmapped = sapply(rse$numUnmapped, sum)\n",
    "    rse$mitoMapped = sapply(rse$mitoMapped, sum)\n",
    "    rse$totalMapped = sapply(rse$totalMapped, sum)\n",
    "    return(rse)\n",
    "}\n",
    "\n",
    "\n",
    "save_volcanoPlot <- function(top, label, feature){\n",
    "    pdf(file=paste0(feature, \"/volcanoPlot_\", label, \".pdf\"), 8, 6)\n",
    "    with(top, plot(logFC, -log10(P.Value), pch=20, cex=0.6))\n",
    "    with(subset(top, adj.P.Val<=0.05), points(logFC, -log10(P.Value),\n",
    "                                              pch=20, col='red', cex=0.6))\n",
    "    with(subset(top, abs(logFC)>0.50), points(logFC, -log10(P.Value),\n",
    "                                              pch=20, col='orange', cex=0.6))\n",
    "    with(subset(top, adj.P.Val<=0.05 & abs(logFC)>0.50),\n",
    "         points(logFC, -log10(P.Value), pch=20, col='green', cex=0.6))\n",
    "    dev.off()\n",
    "}\n",
    "\n",
    "\n",
    "save_MAplot <- function(top, label, feature){\n",
    "    pdf(file=paste0(feature, \"/MAplot_\", label, \".pdf\"), 8, 6)\n",
    "    with(top, plot(AveExpr, logFC, pch=20, cex=0.5))\n",
    "    with(subset(top, adj.P.Val<0.05),\n",
    "         points(AveExpr, logFC, col=\"red\", pch=20, cex=0.5))\n",
    "    dev.off()\n",
    "}\n",
    "\n",
    "\n",
    "extract_de <- function(contrast, label, efit, feature){\n",
    "    top <- topTable(efit, coef=contrast, number=Inf, sort.by=\"P\")\n",
    "    top <- top[order(top$P.Value), ]\n",
    "    top.fdr <- top %>% filter(adj.P.Val<=0.05)\n",
    "    print(paste(\"Comparison for:\", label))\n",
    "    print(paste('There are:', dim(top.fdr)[1], 'DE features!'))\n",
    "    fwrite(top, \n",
    "           file=paste0(feature, \"/diffExpr_\", label, \"_full.txt\"), \n",
    "           sep='\\t', row.names=TRUE)\n",
    "    fwrite(top.fdr, \n",
    "           file=paste0(feature, \"/diffExpr_\", label, \"_FDR05.txt\"), \n",
    "           sep='\\t', row.names=TRUE)\n",
    "    save_volcanoPlot(top, label, feature)\n",
    "    save_MAplot(top, label, feature)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cached functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "prep_data <- function(feature){\n",
    "    ancestry = \"../../../input/ancestry_structure/structure.out_ancestry_proportion_raceDemo_compare\"\n",
    "    counts_lt = list(\"genes\"=\"../../../input/counts/_m/dlpfc_ribozero_brainseq_phase2_hg38_rseGene_merged_n453.rda\", \n",
    "                     \"transcripts\"=\"../../../input/counts/_m/dlpfc_ribozero_brainseq_phase2_hg38_rseTx_merged_n453.rda\",\n",
    "                     \"exons\"=\"../../../input/counts/_m/dlpfc_ribozero_brainseq_phase2_hg38_rseExon_merged_n453.rda\",\n",
    "                     \"junctions\"=\"../../../input/counts/_m/dlpfc_ribozero_brainseq_phase2_hg38_rseJxn_merged_n453.rda\")\n",
    "    tx_file = '../../../input/counts/_m/transcripts_counts/dlpfc_counts.txt'\n",
    "    load(counts_lt[[feature]])\n",
    "    if(exists(\"rse_gene\")){\n",
    "        rse_df = rse_gene\n",
    "    } else if (exists(\"rse_tx\")){\n",
    "        counts <- fread(tx_file) %>% column_to_rownames(\"transcript_id\") %>%\n",
    "            select(colnames(rse_tx)) %>% as.matrix\n",
    "        annot <- fread(tx_file) %>% column_to_rownames(\"transcript_id\") %>%\n",
    "            select(-starts_with(\"R\"))\n",
    "        rse_df <- SummarizedExperiment(assays=SimpleList(counts=counts),\n",
    "                                       rowData=annot, \n",
    "                                       colData=colData(rse_tx)) %>%\n",
    "            as(\"RangedSummarizedExperiment\")\n",
    "    } else if (exists(\"rse_exon\")){\n",
    "        rse_df = rse_exon\n",
    "    } else {\n",
    "        rse_df = rse_jxn\n",
    "    }\n",
    "    keepIndex = which((rse_df$Dx %in% c(\"Control\")) & \n",
    "                      rse_df$Age > 17 & \n",
    "                      rse_df$Race %in% c(\"AA\", \"CAUC\"))\n",
    "    rse_df = rse_df[, keepIndex]\n",
    "    rse_df$Sex <- factor(rse_df$Sex)\n",
    "    rse_df <- merge_rse_metrics(rse_df)\n",
    "    colData(rse_df)$RIN = sapply(colData(rse_df)$RIN,\"[\",1)\n",
    "    rownames(colData(rse_df)) <- sapply(strsplit(rownames(colData(rse_df)), \"_\"), \"[\", 1)\n",
    "    pheno = colData(rse_df) %>% as.data.frame %>% \n",
    "        inner_join(fread(ancestry), by=c(\"BrNum\"=\"id\", \"Race\"=\"group\"))\n",
    "    # Generate DGE list\n",
    "    x <- DGEList(counts=assays(rse_df)$counts[, pheno$RNum], \n",
    "                 genes=rowData(rse_df), \n",
    "                 samples=pheno)\n",
    "    # Filter by expression\n",
    "    design0 <- model.matrix(~Race, data=x$samples)\n",
    "    keep.x <- filterByExpr(x, design=design0)\n",
    "    x <- x[keep.x, , keep.lib.sizes=FALSE]\n",
    "    print(paste('There are:', sum(keep.x), 'features left!', sep=' '))\n",
    "    # Normalize library size\n",
    "    x <- calcNormFactors(x, method=\"TMM\")\n",
    "    return(x)\n",
    "}\n",
    "\n",
    "memo_prepData <- memoise(prep_data)\n",
    "\n",
    "qSV_model <- function(feature){\n",
    "    x <- memo_prepData(feature)\n",
    "    # Design matrix\n",
    "    mod = model.matrix(~Eur + Sex + Age + mitoRate + rRNA_rate + \n",
    "                       totalAssignedGene + overallMapRate, data = x$samples)\n",
    "    colnames(mod) <- gsub(\"SexM\", \"Male\", colnames(mod))\n",
    "    colnames(mod) <- gsub(\"Eur\", \"EA\", colnames(mod))\n",
    "    colnames(mod) <- gsub(\"\\\\(Intercept\\\\)\", \"Intercept\", colnames(mod))\n",
    "    # Load qSV\n",
    "    qsv_file = \"../../../input/phenotypes/_m/qSV_dlpfc.csv\"\n",
    "    modQsva <- mod %>% as.data.frame %>% rownames_to_column() %>%\n",
    "        inner_join(fread(qsv_file), by=c(\"rowname\"=\"V1\")) %>% \n",
    "        rename_all(list(~str_replace_all(., 'PC', 'qPC'))) %>% \n",
    "        column_to_rownames(\"rowname\") %>% as.matrix\n",
    "    return(modQsva)\n",
    "}\n",
    "\n",
    "memo_qsvModel <- memoise(qSV_model)\n",
    "\n",
    "get_voom <- function(feature){\n",
    "    ### Preform voom\n",
    "    x <- memo_prepData(feature)\n",
    "    modQsva <- memo_qsvModel(feature)\n",
    "    v <- voom(x[, rownames(modQsva)], modQsva, plot=TRUE)\n",
    "    return(v)\n",
    "}\n",
    "\n",
    "memo_voom <- memoise(get_voom)\n",
    "\n",
    "cal_res <- function(feature){\n",
    "    ### Calculate residuals\n",
    "    v <- memo_voom(feature)\n",
    "    null_model <- v$design %>% as.data.frame %>% select(-c(\"EA\")) %>% as.matrix\n",
    "    fit_res <- lmFit(v, design=null_model)\n",
    "    res = v$E - ( fit_res$coefficients %*% t(null_model) )\n",
    "    res_sd = apply(res, 1, sd)\n",
    "    res_mean = apply(res, 1, mean)\n",
    "    res_norm = (res - res_mean) / res_sd\n",
    "    write.table(res_norm, file=paste0(feature, '/residualized_expression.tsv'), \n",
    "                sep=\"\\t\", quote=FALSE)\n",
    "}\n",
    "\n",
    "memo_res <- memoise(cal_res)\n",
    "\n",
    "fit_voom <- function(feature){\n",
    "    v <- memo_voom(feature)\n",
    "    modQsva <- memo_qsvModel(feature)\n",
    "    fit0 <- lmFit(v, modQsva)\n",
    "    contr.matrix <- makeContrasts(EAvsAA=EA, \n",
    "                                  levels=colnames(modQsva))\n",
    "    fit <- contrasts.fit(fit0, contrasts=contr.matrix)\n",
    "    esv <- eBayes(fit)\n",
    "    return(esv)\n",
    "}\n",
    "\n",
    "memo_efit <- memoise(fit_voom)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential Expression Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for(feature in c('genes', 'transcripts', 'junctions', 'exons')){\n",
    "    dir.create(feature)\n",
    "    # Preform voom\n",
    "    v <- memo_voom(feature)\n",
    "    save(v, file=paste0(feature,'/voomSVA.RData'))\n",
    "    # Fit model and apply eBayes\n",
    "    efit = memo_efit(feature)\n",
    "    # Save differential expression\n",
    "    extract_de(1, \"EAvsAA\", efit, feature)\n",
    "    # Calculate residuals\n",
    "    memo_res(feature)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reproducibility Information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"2020-01-09 09:00:07 EST\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "     user    system   elapsed \n",
       "15709.702   132.946   376.450 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "- Session info -------------------------------------------------------------------------------------------------------\n",
       " setting  value                       \n",
       " version  R version 3.6.1 (2019-07-05)\n",
       " os       Arch Linux                  \n",
       " system   x86_64, linux-gnu           \n",
       " ui       X11                         \n",
       " language (EN)                        \n",
       " collate  C                           \n",
       " ctype    C                           \n",
       " tz       America/New_York            \n",
       " date     2020-01-09                  \n",
       "\n",
       "- Packages -----------------------------------------------------------------------------------------------------------\n",
       " package              * version    date       lib source                            \n",
       " annotate               1.62.0     2019-05-02 [1] Bioconductor                      \n",
       " AnnotationDbi          1.46.1     2019-08-20 [1] Bioconductor                      \n",
       " assertthat             0.2.1      2019-03-21 [1] CRAN (R 3.6.1)                    \n",
       " backports              1.1.5      2019-10-02 [1] CRAN (R 3.6.1)                    \n",
       " base64enc              0.1-3      2015-07-28 [1] CRAN (R 3.6.1)                    \n",
       " Biobase              * 2.44.0     2019-05-02 [1] Bioconductor                      \n",
       " BiocGenerics         * 0.30.0     2019-05-02 [1] Bioconductor                      \n",
       " BiocParallel         * 1.18.1     2019-08-06 [1] Bioconductor                      \n",
       " bit                    1.1-14     2018-05-29 [1] CRAN (R 3.6.1)                    \n",
       " bit64                  0.9-7      2017-05-08 [1] CRAN (R 3.6.1)                    \n",
       " bitops                 1.0-6      2013-08-17 [1] CRAN (R 3.6.1)                    \n",
       " blob                   1.2.0      2019-07-09 [1] CRAN (R 3.6.1)                    \n",
       " calibrate            * 1.7.5      2019-10-01 [1] CRAN (R 3.6.1)                    \n",
       " cli                    1.1.0      2019-03-19 [1] CRAN (R 3.6.1)                    \n",
       " colorspace             1.4-1      2019-03-18 [1] CRAN (R 3.6.1)                    \n",
       " crayon                 1.3.4      2017-09-16 [1] CRAN (R 3.6.1)                    \n",
       " data.table           * 1.12.6     2019-10-18 [1] CRAN (R 3.6.1)                    \n",
       " DBI                    1.0.0      2018-05-02 [1] CRAN (R 3.6.1)                    \n",
       " DelayedArray         * 0.10.0     2019-05-02 [1] Bioconductor                      \n",
       " digest                 0.6.23     2019-11-23 [1] CRAN (R 3.6.1)                    \n",
       " dplyr                  0.8.3      2019-07-04 [1] CRAN (R 3.6.1)                    \n",
       " edgeR                * 3.26.8     2019-09-01 [1] Bioconductor                      \n",
       " evaluate               0.14       2019-05-28 [1] CRAN (R 3.6.1)                    \n",
       " farver                 2.0.1      2019-11-13 [1] CRAN (R 3.6.1)                    \n",
       " genefilter           * 1.66.0     2019-05-02 [1] Bioconductor                      \n",
       " GenomeInfoDb         * 1.20.0     2019-05-02 [1] Bioconductor                      \n",
       " GenomeInfoDbData       1.2.1      2019-08-28 [1] Bioconductor                      \n",
       " GenomicRanges        * 1.36.1     2019-09-06 [1] Bioconductor                      \n",
       " ggplot2              * 3.2.1      2019-08-10 [1] CRAN (R 3.6.1)                    \n",
       " glue                   1.3.1      2019-03-12 [1] CRAN (R 3.6.1)                    \n",
       " gtable                 0.3.0      2019-03-25 [1] CRAN (R 3.6.1)                    \n",
       " htmltools              0.4.0      2019-10-04 [1] CRAN (R 3.6.1)                    \n",
       " IRanges              * 2.18.3     2019-09-24 [1] Bioconductor                      \n",
       " IRdisplay              0.7.0      2018-11-29 [1] CRAN (R 3.6.1)                    \n",
       " IRkernel               1.0.2.9000 2019-08-28 [1] Github (IRkernel/IRkernel@d7f0b08)\n",
       " jsonlite               1.6        2018-12-07 [1] CRAN (R 3.6.1)                    \n",
       " labeling               0.3        2014-08-23 [1] CRAN (R 3.6.1)                    \n",
       " lattice                0.20-38    2018-11-04 [2] CRAN (R 3.6.1)                    \n",
       " lazyeval               0.2.2      2019-03-15 [1] CRAN (R 3.6.1)                    \n",
       " lifecycle              0.1.0      2019-08-01 [1] CRAN (R 3.6.1)                    \n",
       " limma                * 3.40.6     2019-07-26 [1] Bioconductor                      \n",
       " locfit                 1.5-9.1    2013-04-20 [1] CRAN (R 3.6.1)                    \n",
       " magrittr               1.5        2014-11-22 [1] CRAN (R 3.6.1)                    \n",
       " MASS                 * 7.3-51.4   2019-03-31 [2] CRAN (R 3.6.1)                    \n",
       " Matrix                 1.2-17     2019-03-22 [2] CRAN (R 3.6.1)                    \n",
       " matrixStats          * 0.55.0     2019-09-07 [1] CRAN (R 3.6.1)                    \n",
       " memoise                1.1.0      2017-04-21 [1] CRAN (R 3.6.1)                    \n",
       " mgcv                 * 1.8-28     2019-03-21 [2] CRAN (R 3.6.1)                    \n",
       " munsell                0.5.0      2018-06-12 [1] CRAN (R 3.6.1)                    \n",
       " nlme                 * 3.1-140    2019-05-12 [2] CRAN (R 3.6.1)                    \n",
       " pbdZMQ                 0.3-3      2018-05-05 [1] CRAN (R 3.6.1)                    \n",
       " pillar                 1.4.2      2019-06-29 [1] CRAN (R 3.6.1)                    \n",
       " pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 3.6.1)                    \n",
       " purrr                  0.3.3      2019-10-18 [1] CRAN (R 3.6.1)                    \n",
       " R6                     2.4.1      2019-11-12 [1] CRAN (R 3.6.1)                    \n",
       " Rcpp                   1.0.3      2019-11-08 [1] CRAN (R 3.6.1)                    \n",
       " RCurl                  1.95-4.12  2019-03-04 [1] CRAN (R 3.6.1)                    \n",
       " repr                   1.0.1      2019-05-14 [1] CRAN (R 3.6.1)                    \n",
       " rlang                  0.4.2      2019-11-23 [1] CRAN (R 3.6.1)                    \n",
       " RSQLite                2.1.4      2019-12-04 [1] CRAN (R 3.6.1)                    \n",
       " S4Vectors            * 0.22.1     2019-09-09 [1] Bioconductor                      \n",
       " scales                 1.1.0      2019-11-18 [1] CRAN (R 3.6.1)                    \n",
       " sessioninfo            1.1.1      2018-11-05 [1] CRAN (R 3.6.1)                    \n",
       " SummarizedExperiment * 1.14.1     2019-07-31 [1] Bioconductor                      \n",
       " survival               2.44-1.1   2019-04-01 [2] CRAN (R 3.6.1)                    \n",
       " sva                  * 3.32.1     2019-05-22 [1] Bioconductor                      \n",
       " tibble                 2.1.3      2019-06-06 [1] CRAN (R 3.6.1)                    \n",
       " tidyselect             0.2.5      2018-10-11 [1] CRAN (R 3.6.1)                    \n",
       " uuid                   0.1-2      2015-07-28 [1] CRAN (R 3.6.1)                    \n",
       " vctrs                  0.2.0      2019-07-05 [1] CRAN (R 3.6.1)                    \n",
       " withr                  2.1.2      2018-03-15 [1] CRAN (R 3.6.1)                    \n",
       " XML                    3.98-1.20  2019-06-06 [1] CRAN (R 3.6.1)                    \n",
       " xtable                 1.8-4      2019-04-21 [1] CRAN (R 3.6.1)                    \n",
       " XVector                0.24.0     2019-05-02 [1] Bioconductor                      \n",
       " zeallot                0.1.0      2018-01-28 [1] CRAN (R 3.6.1)                    \n",
       " zlibbioc               1.30.0     2019-05-02 [1] Bioconductor                      \n",
       "\n",
       "[1] /home/jbenja13/R/x86_64-pc-linux-gnu-library/3.6\n",
       "[2] /usr/lib/R/library"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Sys.time()\n",
    "proc.time()\n",
    "options(width = 120)\n",
    "sessioninfo::session_info()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
