{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GO module enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import functools\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import collections as cx\n",
    "from pybiomart import Dataset\n",
    "from gtfparse import read_gtf\n",
    "from scipy.stats import fisher_exact\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "# GO analysis\n",
    "from goatools.base import download_go_basic_obo\n",
    "from goatools.base import download_ncbi_associations\n",
    "from goatools.obo_parser import GODag\n",
    "from goatools.anno.genetogo_reader import Gene2GoReader\n",
    "from goatools.goea.go_enrichment_ns import GOEnrichmentStudyNS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cached functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "@functools.lru_cache()\n",
    "def get_gtf_genes_df():\n",
    "    gtf_df = read_gtf(\"/ceph/genome/human/gencode25/gtf.CHR/_m/gencode.v25.annotation.gtf\")\n",
    "    return gtf_df[gtf_df[\"feature\"] == \"gene\"][['gene_id', 'gene_name']]\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_wgcna_modules():\n",
    "    return pd.read_csv(\"../../_m/modules.csv\", index_col=0)\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_database():\n",
    "    dataset = Dataset(name=\"hsapiens_gene_ensembl\", \n",
    "                      host=\"http://www.ensembl.org\",\n",
    "                      use_cache=True)\n",
    "    db = dataset.query(attributes=[\"ensembl_gene_id\", \n",
    "                                   \"external_gene_name\", \n",
    "                                   \"entrezgene_id\"], \n",
    "                       use_attr_names=True).dropna(subset=['entrezgene_id'])\n",
    "    return db"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def convert2entrez(mod):\n",
    "    df = get_wgcna_modules()[(get_wgcna_modules().module) == mod].copy()\n",
    "    df[\"ensemblID\"] = df.index.str.replace(\"\\\\..*\", \"\", regex=True)\n",
    "    return df.merge(get_database(), left_on='ensemblID', \n",
    "                    right_on='ensembl_gene_id')\n",
    "\n",
    "\n",
    "def obo_annotation(alpha=0.05):\n",
    "    # database annotation\n",
    "    fn_obo = download_go_basic_obo()\n",
    "    fn_gene2go = download_ncbi_associations() # must be gunzip to work\n",
    "    obodag = GODag(fn_obo) # downloads most up-to-date\n",
    "    anno_hs = Gene2GoReader(fn_gene2go, taxids=[9606])\n",
    "    # get associations\n",
    "    ns2assoc = anno_hs.get_ns2assc()\n",
    "    for nspc, id2gos in ns2assoc.items():\n",
    "        print(\"{NS} {N:,} annotated human genes\".format(NS=nspc, N=len(id2gos)))\n",
    "    goeaobj = GOEnrichmentStudyNS(\n",
    "        get_database()['entrezgene_id'], # List of human genes with entrez IDs\n",
    "        ns2assoc, # geneid/GO associations\n",
    "        obodag, # Ontologies\n",
    "        propagate_counts = False,\n",
    "        alpha = alpha, # default significance cut-off\n",
    "        methods = ['fdr_bh'])\n",
    "    return goeaobj\n",
    "\n",
    "\n",
    "def run_goea(mod):\n",
    "    df = convert2entrez(mod)\n",
    "    geneids_study = {z[0]:z[1] for z in zip(df['entrezgene_id'], df['external_gene_name'])}\n",
    "    goeaobj = obo_annotation()\n",
    "    goea_results_all = goeaobj.run_study(geneids_study)\n",
    "    goea_results_sig = [r for r in goea_results_all if r.p_fdr_bh < 0.05]\n",
    "    ctr = cx.Counter([r.NS for r in goea_results_sig])\n",
    "    print('Significant results[{TOTAL}] = {BP} BP + {MF} MF + {CC} CC'.format(\n",
    "        TOTAL=len(goea_results_sig),\n",
    "        BP=ctr['BP'],  # biological_process\n",
    "        MF=ctr['MF'],  # molecular_function\n",
    "        CC=ctr['CC'])) # cellular_component\n",
    "    goeaobj.wr_xlsx(\"GO_analysis_module_%s.xlsx\" % mod, goea_results_sig)\n",
    "    goeaobj.wr_txt(\"GO_analysis_module_%s.txt\" % mod, goea_results_sig)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_status', 'gene_name', 'level', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_status', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'protein_id', 'ccdsid']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id gene_name\n",
       "0   ENSG00000223972.5   DDX11L1\n",
       "12  ENSG00000227232.5    WASH7P"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf = get_gtf_genes_df()\n",
    "gtf.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load WGCNA module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000279457.3</td>\n",
       "      <td>FO538757.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    module            gene_id   gene_name\n",
       "12    grey  ENSG00000227232.5      WASH7P\n",
       "139   grey  ENSG00000279457.3  FO538757.1"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df = get_wgcna_modules().merge(gtf, left_index=True, right_on=\"gene_id\", how=\"left\")\n",
    "wgcna_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [module, gene_id, gene_name]\n",
       "Index: []"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_id.str.startswith(\"chr\"))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1445279</th>\n",
       "      <td>purple</td>\n",
       "      <td>ENSG00000149295.13</td>\n",
       "      <td>DRD2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         module             gene_id gene_name\n",
       "1445279  purple  ENSG00000149295.13      DRD2"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_name == 'DRD2')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "wgcna_df.to_csv(\"module_annotated.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GO enrichment for each cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-3-428e5e18961c>:3: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  df[\"ensemblID\"] = df.index.str.replace(\"\\\\..*\", \"\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "requests.get(http://purl.obolibrary.org/obo/go/go-basic.obo, stream=True)\n",
      "  WROTE: go-basic.obo\n",
      "\n",
      "FTP RETR ftp.ncbi.nlm.nih.gov gene/DATA gene2go.gz -> gene2go.gz\n",
      "  gunzip gene2go.gz\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.108239 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 8079 IDs ... 75%  6,073 of  8,079 study items found in association\n",
      "100%  8,079 of  8,079 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   8,494 GO terms are associated with  6,073 of  8,079 study items\n",
      "  METHOD fdr_bh:\n",
      "      59 GO terms found significant (< 0.05=alpha) ( 54 enriched +   5 purified): statsmodels fdr_bh\n",
      "   2,236 study items associated with significant GO IDs (enriched)\n",
      "     275 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 8079 IDs ... 79%  6,408 of  8,079 study items found in association\n",
      "100%  8,079 of  8,079 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "   1,298 GO terms are associated with  6,408 of  8,079 study items\n",
      "  METHOD fdr_bh:\n",
      "      46 GO terms found significant (< 0.05=alpha) ( 43 enriched +   3 purified): statsmodels fdr_bh\n",
      "   5,873 study items associated with significant GO IDs (enriched)\n",
      "      34 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 8079 IDs ... 77%  6,190 of  8,079 study items found in association\n",
      "100%  8,079 of  8,079 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "   2,906 GO terms are associated with  6,190 of  8,079 study items\n",
      "  METHOD fdr_bh:\n",
      "      34 GO terms found significant (< 0.05=alpha) ( 30 enriched +   4 purified): statsmodels fdr_bh\n",
      "   5,342 study items associated with significant GO IDs (enriched)\n",
      "     119 study items associated with significant GO IDs (purified)\n",
      "Significant results[139] = 59 BP + 34 MF + 46 CC\n",
      "    139 items WROTE: GO_analysis_module_grey.xlsx\n",
      "    139 GOEA results for  6504 study items. WROTE: GO_analysis_module_grey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-3-428e5e18961c>:3: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  df[\"ensemblID\"] = df.index.str.replace(\"\\\\..*\", \"\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.249286 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1092 IDs ... 78%    855 of  1,092 study items found in association\n",
      "100%  1,092 of  1,092 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   3,186 GO terms are associated with    855 of  1,092 study items\n",
      "  METHOD fdr_bh:\n",
      "      62 GO terms found significant (< 0.05=alpha) ( 59 enriched +   3 purified): statsmodels fdr_bh\n",
      "     372 study items associated with significant GO IDs (enriched)\n",
      "       1 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1092 IDs ... 83%    911 of  1,092 study items found in association\n",
      "100%  1,092 of  1,092 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     548 GO terms are associated with    911 of  1,092 study items\n",
      "  METHOD fdr_bh:\n",
      "      68 GO terms found significant (< 0.05=alpha) ( 67 enriched +   1 purified): statsmodels fdr_bh\n",
      "     725 study items associated with significant GO IDs (enriched)\n",
      "      23 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1092 IDs ... 80%    873 of  1,092 study items found in association\n",
      "100%  1,092 of  1,092 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     832 GO terms are associated with    873 of  1,092 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 27 enriched +   3 purified): statsmodels fdr_bh\n",
      "     725 study items associated with significant GO IDs (enriched)\n",
      "      20 study items associated with significant GO IDs (purified)\n",
      "Significant results[160] = 62 BP + 30 MF + 68 CC\n",
      "    160 items WROTE: GO_analysis_module_blue.xlsx\n",
      "    160 GOEA results for   886 study items. WROTE: GO_analysis_module_blue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.951413 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 223 IDs ... 96%    213 of    223 study items found in association\n",
      "100%    223 of    223 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   1,129 GO terms are associated with    213 of    223 study items\n",
      "  METHOD fdr_bh:\n",
      "      32 GO terms found significant (< 0.05=alpha) ( 32 enriched +   0 purified): statsmodels fdr_bh\n",
      "      99 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 223 IDs ... 98%    219 of    223 study items found in association\n",
      "100%    223 of    223 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     348 GO terms are associated with    219 of    223 study items\n",
      "  METHOD fdr_bh:\n",
      "      47 GO terms found significant (< 0.05=alpha) ( 47 enriched +   0 purified): statsmodels fdr_bh\n",
      "     212 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 223 IDs ... 96%    215 of    223 study items found in association\n",
      "100%    223 of    223 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     398 GO terms are associated with    215 of    223 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "     190 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[86] = 32 BP + 7 MF + 47 CC\n",
      "     86 items WROTE: GO_analysis_module_cyan.xlsx\n",
      "     86 GOEA results for   221 study items. WROTE: GO_analysis_module_cyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.343808 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 56 IDs ... 93%     52 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     375 GO terms are associated with     52 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "      45 GO terms found significant (< 0.05=alpha) ( 45 enriched +   0 purified): statsmodels fdr_bh\n",
      "      43 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 56 IDs ... 95%     53 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "      90 GO terms are associated with     53 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      42 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 56 IDs ... 96%     54 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     111 GO terms are associated with     54 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "      13 GO terms found significant (< 0.05=alpha) ( 13 enriched +   0 purified): statsmodels fdr_bh\n",
      "      51 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[63] = 45 BP + 13 MF + 5 CC\n",
      "     63 items WROTE: GO_analysis_module_violet.xlsx\n",
      "     63 GOEA results for    52 study items. WROTE: GO_analysis_module_violet.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.848427 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 225 IDs ... 91%    204 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     950 GO terms are associated with    204 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 225 IDs ... 92%    208 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     284 GO terms are associated with    208 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "     169 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 225 IDs ... 95%    214 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     317 GO terms are associated with    214 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     187 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[8] = 0 BP + 2 MF + 6 CC\n",
      "      8 items WROTE: GO_analysis_module_tan.xlsx\n",
      "      8 GOEA results for   207 study items. WROTE: GO_analysis_module_tan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.146849 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 112 IDs ... 87%     97 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     521 GO terms are associated with     97 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 112 IDs ... 88%     99 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     170 GO terms are associated with     99 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      17 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 112 IDs ... 96%    108 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     192 GO terms are associated with    108 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      91 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 2 MF + 1 CC\n",
      "      3 items WROTE: GO_analysis_module_darkred.xlsx\n",
      "      3 GOEA results for    94 study items. WROTE: GO_analysis_module_darkred.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.153119 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 101 IDs ... 86%     87 of    101 study items found in association\n",
      "100%    101 of    101 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     530 GO terms are associated with     87 of    101 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      15 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 101 IDs ... 90%     91 of    101 study items found in association\n",
      "100%    101 of    101 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     187 GO terms are associated with     91 of    101 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      45 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 101 IDs ... 95%     96 of    101 study items found in association\n",
      "100%    101 of    101 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     196 GO terms are associated with     96 of    101 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      83 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[9] = 2 BP + 4 MF + 3 CC\n",
      "      9 items WROTE: GO_analysis_module_darkgreen.xlsx\n",
      "      9 GOEA results for    87 study items. WROTE: GO_analysis_module_darkgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.199571 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1681 IDs ... 89%  1,495 of  1,681 study items found in association\n",
      "100%  1,681 of  1,681 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   4,616 GO terms are associated with  1,495 of  1,681 study items\n",
      "  METHOD fdr_bh:\n",
      "      81 GO terms found significant (< 0.05=alpha) ( 78 enriched +   3 purified): statsmodels fdr_bh\n",
      "     847 study items associated with significant GO IDs (enriched)\n",
      "      29 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1681 IDs ... 93%  1,555 of  1,681 study items found in association\n",
      "100%  1,681 of  1,681 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     837 GO terms are associated with  1,555 of  1,681 study items\n",
      "  METHOD fdr_bh:\n",
      "      97 GO terms found significant (< 0.05=alpha) ( 97 enriched +   0 purified): statsmodels fdr_bh\n",
      "   1,501 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1681 IDs ... 91%  1,530 of  1,681 study items found in association\n",
      "100%  1,681 of  1,681 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "   1,395 GO terms are associated with  1,530 of  1,681 study items\n",
      "  METHOD fdr_bh:\n",
      "      43 GO terms found significant (< 0.05=alpha) ( 40 enriched +   3 purified): statsmodels fdr_bh\n",
      "   1,386 study items associated with significant GO IDs (enriched)\n",
      "       9 study items associated with significant GO IDs (purified)\n",
      "Significant results[221] = 81 BP + 43 MF + 97 CC\n",
      "    221 items WROTE: GO_analysis_module_turquoise.xlsx\n",
      "    221 GOEA results for  1580 study items. WROTE: GO_analysis_module_turquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.279343 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 222 IDs ... 88%    195 of    222 study items found in association\n",
      "100%    222 of    222 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   1,202 GO terms are associated with    195 of    222 study items\n",
      "  METHOD fdr_bh:\n",
      "      15 GO terms found significant (< 0.05=alpha) ( 15 enriched +   0 purified): statsmodels fdr_bh\n",
      "      50 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 222 IDs ... 90%    200 of    222 study items found in association\n",
      "100%    222 of    222 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     309 GO terms are associated with    200 of    222 study items\n",
      "  METHOD fdr_bh:\n",
      "      27 GO terms found significant (< 0.05=alpha) ( 27 enriched +   0 purified): statsmodels fdr_bh\n",
      "     166 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 222 IDs ... 87%    194 of    222 study items found in association\n",
      "100%    222 of    222 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     359 GO terms are associated with    194 of    222 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "     152 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[51] = 15 BP + 9 MF + 27 CC\n",
      "     51 items WROTE: GO_analysis_module_purple.xlsx\n",
      "     51 GOEA results for   197 study items. WROTE: GO_analysis_module_purple.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.915913 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 352 IDs ... 84%    294 of    352 study items found in association\n",
      "100%    352 of    352 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   1,420 GO terms are associated with    294 of    352 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      48 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 352 IDs ... 89%    315 of    352 study items found in association\n",
      "100%    352 of    352 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     357 GO terms are associated with    315 of    352 study items\n",
      "  METHOD fdr_bh:\n",
      "      23 GO terms found significant (< 0.05=alpha) ( 22 enriched +   1 purified): statsmodels fdr_bh\n",
      "     251 study items associated with significant GO IDs (enriched)\n",
      "       6 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 352 IDs ... 85%    299 of    352 study items found in association\n",
      "100%    352 of    352 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     433 GO terms are associated with    299 of    352 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "     234 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[35] = 8 BP + 4 MF + 23 CC\n",
      "     35 items WROTE: GO_analysis_module_green.xlsx\n",
      "     35 GOEA results for   296 study items. WROTE: GO_analysis_module_green.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.834597 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 81 IDs ... 88%     71 of     81 study items found in association\n",
      "100%     81 of     81 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     532 GO terms are associated with     71 of     81 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 81 IDs ... 93%     75 of     81 study items found in association\n",
      "100%     81 of     81 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     119 GO terms are associated with     75 of     81 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      57 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 81 IDs ... 88%     71 of     81 study items found in association\n",
      "100%     81 of     81 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     178 GO terms are associated with     71 of     81 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 0 BP + 0 MF + 4 CC\n",
      "      4 items WROTE: GO_analysis_module_white.xlsx\n",
      "      4 GOEA results for    57 study items. WROTE: GO_analysis_module_white.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.200278 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 416 IDs ... 89%    371 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   2,066 GO terms are associated with    371 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      63 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 416 IDs ... 94%    391 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     348 GO terms are associated with    391 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "      32 GO terms found significant (< 0.05=alpha) ( 32 enriched +   0 purified): statsmodels fdr_bh\n",
      "     360 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 416 IDs ... 90%    376 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     657 GO terms are associated with    376 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "     306 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[49] = 7 BP + 10 MF + 32 CC\n",
      "     49 items WROTE: GO_analysis_module_yellow.xlsx\n",
      "     49 GOEA results for   384 study items. WROTE: GO_analysis_module_yellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.156049 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 330 IDs ... 91%    301 of    330 study items found in association\n",
      "100%    330 of    330 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   1,587 GO terms are associated with    301 of    330 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      41 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 330 IDs ... 93%    307 of    330 study items found in association\n",
      "100%    330 of    330 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     329 GO terms are associated with    307 of    330 study items\n",
      "  METHOD fdr_bh:\n",
      "      24 GO terms found significant (< 0.05=alpha) ( 24 enriched +   0 purified): statsmodels fdr_bh\n",
      "     207 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 330 IDs ... 91%    299 of    330 study items found in association\n",
      "100%    330 of    330 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     493 GO terms are associated with    299 of    330 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     214 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[30] = 4 BP + 2 MF + 24 CC\n",
      "     30 items WROTE: GO_analysis_module_red.xlsx\n",
      "     30 GOEA results for   288 study items. WROTE: GO_analysis_module_red.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.216528 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 73 IDs ... 97%     71 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     277 GO terms are associated with     71 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "      19 GO terms found significant (< 0.05=alpha) ( 19 enriched +   0 purified): statsmodels fdr_bh\n",
      "      54 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 73 IDs ... 97%     71 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "      98 GO terms are associated with     71 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 22 enriched +   0 purified): statsmodels fdr_bh\n",
      "      68 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 73 IDs ... 96%     70 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "      77 GO terms are associated with     70 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "      68 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[51] = 19 BP + 10 MF + 22 CC\n",
      "     51 items WROTE: GO_analysis_module_skyblue.xlsx\n",
      "     51 GOEA results for    71 study items. WROTE: GO_analysis_module_skyblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.281857 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 52 IDs ... 94%     49 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     348 GO terms are associated with     49 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "      13 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 52 IDs ... 98%     51 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     146 GO terms are associated with     51 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "      15 GO terms found significant (< 0.05=alpha) ( 15 enriched +   0 purified): statsmodels fdr_bh\n",
      "      48 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 52 IDs ... 98%     51 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     126 GO terms are associated with     51 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      47 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[30] = 11 BP + 4 MF + 15 CC\n",
      "     30 items WROTE: GO_analysis_module_darkmagenta.xlsx\n",
      "     30 GOEA results for    51 study items. WROTE: GO_analysis_module_darkmagenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.949910 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 72 IDs ... 86%     62 of     72 study items found in association\n",
      "100%     72 of     72 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     479 GO terms are associated with     62 of     72 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 72 IDs ... 89%     64 of     72 study items found in association\n",
      "100%     72 of     72 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     121 GO terms are associated with     64 of     72 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      29 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 72 IDs ... 82%     59 of     72 study items found in association\n",
      "100%     72 of     72 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     172 GO terms are associated with     59 of     72 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 0 MF + 3 CC\n",
      "      3 items WROTE: GO_analysis_module_saddlebrown.xlsx\n",
      "      3 GOEA results for    29 study items. WROTE: GO_analysis_module_saddlebrown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.191334 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 100 IDs ... 94%     94 of    100 study items found in association\n",
      "100%    100 of    100 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   1,032 GO terms are associated with     94 of    100 study items\n",
      "  METHOD fdr_bh:\n",
      "      62 GO terms found significant (< 0.05=alpha) ( 62 enriched +   0 purified): statsmodels fdr_bh\n",
      "      76 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 100 IDs ... 95%     95 of    100 study items found in association\n",
      "100%    100 of    100 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     178 GO terms are associated with     95 of    100 study items\n",
      "  METHOD fdr_bh:\n",
      "      18 GO terms found significant (< 0.05=alpha) ( 18 enriched +   0 purified): statsmodels fdr_bh\n",
      "      77 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 100 IDs ... 95%     95 of    100 study items found in association\n",
      "100%    100 of    100 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     293 GO terms are associated with     95 of    100 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      78 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[85] = 62 BP + 5 MF + 18 CC\n",
      "     85 items WROTE: GO_analysis_module_darkturquoise.xlsx\n",
      "     85 GOEA results for    95 study items. WROTE: GO_analysis_module_darkturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.209102 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 164 IDs ... 91%    150 of    164 study items found in association\n",
      "100%    164 of    164 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   1,263 GO terms are associated with    150 of    164 study items\n",
      "  METHOD fdr_bh:\n",
      "     192 GO terms found significant (< 0.05=alpha) (192 enriched +   0 purified): statsmodels fdr_bh\n",
      "     135 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 164 IDs ... 98%    160 of    164 study items found in association\n",
      "100%    164 of    164 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     200 GO terms are associated with    160 of    164 study items\n",
      "  METHOD fdr_bh:\n",
      "      41 GO terms found significant (< 0.05=alpha) ( 41 enriched +   0 purified): statsmodels fdr_bh\n",
      "     151 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 164 IDs ... 93%    153 of    164 study items found in association\n",
      "100%    164 of    164 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     297 GO terms are associated with    153 of    164 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 22 enriched +   0 purified): statsmodels fdr_bh\n",
      "     137 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[255] = 192 BP + 22 MF + 41 CC\n",
      "    255 items WROTE: GO_analysis_module_grey60.xlsx\n",
      "    255 GOEA results for   160 study items. WROTE: GO_analysis_module_grey60.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.172686 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 898 IDs ... 83%    748 of    898 study items found in association\n",
      "100%    898 of    898 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   2,932 GO terms are associated with    748 of    898 study items\n",
      "  METHOD fdr_bh:\n",
      "      60 GO terms found significant (< 0.05=alpha) ( 58 enriched +   2 purified): statsmodels fdr_bh\n",
      "     318 study items associated with significant GO IDs (enriched)\n",
      "       3 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 898 IDs ... 88%    792 of    898 study items found in association\n",
      "100%    898 of    898 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     551 GO terms are associated with    792 of    898 study items\n",
      "  METHOD fdr_bh:\n",
      "      72 GO terms found significant (< 0.05=alpha) ( 72 enriched +   0 purified): statsmodels fdr_bh\n",
      "     729 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 898 IDs ... 85%    766 of    898 study items found in association\n",
      "100%    898 of    898 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     844 GO terms are associated with    766 of    898 study items\n",
      "  METHOD fdr_bh:\n",
      "      32 GO terms found significant (< 0.05=alpha) ( 31 enriched +   1 purified): statsmodels fdr_bh\n",
      "     601 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[164] = 60 BP + 32 MF + 72 CC\n",
      "    164 items WROTE: GO_analysis_module_brown.xlsx\n",
      "    164 GOEA results for   798 study items. WROTE: GO_analysis_module_brown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.912858 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 48 IDs ... 90%     43 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     395 GO terms are associated with     43 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       4 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 48 IDs ... 94%     45 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     135 GO terms are associated with     45 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "      20 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 48 IDs ... 94%     45 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     110 GO terms are associated with     45 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      36 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[13] = 1 BP + 1 MF + 11 CC\n",
      "     13 items WROTE: GO_analysis_module_sienna3.xlsx\n",
      "     13 GOEA results for    43 study items. WROTE: GO_analysis_module_sienna3.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.308728 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 196 IDs ... 92%    181 of    196 study items found in association\n",
      "100%    196 of    196 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "   1,503 GO terms are associated with    181 of    196 study items\n",
      "  METHOD fdr_bh:\n",
      "      40 GO terms found significant (< 0.05=alpha) ( 40 enriched +   0 purified): statsmodels fdr_bh\n",
      "      83 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 196 IDs ... 93%    182 of    196 study items found in association\n",
      "100%    196 of    196 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     241 GO terms are associated with    182 of    196 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "     147 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 196 IDs ... 91%    179 of    196 study items found in association\n",
      "100%    196 of    196 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     354 GO terms are associated with    179 of    196 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "     143 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[62] = 40 BP + 11 MF + 11 CC\n",
      "     62 items WROTE: GO_analysis_module_lightcyan.xlsx\n",
      "     62 GOEA results for   170 study items. WROTE: GO_analysis_module_lightcyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.234889 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 154 IDs ... 91%    140 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     983 GO terms are associated with    140 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      36 GO terms found significant (< 0.05=alpha) ( 36 enriched +   0 purified): statsmodels fdr_bh\n",
      "      67 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 154 IDs ... 95%    146 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     246 GO terms are associated with    146 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      28 GO terms found significant (< 0.05=alpha) ( 28 enriched +   0 purified): statsmodels fdr_bh\n",
      "     137 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 154 IDs ... 96%    148 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     310 GO terms are associated with    148 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      20 GO terms found significant (< 0.05=alpha) ( 20 enriched +   0 purified): statsmodels fdr_bh\n",
      "     136 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[84] = 36 BP + 20 MF + 28 CC\n",
      "     84 items WROTE: GO_analysis_module_lightgreen.xlsx\n",
      "     84 GOEA results for   149 study items. WROTE: GO_analysis_module_lightgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.129901 341,880 annotations, 20,685 genes, 18,610 GOs, 1 taxids READ: gene2go \n",
      "BP 18,709 annotated human genes\n",
      "CC 19,421 annotated human genes\n",
      "MF 18,179 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 71% 20,538 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,427 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,342 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 138 IDs ... 93%    129 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 12,433 uncorrected p-values using fisher\n",
      "  12,433 GO terms are associated with 18,051 of 29,107 population items\n",
      "     929 GO terms are associated with    129 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 26 enriched +   0 purified): statsmodels fdr_bh\n",
      "      50 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 138 IDs ... 96%    133 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 1,755 uncorrected p-values using fisher\n",
      "   1,755 GO terms are associated with 18,710 of 29,107 population items\n",
      "     264 GO terms are associated with    133 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "      19 GO terms found significant (< 0.05=alpha) ( 19 enriched +   0 purified): statsmodels fdr_bh\n",
      "     124 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 138 IDs ... 94%    130 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 4,361 uncorrected p-values using fisher\n",
      "   4,361 GO terms are associated with 17,827 of 29,107 population items\n",
      "     288 GO terms are associated with    130 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     121 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[50] = 26 BP + 5 MF + 19 CC\n",
      "     50 items WROTE: GO_analysis_module_royalblue.xlsx\n",
      "     50 GOEA results for   133 study items. WROTE: GO_analysis_module_royalblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-738bac64b246>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mmod\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mget_wgcna_modules\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmodule\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mrun_goea\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmod\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-3-428e5e18961c>\u001b[0m in \u001b[0;36mrun_goea\u001b[0;34m(mod)\u001b[0m\n\u001b[1;32m     29\u001b[0m     \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconvert2entrez\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmod\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     30\u001b[0m     \u001b[0mgeneids_study\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0mz\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mz\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mz\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mzip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'entrezgene_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'external_gene_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 31\u001b[0;31m     \u001b[0mgoeaobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mobo_annotation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     32\u001b[0m     \u001b[0mgoea_results_all\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgoeaobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_study\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgeneids_study\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m     \u001b[0mgoea_results_sig\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mr\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mr\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgoea_results_all\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mp_fdr_bh\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m0.05\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-3-428e5e18961c>\u001b[0m in \u001b[0;36mobo_annotation\u001b[0;34m(alpha)\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0mfn_gene2go\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdownload_ncbi_associations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# must be gunzip to work\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m     \u001b[0mobodag\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mGODag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfn_obo\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# downloads most up-to-date\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 13\u001b[0;31m     \u001b[0manno_hs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mGene2GoReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfn_gene2go\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtaxids\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m9606\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     14\u001b[0m     \u001b[0;31m# get associations\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0mns2assoc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0manno_hs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_ns2assc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.9/site-packages/goatools/anno/genetogo_reader.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, filename, **kws)\u001b[0m\n\u001b[1;32m     25\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkws\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m         \u001b[0;31m# kws: taxids or taxid\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 27\u001b[0;31m         \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mGene2GoReader\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'gene2go'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkws\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     28\u001b[0m         \u001b[0;31m# Each taxid has a list of namedtuples - one for each line in the annotations\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     29\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtaxid2asscs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_init_taxid2asscs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.9/site-packages/goatools/anno/annoreader_base.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, name, filename, **kws)\u001b[0m\n\u001b[1;32m     45\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdatobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     46\u001b[0m         \u001b[0;31m# pylint: disable=no-member\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 47\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0massociations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_init_associations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkws\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     48\u001b[0m         \u001b[0;31m# assert self.associations, 'NO ANNOTATIONS FOUND: {ANNO}'.format(ANNO=filename)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     49\u001b[0m         \u001b[0;32massert\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnamespaces\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnamespaces\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.9/site-packages/goatools/anno/genetogo_reader.py\u001b[0m in \u001b[0;36m_init_associations\u001b[0;34m(fin_anno, taxid, taxids, namespaces, **kws)\u001b[0m\n\u001b[1;32m    155\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_init_associations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfin_anno\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtaxid\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtaxids\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespaces\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkws\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    156\u001b[0m         \u001b[0;34m\"\"\"Read annotation file and store a list of namedtuples.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 157\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mInitAssc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtaxid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtaxids\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minit_associations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfin_anno\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtaxids\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespaces\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    158\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    159\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_init_taxid2asscs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.9/site-packages/goatools/anno/init/reader_genetogo.py\u001b[0m in \u001b[0;36minit_associations\u001b[0;34m(self, fin_anno, taxids, namespaces)\u001b[0m\n\u001b[1;32m     88\u001b[0m                                 \u001b[0mQualifier\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_qualifiers\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvals\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     89\u001b[0m                                 \u001b[0mGO_term\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mvals\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 90\u001b[0;31m                                 \u001b[0mDB_Reference\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_pmids\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvals\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m6\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     91\u001b[0m                                 NS=nspc)\n\u001b[1;32m     92\u001b[0m                             \u001b[0;31m#self._chk_qualifiers(qualifiers, lnum, ntd)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.9/site-packages/goatools/anno/init/reader_genetogo.py\u001b[0m in \u001b[0;36m_get_pmids\u001b[0;34m(pmidstr)\u001b[0m\n\u001b[1;32m    121\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mpmidstr\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'-'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    122\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 123\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'PMID:{N}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mN\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mn\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mpmidstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'|'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    124\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    125\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_prt_line_detail\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlnum\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.9/site-packages/goatools/anno/init/reader_genetogo.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    121\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mpmidstr\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'-'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    122\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 123\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'PMID:{N}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mN\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mn\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mpmidstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'|'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    124\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    125\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_prt_line_detail\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlnum\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for mod in get_wgcna_modules().module.unique():\n",
    "    run_goea(mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
