{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GO module enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:46:56.820445Z",
     "iopub.status.busy": "2021-07-12T14:46:56.818191Z",
     "iopub.status.idle": "2021-07-12T14:46:57.464386Z",
     "shell.execute_reply": "2021-07-12T14:46:57.463774Z"
    }
   },
   "outputs": [],
   "source": [
    "import functools\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import collections as cx\n",
    "from pybiomart import Dataset\n",
    "from gtfparse import read_gtf\n",
    "from scipy.stats import fisher_exact\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "# GO analysis\n",
    "from goatools.base import download_go_basic_obo\n",
    "from goatools.base import download_ncbi_associations\n",
    "from goatools.obo_parser import GODag\n",
    "from goatools.anno.genetogo_reader import Gene2GoReader\n",
    "from goatools.goea.go_enrichment_ns import GOEnrichmentStudyNS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cached functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:46:57.471380Z",
     "iopub.status.busy": "2021-07-12T14:46:57.470563Z",
     "iopub.status.idle": "2021-07-12T14:46:57.473315Z",
     "shell.execute_reply": "2021-07-12T14:46:57.472817Z"
    }
   },
   "outputs": [],
   "source": [
    "@functools.lru_cache()\n",
    "def get_gtf_genes_df():\n",
    "    gtf_df = read_gtf(\"/ceph/genome/human/gencode25/gtf.CHR/_m/gencode.v25.annotation.gtf\")\n",
    "    return gtf_df[gtf_df[\"feature\"] == \"gene\"][['gene_id', 'gene_name']]\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_wgcna_modules():\n",
    "    return pd.read_csv(\"../../_m/modules.csv\", index_col=0)\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_database():\n",
    "    dataset = Dataset(name=\"hsapiens_gene_ensembl\", \n",
    "                      host=\"http://www.ensembl.org\",\n",
    "                      use_cache=True)\n",
    "    db = dataset.query(attributes=[\"ensembl_gene_id\", \n",
    "                                   \"external_gene_name\", \n",
    "                                   \"entrezgene_id\"], \n",
    "                       use_attr_names=True).dropna(subset=['entrezgene_id'])\n",
    "    return db"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:46:57.483593Z",
     "iopub.status.busy": "2021-07-12T14:46:57.482764Z",
     "iopub.status.idle": "2021-07-12T14:46:57.485045Z",
     "shell.execute_reply": "2021-07-12T14:46:57.485492Z"
    }
   },
   "outputs": [],
   "source": [
    "def convert2entrez(mod):\n",
    "    df = get_wgcna_modules()[(get_wgcna_modules().module) == mod].copy()\n",
    "    df[\"ensemblID\"] = df.index.str.replace(\"\\\\..*\", \"\", regex=True)\n",
    "    return df.merge(get_database(), left_on='ensemblID', \n",
    "                    right_on='ensembl_gene_id')\n",
    "\n",
    "\n",
    "def obo_annotation(alpha=0.05):\n",
    "    # database annotation\n",
    "    fn_obo = download_go_basic_obo()\n",
    "    fn_gene2go = download_ncbi_associations() # must be gunzip to work\n",
    "    obodag = GODag(fn_obo) # downloads most up-to-date\n",
    "    anno_hs = Gene2GoReader(fn_gene2go, taxids=[9606])\n",
    "    # get associations\n",
    "    ns2assoc = anno_hs.get_ns2assc()\n",
    "    for nspc, id2gos in ns2assoc.items():\n",
    "        print(\"{NS} {N:,} annotated human genes\".format(NS=nspc, N=len(id2gos)))\n",
    "    goeaobj = GOEnrichmentStudyNS(\n",
    "        get_database()['entrezgene_id'], # List of human genes with entrez IDs\n",
    "        ns2assoc, # geneid/GO associations\n",
    "        obodag, # Ontologies\n",
    "        propagate_counts = False,\n",
    "        alpha = alpha, # default significance cut-off\n",
    "        methods = ['fdr_bh'])\n",
    "    return goeaobj\n",
    "\n",
    "\n",
    "def run_goea(mod):\n",
    "    df = convert2entrez(mod)\n",
    "    geneids_study = {z[0]:z[1] for z in zip(df['entrezgene_id'], df['external_gene_name'])}\n",
    "    goeaobj = obo_annotation()\n",
    "    goea_results_all = goeaobj.run_study(geneids_study)\n",
    "    goea_results_sig = [r for r in goea_results_all if r.p_fdr_bh < 0.05]\n",
    "    ctr = cx.Counter([r.NS for r in goea_results_sig])\n",
    "    print('Significant results[{TOTAL}] = {BP} BP + {MF} MF + {CC} CC'.format(\n",
    "        TOTAL=len(goea_results_sig),\n",
    "        BP=ctr['BP'],  # biological_process\n",
    "        MF=ctr['MF'],  # molecular_function\n",
    "        CC=ctr['CC'])) # cellular_component\n",
    "    goeaobj.wr_xlsx(\"GO_analysis_module_%s.xlsx\" % mod, goea_results_sig)\n",
    "    goeaobj.wr_txt(\"GO_analysis_module_%s.txt\" % mod, goea_results_sig)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:46:57.490814Z",
     "iopub.status.busy": "2021-07-12T14:46:57.489416Z",
     "iopub.status.idle": "2021-07-12T14:48:10.336242Z",
     "shell.execute_reply": "2021-07-12T14:48:10.337295Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_status', 'gene_name', 'level', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_status', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'protein_id', 'ccdsid']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id gene_name\n",
       "0   ENSG00000223972.5   DDX11L1\n",
       "12  ENSG00000227232.5    WASH7P"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf = get_gtf_genes_df()\n",
    "gtf.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load WGCNA module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:48:10.344196Z",
     "iopub.status.busy": "2021-07-12T14:48:10.342814Z",
     "iopub.status.idle": "2021-07-12T14:48:10.402431Z",
     "shell.execute_reply": "2021-07-12T14:48:10.403473Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000279457.3</td>\n",
       "      <td>FO538757.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    module            gene_id   gene_name\n",
       "12    grey  ENSG00000227232.5      WASH7P\n",
       "139   grey  ENSG00000279457.3  FO538757.1"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df = get_wgcna_modules().merge(gtf, left_index=True, right_on=\"gene_id\", how=\"left\")\n",
    "wgcna_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:48:10.422810Z",
     "iopub.status.busy": "2021-07-12T14:48:10.420223Z",
     "iopub.status.idle": "2021-07-12T14:48:10.430912Z",
     "shell.execute_reply": "2021-07-12T14:48:10.431924Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [module, gene_id, gene_name]\n",
       "Index: []"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_id.str.startswith(\"chr\"))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:48:10.444427Z",
     "iopub.status.busy": "2021-07-12T14:48:10.440497Z",
     "iopub.status.idle": "2021-07-12T14:48:10.447426Z",
     "shell.execute_reply": "2021-07-12T14:48:10.448434Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1445279</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000149295.13</td>\n",
       "      <td>DRD2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        module             gene_id gene_name\n",
       "1445279   grey  ENSG00000149295.13      DRD2"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_name == 'DRD2')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:48:10.454137Z",
     "iopub.status.busy": "2021-07-12T14:48:10.452853Z",
     "iopub.status.idle": "2021-07-12T14:48:10.507132Z",
     "shell.execute_reply": "2021-07-12T14:48:10.505971Z"
    }
   },
   "outputs": [],
   "source": [
    "wgcna_df.to_csv(\"module_annotated.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GO enrichment for each cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T14:48:10.514066Z",
     "iopub.status.busy": "2021-07-12T14:48:10.512770Z",
     "iopub.status.idle": "2021-07-12T14:53:14.555529Z",
     "shell.execute_reply": "2021-07-12T14:53:14.554951Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "requests.get(http://purl.obolibrary.org/obo/go/go-basic.obo, stream=True)\n",
      "  WROTE: go-basic.obo\n",
      "\n",
      "FTP RETR ftp.ncbi.nlm.nih.gov gene/DATA gene2go.gz -> gene2go.gz\n",
      "  gunzip gene2go.gz\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.049954 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 13472 IDs ... 81% 10,935 of 13,472 study items found in association\n",
      "100% 13,472 of 13,472 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "  10,697 GO terms are associated with 10,935 of 13,472 study items\n",
      "  METHOD fdr_bh:\n",
      "     632 GO terms found significant (< 0.05=alpha) (612 enriched +  20 purified): statsmodels fdr_bh\n",
      "   9,202 study items associated with significant GO IDs (enriched)\n",
      "     407 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 13472 IDs ... 86% 11,525 of 13,472 study items found in association\n",
      "100% 13,472 of 13,472 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "   1,606 GO terms are associated with 11,525 of 13,472 study items\n",
      "  METHOD fdr_bh:\n",
      "     371 GO terms found significant (< 0.05=alpha) (363 enriched +   8 purified): statsmodels fdr_bh\n",
      "  11,275 study items associated with significant GO IDs (enriched)\n",
      "     716 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 13472 IDs ... 83% 11,185 of 13,472 study items found in association\n",
      "100% 13,472 of 13,472 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   3,822 GO terms are associated with 11,185 of 13,472 study items\n",
      "  METHOD fdr_bh:\n",
      "     270 GO terms found significant (< 0.05=alpha) (261 enriched +   9 purified): statsmodels fdr_bh\n",
      "  10,403 study items associated with significant GO IDs (enriched)\n",
      "     246 study items associated with significant GO IDs (purified)\n",
      "Significant results[1273] = 632 BP + 270 MF + 371 CC\n",
      "   1273 items WROTE: GO_analysis_module_grey.xlsx\n",
      "   1273 GOEA results for 11957 study items. WROTE: GO_analysis_module_grey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.712296 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 146 IDs ... 90%    131 of    146 study items found in association\n",
      "100%    146 of    146 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     781 GO terms are associated with    131 of    146 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      12 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 146 IDs ... 92%    135 of    146 study items found in association\n",
      "100%    146 of    146 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     210 GO terms are associated with    135 of    146 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 21 enriched +   0 purified): statsmodels fdr_bh\n",
      "     107 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 146 IDs ... 89%    130 of    146 study items found in association\n",
      "100%    146 of    146 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     247 GO terms are associated with    130 of    146 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     102 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[33] = 7 BP + 5 MF + 21 CC\n",
      "     33 items WROTE: GO_analysis_module_pink.xlsx\n",
      "     33 GOEA results for   125 study items. WROTE: GO_analysis_module_pink.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.936176 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 39 IDs ... 82%     32 of     39 study items found in association\n",
      "100%     39 of     39 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     261 GO terms are associated with     32 of     39 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "       4 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 39 IDs ... 90%     35 of     39 study items found in association\n",
      "100%     39 of     39 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      98 GO terms are associated with     35 of     39 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "       7 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 39 IDs ... 85%     33 of     39 study items found in association\n",
      "100%     39 of     39 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      82 GO terms are associated with     33 of     39 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[5] = 2 BP + 0 MF + 3 CC\n",
      "      5 items WROTE: GO_analysis_module_darkgreen.xlsx\n",
      "      5 GOEA results for     7 study items. WROTE: GO_analysis_module_darkgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.487098 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 156 IDs ... 97%    152 of    156 study items found in association\n",
      "100%    156 of    156 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,026 GO terms are associated with    152 of    156 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 22 enriched +   0 purified): statsmodels fdr_bh\n",
      "      75 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 156 IDs ... 99%    154 of    156 study items found in association\n",
      "100%    156 of    156 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     187 GO terms are associated with    154 of    156 study items\n",
      "  METHOD fdr_bh:\n",
      "      27 GO terms found significant (< 0.05=alpha) ( 27 enriched +   0 purified): statsmodels fdr_bh\n",
      "     130 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 156 IDs ... 96%    149 of    156 study items found in association\n",
      "100%    156 of    156 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     258 GO terms are associated with    149 of    156 study items\n",
      "  METHOD fdr_bh:\n",
      "      20 GO terms found significant (< 0.05=alpha) ( 20 enriched +   0 purified): statsmodels fdr_bh\n",
      "     134 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[69] = 22 BP + 20 MF + 27 CC\n",
      "     69 items WROTE: GO_analysis_module_black.xlsx\n",
      "     69 GOEA results for   149 study items. WROTE: GO_analysis_module_black.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.659221 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 464 IDs ... 86%    397 of    464 study items found in association\n",
      "100%    464 of    464 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,997 GO terms are associated with    397 of    464 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "      90 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 464 IDs ... 89%    412 of    464 study items found in association\n",
      "100%    464 of    464 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     407 GO terms are associated with    412 of    464 study items\n",
      "  METHOD fdr_bh:\n",
      "      42 GO terms found significant (< 0.05=alpha) ( 42 enriched +   0 purified): statsmodels fdr_bh\n",
      "     374 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 464 IDs ... 88%    408 of    464 study items found in association\n",
      "100%    464 of    464 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     614 GO terms are associated with    408 of    464 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     305 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[58] = 11 BP + 5 MF + 42 CC\n",
      "     58 items WROTE: GO_analysis_module_turquoise.xlsx\n",
      "     58 GOEA results for   413 study items. WROTE: GO_analysis_module_turquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.559263 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 463 IDs ... 67%    311 of    463 study items found in association\n",
      "100%    463 of    463 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,061 GO terms are associated with    311 of    463 study items\n",
      "  METHOD fdr_bh:\n",
      "      31 GO terms found significant (< 0.05=alpha) ( 30 enriched +   1 purified): statsmodels fdr_bh\n",
      "     137 study items associated with significant GO IDs (enriched)\n",
      "       1 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 463 IDs ... 75%    345 of    463 study items found in association\n",
      "100%    463 of    463 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     265 GO terms are associated with    345 of    463 study items\n",
      "  METHOD fdr_bh:\n",
      "      33 GO terms found significant (< 0.05=alpha) ( 29 enriched +   4 purified): statsmodels fdr_bh\n",
      "     221 study items associated with significant GO IDs (enriched)\n",
      "      65 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 463 IDs ... 69%    321 of    463 study items found in association\n",
      "100%    463 of    463 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     345 GO terms are associated with    321 of    463 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) ( 11 enriched +   1 purified): statsmodels fdr_bh\n",
      "      71 study items associated with significant GO IDs (enriched)\n",
      "       3 study items associated with significant GO IDs (purified)\n",
      "Significant results[76] = 31 BP + 12 MF + 33 CC\n",
      "     76 items WROTE: GO_analysis_module_blue.xlsx\n",
      "     76 GOEA results for   271 study items. WROTE: GO_analysis_module_blue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.626144 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 206 IDs ... 78%    161 of    206 study items found in association\n",
      "100%    206 of    206 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,033 GO terms are associated with    161 of    206 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 206 IDs ... 83%    171 of    206 study items found in association\n",
      "100%    206 of    206 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     242 GO terms are associated with    171 of    206 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      59 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 206 IDs ... 80%    165 of    206 study items found in association\n",
      "100%    206 of    206 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     300 GO terms are associated with    165 of    206 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[2] = 0 BP + 0 MF + 2 CC\n",
      "      2 items WROTE: GO_analysis_module_green.xlsx\n",
      "      2 GOEA results for    59 study items. WROTE: GO_analysis_module_green.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.371620 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 308 IDs ... 81%    249 of    308 study items found in association\n",
      "100%    308 of    308 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,346 GO terms are associated with    249 of    308 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 308 IDs ... 87%    268 of    308 study items found in association\n",
      "100%    308 of    308 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     307 GO terms are associated with    268 of    308 study items\n",
      "  METHOD fdr_bh:\n",
      "      13 GO terms found significant (< 0.05=alpha) ( 13 enriched +   0 purified): statsmodels fdr_bh\n",
      "     158 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 308 IDs ... 84%    260 of    308 study items found in association\n",
      "100%    308 of    308 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     470 GO terms are associated with    260 of    308 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      27 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[14] = 0 BP + 1 MF + 13 CC\n",
      "     14 items WROTE: GO_analysis_module_yellow.xlsx\n",
      "     14 GOEA results for   166 study items. WROTE: GO_analysis_module_yellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.645400 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 91 IDs ... 88%     80 of     91 study items found in association\n",
      "100%     91 of     91 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     544 GO terms are associated with     80 of     91 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      29 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 91 IDs ... 95%     86 of     91 study items found in association\n",
      "100%     91 of     91 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     185 GO terms are associated with     86 of     91 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 22 enriched +   0 purified): statsmodels fdr_bh\n",
      "      58 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 91 IDs ... 89%     81 of     91 study items found in association\n",
      "100%     91 of     91 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     197 GO terms are associated with     81 of     91 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "       9 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[30] = 5 BP + 3 MF + 22 CC\n",
      "     30 items WROTE: GO_analysis_module_purple.xlsx\n",
      "     30 GOEA results for    62 study items. WROTE: GO_analysis_module_purple.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.379890 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 29 IDs ... 83%     24 of     29 study items found in association\n",
      "100%     29 of     29 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     293 GO terms are associated with     24 of     29 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 29 IDs ... 83%     24 of     29 study items found in association\n",
      "100%     29 of     29 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      88 GO terms are associated with     24 of     29 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      10 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 29 IDs ... 86%     25 of     29 study items found in association\n",
      "100%     29 of     29 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      81 GO terms are associated with     25 of     29 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[7] = 0 BP + 0 MF + 7 CC\n",
      "      7 items WROTE: GO_analysis_module_orange.xlsx\n",
      "      7 GOEA results for    10 study items. WROTE: GO_analysis_module_orange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.650817 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 70 IDs ... 87%     61 of     70 study items found in association\n",
      "100%     70 of     70 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     404 GO terms are associated with     61 of     70 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "       7 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 70 IDs ... 91%     64 of     70 study items found in association\n",
      "100%     70 of     70 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     155 GO terms are associated with     64 of     70 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      18 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 70 IDs ... 84%     59 of     70 study items found in association\n",
      "100%     70 of     70 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     143 GO terms are associated with     59 of     70 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "       8 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[14] = 4 BP + 3 MF + 7 CC\n",
      "     14 items WROTE: GO_analysis_module_greenyellow.xlsx\n",
      "     14 GOEA results for    19 study items. WROTE: GO_analysis_module_greenyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.644683 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 54 IDs ... 91%     49 of     54 study items found in association\n",
      "100%     54 of     54 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     407 GO terms are associated with     49 of     54 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      11 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 54 IDs ... 96%     52 of     54 study items found in association\n",
      "100%     54 of     54 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     172 GO terms are associated with     52 of     54 study items\n",
      "  METHOD fdr_bh:\n",
      "      19 GO terms found significant (< 0.05=alpha) ( 19 enriched +   0 purified): statsmodels fdr_bh\n",
      "      49 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 54 IDs ... 91%     49 of     54 study items found in association\n",
      "100%     54 of     54 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     157 GO terms are associated with     49 of     54 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "      44 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[29] = 4 BP + 6 MF + 19 CC\n",
      "     29 items WROTE: GO_analysis_module_midnightblue.xlsx\n",
      "     29 GOEA results for    52 study items. WROTE: GO_analysis_module_midnightblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.313489 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 405 IDs ... 86%    349 of    405 study items found in association\n",
      "100%    405 of    405 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,052 GO terms are associated with    349 of    405 study items\n",
      "  METHOD fdr_bh:\n",
      "     240 GO terms found significant (< 0.05=alpha) (239 enriched +   1 purified): statsmodels fdr_bh\n",
      "     291 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 405 IDs ... 93%    375 of    405 study items found in association\n",
      "100%    405 of    405 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     300 GO terms are associated with    375 of    405 study items\n",
      "  METHOD fdr_bh:\n",
      "      69 GO terms found significant (< 0.05=alpha) ( 69 enriched +   0 purified): statsmodels fdr_bh\n",
      "     354 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 405 IDs ... 90%    366 of    405 study items found in association\n",
      "100%    405 of    405 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     557 GO terms are associated with    366 of    405 study items\n",
      "  METHOD fdr_bh:\n",
      "      55 GO terms found significant (< 0.05=alpha) ( 55 enriched +   0 purified): statsmodels fdr_bh\n",
      "     323 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[364] = 240 BP + 55 MF + 69 CC\n",
      "    364 items WROTE: GO_analysis_module_brown.xlsx\n",
      "    364 GOEA results for   373 study items. WROTE: GO_analysis_module_brown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.560230 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 186 IDs ... 90%    167 of    186 study items found in association\n",
      "100%    186 of    186 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,195 GO terms are associated with    167 of    186 study items\n",
      "  METHOD fdr_bh:\n",
      "      13 GO terms found significant (< 0.05=alpha) ( 13 enriched +   0 purified): statsmodels fdr_bh\n",
      "      46 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 186 IDs ... 92%    171 of    186 study items found in association\n",
      "100%    186 of    186 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     196 GO terms are associated with    171 of    186 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "     138 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 186 IDs ... 89%    166 of    186 study items found in association\n",
      "100%    186 of    186 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     343 GO terms are associated with    166 of    186 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "     118 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[35] = 13 BP + 6 MF + 16 CC\n",
      "     35 items WROTE: GO_analysis_module_red.xlsx\n",
      "     35 GOEA results for   163 study items. WROTE: GO_analysis_module_red.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.396688 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 45 IDs ... 98%     44 of     45 study items found in association\n",
      "100%     45 of     45 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     536 GO terms are associated with     44 of     45 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) ( 12 enriched +   0 purified): statsmodels fdr_bh\n",
      "      34 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 45 IDs ... 98%     44 of     45 study items found in association\n",
      "100%     45 of     45 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      94 GO terms are associated with     44 of     45 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      26 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 45 IDs ... 98%     44 of     45 study items found in association\n",
      "100%     45 of     45 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     132 GO terms are associated with     44 of     45 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      39 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[18] = 12 BP + 4 MF + 2 CC\n",
      "     18 items WROTE: GO_analysis_module_lightyellow.xlsx\n",
      "     18 GOEA results for    43 study items. WROTE: GO_analysis_module_lightyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.574669 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 19 IDs ...100%     19 of     19 study items found in association\n",
      "100%     19 of     19 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     164 GO terms are associated with     19 of     19 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 19 IDs ...100%     19 of     19 study items found in association\n",
      "100%     19 of     19 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      61 GO terms are associated with     19 of     19 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 19 IDs ...100%     19 of     19 study items found in association\n",
      "100%     19 of     19 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      64 GO terms are associated with     19 of     19 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_sienna3.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_sienna3.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.632776 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 33 IDs ... 82%     27 of     33 study items found in association\n",
      "100%     33 of     33 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     218 GO terms are associated with     27 of     33 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 33 IDs ... 88%     29 of     33 study items found in association\n",
      "100%     33 of     33 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      86 GO terms are associated with     29 of     33 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 33 IDs ... 91%     30 of     33 study items found in association\n",
      "100%     33 of     33 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      95 GO terms are associated with     30 of     33 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_darkgrey.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_darkgrey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.545939 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 28 IDs ...100%     28 of     28 study items found in association\n",
      "100%     28 of     28 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     202 GO terms are associated with     28 of     28 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "       8 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 28 IDs ...100%     28 of     28 study items found in association\n",
      "100%     28 of     28 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      77 GO terms are associated with     28 of     28 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "      24 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 28 IDs ...100%     28 of     28 study items found in association\n",
      "100%     28 of     28 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     111 GO terms are associated with     28 of     28 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "      27 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[25] = 4 BP + 11 MF + 10 CC\n",
      "     25 items WROTE: GO_analysis_module_white.xlsx\n",
      "     25 GOEA results for    27 study items. WROTE: GO_analysis_module_white.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.633881 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 43 IDs ... 84%     36 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     207 GO terms are associated with     36 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 43 IDs ... 77%     33 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      85 GO terms are associated with     33 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       4 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 43 IDs ... 74%     32 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      61 GO terms are associated with     32 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "       4 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 2 MF + 1 CC\n",
      "      3 items WROTE: GO_analysis_module_lightgreen.xlsx\n",
      "      3 GOEA results for     4 study items. WROTE: GO_analysis_module_lightgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.572526 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 24 IDs ... 92%     22 of     24 study items found in association\n",
      "100%     24 of     24 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     178 GO terms are associated with     22 of     24 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 24 IDs ... 92%     22 of     24 study items found in association\n",
      "100%     24 of     24 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      68 GO terms are associated with     22 of     24 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 24 IDs ... 79%     19 of     24 study items found in association\n",
      "100%     24 of     24 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      81 GO terms are associated with     19 of     24 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_saddlebrown.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_saddlebrown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.583317 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 53 IDs ... 96%     51 of     53 study items found in association\n",
      "100%     53 of     53 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     337 GO terms are associated with     51 of     53 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "       4 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 53 IDs ... 98%     52 of     53 study items found in association\n",
      "100%     53 of     53 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     125 GO terms are associated with     52 of     53 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      46 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 53 IDs ... 94%     50 of     53 study items found in association\n",
      "100%     53 of     53 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     114 GO terms are associated with     50 of     53 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      43 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[17] = 4 BP + 8 MF + 5 CC\n",
      "     17 items WROTE: GO_analysis_module_lightcyan.xlsx\n",
      "     17 GOEA results for    50 study items. WROTE: GO_analysis_module_lightcyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.619552 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 26 IDs ... 88%     23 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     154 GO terms are associated with     23 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 26 enriched +   0 purified): statsmodels fdr_bh\n",
      "      21 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 26 IDs ... 92%     24 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      34 GO terms are associated with     24 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      22 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 26 IDs ... 96%     25 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      45 GO terms are associated with     25 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "      23 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[38] = 26 BP + 10 MF + 2 CC\n",
      "     38 items WROTE: GO_analysis_module_steelblue.xlsx\n",
      "     38 GOEA results for    25 study items. WROTE: GO_analysis_module_steelblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.637184 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 96 IDs ... 82%     79 of     96 study items found in association\n",
      "100%     96 of     96 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     408 GO terms are associated with     79 of     96 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       6 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 96 IDs ... 90%     86 of     96 study items found in association\n",
      "100%     96 of     96 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     114 GO terms are associated with     86 of     96 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 96 IDs ... 80%     77 of     96 study items found in association\n",
      "100%     96 of     96 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     161 GO terms are associated with     77 of     96 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[1] = 1 BP + 0 MF + 0 CC\n",
      "      1 items WROTE: GO_analysis_module_magenta.xlsx\n",
      "      1 GOEA results for     6 study items. WROTE: GO_analysis_module_magenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.325556 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 70 IDs ... 84%     59 of     70 study items found in association\n",
      "100%     70 of     70 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     409 GO terms are associated with     59 of     70 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 70 IDs ... 87%     61 of     70 study items found in association\n",
      "100%     70 of     70 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     109 GO terms are associated with     61 of     70 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 70 IDs ... 90%     63 of     70 study items found in association\n",
      "100%     70 of     70 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     139 GO terms are associated with     63 of     70 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_tan.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_tan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.675666 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 36 IDs ... 89%     32 of     36 study items found in association\n",
      "100%     36 of     36 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     310 GO terms are associated with     32 of     36 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 36 IDs ... 94%     34 of     36 study items found in association\n",
      "100%     36 of     36 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      95 GO terms are associated with     34 of     36 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      22 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 36 IDs ... 97%     35 of     36 study items found in association\n",
      "100%     36 of     36 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     102 GO terms are associated with     35 of     36 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 0 MF + 3 CC\n",
      "      3 items WROTE: GO_analysis_module_darkturquoise.xlsx\n",
      "      3 GOEA results for    22 study items. WROTE: GO_analysis_module_darkturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.584321 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 56 IDs ... 93%     52 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     447 GO terms are associated with     52 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      18 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 56 IDs ... 95%     53 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     132 GO terms are associated with     53 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      40 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 56 IDs ... 96%     54 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     164 GO terms are associated with     54 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      46 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[14] = 3 BP + 4 MF + 7 CC\n",
      "     14 items WROTE: GO_analysis_module_cyan.xlsx\n",
      "     14 GOEA results for    50 study items. WROTE: GO_analysis_module_cyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.724443 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 26 IDs ... 92%     24 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     142 GO terms are associated with     24 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 26 IDs ... 96%     25 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      56 GO terms are associated with     25 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "       6 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 26 IDs ... 85%     22 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      50 GO terms are associated with     22 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "       2 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 0 BP + 2 MF + 2 CC\n",
      "      4 items WROTE: GO_analysis_module_violet.xlsx\n",
      "      4 GOEA results for     6 study items. WROTE: GO_analysis_module_violet.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.349747 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 17 IDs ... 88%     15 of     17 study items found in association\n",
      "100%     17 of     17 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     133 GO terms are associated with     15 of     17 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 17 IDs ... 88%     15 of     17 study items found in association\n",
      "100%     17 of     17 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      44 GO terms are associated with     15 of     17 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 17 IDs ... 82%     14 of     17 study items found in association\n",
      "100%     17 of     17 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      51 GO terms are associated with     14 of     17 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_paleturquoise.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_paleturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.661904 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 21 IDs ...100%     21 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     165 GO terms are associated with     21 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "      13 GO terms found significant (< 0.05=alpha) ( 13 enriched +   0 purified): statsmodels fdr_bh\n",
      "      19 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 21 IDs ...100%     21 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      51 GO terms are associated with     21 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      12 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 21 IDs ...100%     21 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      68 GO terms are associated with     21 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[16] = 13 BP + 0 MF + 3 CC\n",
      "     16 items WROTE: GO_analysis_module_yellowgreen.xlsx\n",
      "     16 GOEA results for    20 study items. WROTE: GO_analysis_module_yellowgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.364474 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 23 IDs ... 91%     21 of     23 study items found in association\n",
      "100%     23 of     23 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     411 GO terms are associated with     21 of     23 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "      15 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 23 IDs ...100%     23 of     23 study items found in association\n",
      "100%     23 of     23 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      46 GO terms are associated with     23 of     23 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      15 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 23 IDs ...100%     23 of     23 study items found in association\n",
      "100%     23 of     23 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      87 GO terms are associated with     23 of     23 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "      21 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[19] = 9 BP + 6 MF + 4 CC\n",
      "     19 items WROTE: GO_analysis_module_darkolivegreen.xlsx\n",
      "     19 GOEA results for    21 study items. WROTE: GO_analysis_module_darkolivegreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.967264 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 22 IDs ... 91%     20 of     22 study items found in association\n",
      "100%     22 of     22 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     204 GO terms are associated with     20 of     22 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 22 IDs ... 91%     20 of     22 study items found in association\n",
      "100%     22 of     22 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      77 GO terms are associated with     20 of     22 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 22 IDs ... 91%     20 of     22 study items found in association\n",
      "100%     22 of     22 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      82 GO terms are associated with     20 of     22 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_darkmagenta.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_darkmagenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.777650 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 57 IDs ... 88%     50 of     57 study items found in association\n",
      "100%     57 of     57 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     521 GO terms are associated with     50 of     57 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "       2 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 57 IDs ... 93%     53 of     57 study items found in association\n",
      "100%     57 of     57 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     105 GO terms are associated with     53 of     57 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 57 IDs ... 93%     53 of     57 study items found in association\n",
      "100%     57 of     57 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     147 GO terms are associated with     53 of     57 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[10] = 10 BP + 0 MF + 0 CC\n",
      "     10 items WROTE: GO_analysis_module_salmon.xlsx\n",
      "     10 GOEA results for     2 study items. WROTE: GO_analysis_module_salmon.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.876665 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 37 IDs ... 92%     34 of     37 study items found in association\n",
      "100%     37 of     37 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     219 GO terms are associated with     34 of     37 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 37 IDs ... 97%     36 of     37 study items found in association\n",
      "100%     37 of     37 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      72 GO terms are associated with     36 of     37 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      24 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 37 IDs ... 86%     32 of     37 study items found in association\n",
      "100%     37 of     37 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      90 GO terms are associated with     32 of     37 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       2 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 0 BP + 1 MF + 3 CC\n",
      "      4 items WROTE: GO_analysis_module_darkred.xlsx\n",
      "      4 GOEA results for    24 study items. WROTE: GO_analysis_module_darkred.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.846772 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 39 IDs ... 85%     33 of     39 study items found in association\n",
      "100%     39 of     39 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     209 GO terms are associated with     33 of     39 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 39 IDs ... 92%     36 of     39 study items found in association\n",
      "100%     39 of     39 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      66 GO terms are associated with     36 of     39 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 39 IDs ... 82%     32 of     39 study items found in association\n",
      "100%     39 of     39 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      91 GO terms are associated with     32 of     39 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_royalblue.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_royalblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.579317 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 47 IDs ... 87%     41 of     47 study items found in association\n",
      "100%     47 of     47 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     262 GO terms are associated with     41 of     47 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 47 IDs ... 89%     42 of     47 study items found in association\n",
      "100%     47 of     47 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     101 GO terms are associated with     42 of     47 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      20 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 47 IDs ... 89%     42 of     47 study items found in association\n",
      "100%     47 of     47 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     101 GO terms are associated with     42 of     47 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[1] = 0 BP + 0 MF + 1 CC\n",
      "      1 items WROTE: GO_analysis_module_grey60.xlsx\n",
      "      1 GOEA results for    20 study items. WROTE: GO_analysis_module_grey60.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.567898 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 32 IDs ...100%     32 of     32 study items found in association\n",
      "100%     32 of     32 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     259 GO terms are associated with     32 of     32 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "       4 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 32 IDs ... 97%     31 of     32 study items found in association\n",
      "100%     32 of     32 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      39 GO terms are associated with     31 of     32 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      18 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 32 IDs ...100%     32 of     32 study items found in association\n",
      "100%     32 of     32 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      86 GO terms are associated with     32 of     32 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[8] = 7 BP + 0 MF + 1 CC\n",
      "      8 items WROTE: GO_analysis_module_darkorange.xlsx\n",
      "      8 GOEA results for    18 study items. WROTE: GO_analysis_module_darkorange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.461756 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 21 IDs ... 71%     15 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     124 GO terms are associated with     15 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 21 IDs ... 95%     20 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      48 GO terms are associated with     20 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 21 IDs ... 95%     20 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      49 GO terms are associated with     20 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      19 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[1] = 0 BP + 1 MF + 0 CC\n",
      "      1 items WROTE: GO_analysis_module_skyblue3.xlsx\n",
      "      1 GOEA results for    19 study items. WROTE: GO_analysis_module_skyblue3.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.287452 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "BP 18,506 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "MF 18,191 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 21 IDs ... 71%     15 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     108 GO terms are associated with     15 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 21 IDs ... 81%     17 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      46 GO terms are associated with     17 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 21 IDs ... 81%     17 of     21 study items found in association\n",
      "100%     21 of     21 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      40 GO terms are associated with     17 of     21 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_skyblue.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_skyblue.txt\n"
     ]
    }
   ],
   "source": [
    "for mod in get_wgcna_modules().module.unique():\n",
    "    run_goea(mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
