{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GO module enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:42:18.498115Z",
     "iopub.status.busy": "2021-07-12T15:42:18.495845Z",
     "iopub.status.idle": "2021-07-12T15:42:19.190163Z",
     "shell.execute_reply": "2021-07-12T15:42:19.188648Z"
    }
   },
   "outputs": [],
   "source": [
    "import functools\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import collections as cx\n",
    "from pybiomart import Dataset\n",
    "from gtfparse import read_gtf\n",
    "from scipy.stats import fisher_exact\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "# GO analysis\n",
    "from goatools.base import download_go_basic_obo\n",
    "from goatools.base import download_ncbi_associations\n",
    "from goatools.obo_parser import GODag\n",
    "from goatools.anno.genetogo_reader import Gene2GoReader\n",
    "from goatools.goea.go_enrichment_ns import GOEnrichmentStudyNS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cached functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:42:19.199346Z",
     "iopub.status.busy": "2021-07-12T15:42:19.198081Z",
     "iopub.status.idle": "2021-07-12T15:42:19.201900Z",
     "shell.execute_reply": "2021-07-12T15:42:19.200672Z"
    }
   },
   "outputs": [],
   "source": [
    "@functools.lru_cache()\n",
    "def get_gtf_genes_df():\n",
    "    gtf_df = read_gtf(\"/ceph/genome/human/gencode25/gtf.CHR/_m/gencode.v25.annotation.gtf\")\n",
    "    return gtf_df[gtf_df[\"feature\"] == \"gene\"][['gene_id', 'gene_name']]\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_wgcna_modules():\n",
    "    return pd.read_csv(\"../../_m/modules.csv\", index_col=0)\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_database():\n",
    "    dataset = Dataset(name=\"hsapiens_gene_ensembl\", \n",
    "                      host=\"http://www.ensembl.org\",\n",
    "                      use_cache=True)\n",
    "    db = dataset.query(attributes=[\"ensembl_gene_id\", \n",
    "                                   \"external_gene_name\", \n",
    "                                   \"entrezgene_id\"], \n",
    "                       use_attr_names=True).dropna(subset=['entrezgene_id'])\n",
    "    return db"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:42:19.212569Z",
     "iopub.status.busy": "2021-07-12T15:42:19.211322Z",
     "iopub.status.idle": "2021-07-12T15:42:19.217497Z",
     "shell.execute_reply": "2021-07-12T15:42:19.216221Z"
    }
   },
   "outputs": [],
   "source": [
    "def convert2entrez(mod):\n",
    "    df = get_wgcna_modules()[(get_wgcna_modules().module) == mod].copy()\n",
    "    df[\"ensemblID\"] = df.index.str.replace(\"\\\\..*\", \"\", regex=True)\n",
    "    return df.merge(get_database(), left_on='ensemblID', \n",
    "                    right_on='ensembl_gene_id')\n",
    "\n",
    "\n",
    "def obo_annotation(alpha=0.05):\n",
    "    # database annotation\n",
    "    fn_obo = download_go_basic_obo()\n",
    "    fn_gene2go = download_ncbi_associations() # must be gunzip to work\n",
    "    obodag = GODag(fn_obo) # downloads most up-to-date\n",
    "    anno_hs = Gene2GoReader(fn_gene2go, taxids=[9606])\n",
    "    # get associations\n",
    "    ns2assoc = anno_hs.get_ns2assc()\n",
    "    for nspc, id2gos in ns2assoc.items():\n",
    "        print(\"{NS} {N:,} annotated human genes\".format(NS=nspc, N=len(id2gos)))\n",
    "    goeaobj = GOEnrichmentStudyNS(\n",
    "        get_database()['entrezgene_id'], # List of human genes with entrez IDs\n",
    "        ns2assoc, # geneid/GO associations\n",
    "        obodag, # Ontologies\n",
    "        propagate_counts = False,\n",
    "        alpha = alpha, # default significance cut-off\n",
    "        methods = ['fdr_bh'])\n",
    "    return goeaobj\n",
    "\n",
    "\n",
    "def run_goea(mod):\n",
    "    df = convert2entrez(mod)\n",
    "    geneids_study = {z[0]:z[1] for z in zip(df['entrezgene_id'], df['external_gene_name'])}\n",
    "    goeaobj = obo_annotation()\n",
    "    goea_results_all = goeaobj.run_study(geneids_study)\n",
    "    goea_results_sig = [r for r in goea_results_all if r.p_fdr_bh < 0.05]\n",
    "    ctr = cx.Counter([r.NS for r in goea_results_sig])\n",
    "    print('Significant results[{TOTAL}] = {BP} BP + {MF} MF + {CC} CC'.format(\n",
    "        TOTAL=len(goea_results_sig),\n",
    "        BP=ctr['BP'],  # biological_process\n",
    "        MF=ctr['MF'],  # molecular_function\n",
    "        CC=ctr['CC'])) # cellular_component\n",
    "    goeaobj.wr_xlsx(\"GO_analysis_module_%s.xlsx\" % mod, goea_results_sig)\n",
    "    goeaobj.wr_txt(\"GO_analysis_module_%s.txt\" % mod, goea_results_sig)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:42:19.223634Z",
     "iopub.status.busy": "2021-07-12T15:42:19.222374Z",
     "iopub.status.idle": "2021-07-12T15:43:45.533804Z",
     "shell.execute_reply": "2021-07-12T15:43:45.534781Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_status', 'gene_name', 'level', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_status', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'protein_id', 'ccdsid']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id gene_name\n",
       "0   ENSG00000223972.5   DDX11L1\n",
       "12  ENSG00000227232.5    WASH7P"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf = get_gtf_genes_df()\n",
    "gtf.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load WGCNA module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:43:45.548616Z",
     "iopub.status.busy": "2021-07-12T15:43:45.547797Z",
     "iopub.status.idle": "2021-07-12T15:43:45.656451Z",
     "shell.execute_reply": "2021-07-12T15:43:45.657363Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000278267.1</td>\n",
       "      <td>MIR6859-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   module            gene_id  gene_name\n",
       "12   grey  ENSG00000227232.5     WASH7P\n",
       "25   grey  ENSG00000278267.1  MIR6859-1"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df = get_wgcna_modules().merge(gtf, left_index=True, right_on=\"gene_id\", how=\"left\")\n",
    "wgcna_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:43:45.670205Z",
     "iopub.status.busy": "2021-07-12T15:43:45.669306Z",
     "iopub.status.idle": "2021-07-12T15:43:45.705665Z",
     "shell.execute_reply": "2021-07-12T15:43:45.706522Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [module, gene_id, gene_name]\n",
       "Index: []"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_id.str.startswith(\"chr\"))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:43:45.726786Z",
     "iopub.status.busy": "2021-07-12T15:43:45.725683Z",
     "iopub.status.idle": "2021-07-12T15:43:45.730096Z",
     "shell.execute_reply": "2021-07-12T15:43:45.730921Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1445279</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000149295.13</td>\n",
       "      <td>DRD2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        module             gene_id gene_name\n",
       "1445279   grey  ENSG00000149295.13      DRD2"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_name == 'DRD2')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:43:45.739095Z",
     "iopub.status.busy": "2021-07-12T15:43:45.737526Z",
     "iopub.status.idle": "2021-07-12T15:43:45.829117Z",
     "shell.execute_reply": "2021-07-12T15:43:45.828166Z"
    }
   },
   "outputs": [],
   "source": [
    "wgcna_df.to_csv(\"module_annotated.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GO enrichment for each cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:43:45.837812Z",
     "iopub.status.busy": "2021-07-12T15:43:45.836947Z",
     "iopub.status.idle": "2021-07-12T15:48:07.289593Z",
     "shell.execute_reply": "2021-07-12T15:48:07.288372Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "requests.get(http://purl.obolibrary.org/obo/go/go-basic.obo, stream=True)\n",
      "  WROTE: go-basic.obo\n",
      "\n",
      "FTP RETR ftp.ncbi.nlm.nih.gov gene/DATA gene2go.gz -> gene2go.gz\n",
      "  gunzip gene2go.gz\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.511399 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 5238 IDs ... 71%  3,721 of  5,238 study items found in association\n",
      "100%  5,238 of  5,238 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   6,528 GO terms are associated with  3,721 of  5,238 study items\n",
      "  METHOD fdr_bh:\n",
      "      18 GO terms found significant (< 0.05=alpha) ( 15 enriched +   3 purified): statsmodels fdr_bh\n",
      "     812 study items associated with significant GO IDs (enriched)\n",
      "     128 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 5238 IDs ... 75%  3,949 of  5,238 study items found in association\n",
      "100%  5,238 of  5,238 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "   1,057 GO terms are associated with  3,949 of  5,238 study items\n",
      "  METHOD fdr_bh:\n",
      "      28 GO terms found significant (< 0.05=alpha) ( 23 enriched +   5 purified): statsmodels fdr_bh\n",
      "   2,060 study items associated with significant GO IDs (enriched)\n",
      "     737 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 5238 IDs ... 72%  3,747 of  5,238 study items found in association\n",
      "100%  5,238 of  5,238 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   2,252 GO terms are associated with  3,747 of  5,238 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) (  7 enriched +   4 purified): statsmodels fdr_bh\n",
      "   2,899 study items associated with significant GO IDs (enriched)\n",
      "      86 study items associated with significant GO IDs (purified)\n",
      "Significant results[57] = 18 BP + 11 MF + 28 CC\n",
      "     57 items WROTE: GO_analysis_module_grey.xlsx\n",
      "     57 GOEA results for  3621 study items. WROTE: GO_analysis_module_grey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.954009 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 17 IDs ... 71%     12 of     17 study items found in association\n",
      "100%     17 of     17 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "      44 GO terms are associated with     12 of     17 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 17 IDs ... 88%     15 of     17 study items found in association\n",
      "100%     17 of     17 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      48 GO terms are associated with     15 of     17 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 17 IDs ... 88%     15 of     17 study items found in association\n",
      "100%     17 of     17 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      30 GO terms are associated with     15 of     17 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_saddlebrown.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_saddlebrown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.614714 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 355 IDs ... 84%    298 of    355 study items found in association\n",
      "100%    355 of    355 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,420 GO terms are associated with    298 of    355 study items\n",
      "  METHOD fdr_bh:\n",
      "      18 GO terms found significant (< 0.05=alpha) ( 18 enriched +   0 purified): statsmodels fdr_bh\n",
      "      83 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 355 IDs ... 88%    314 of    355 study items found in association\n",
      "100%    355 of    355 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     338 GO terms are associated with    314 of    355 study items\n",
      "  METHOD fdr_bh:\n",
      "      47 GO terms found significant (< 0.05=alpha) ( 47 enriched +   0 purified): statsmodels fdr_bh\n",
      "     241 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 355 IDs ... 86%    304 of    355 study items found in association\n",
      "100%    355 of    355 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     480 GO terms are associated with    304 of    355 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "     226 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[74] = 18 BP + 9 MF + 47 CC\n",
      "     74 items WROTE: GO_analysis_module_cyan.xlsx\n",
      "     74 GOEA results for   298 study items. WROTE: GO_analysis_module_cyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.538422 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 786 IDs ... 87%    683 of    786 study items found in association\n",
      "100%    786 of    786 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,664 GO terms are associated with    683 of    786 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      19 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 786 IDs ... 91%    712 of    786 study items found in association\n",
      "100%    786 of    786 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     611 GO terms are associated with    712 of    786 study items\n",
      "  METHOD fdr_bh:\n",
      "      29 GO terms found significant (< 0.05=alpha) ( 29 enriched +   0 purified): statsmodels fdr_bh\n",
      "     620 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 786 IDs ... 89%    696 of    786 study items found in association\n",
      "100%    786 of    786 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     818 GO terms are associated with    696 of    786 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) (  9 enriched +   1 purified): statsmodels fdr_bh\n",
      "     574 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[40] = 1 BP + 10 MF + 29 CC\n",
      "     40 items WROTE: GO_analysis_module_yellow.xlsx\n",
      "     40 GOEA results for   698 study items. WROTE: GO_analysis_module_yellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.289541 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1054 IDs ... 85%    900 of  1,054 study items found in association\n",
      "100%  1,054 of  1,054 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,909 GO terms are associated with    900 of  1,054 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 14 enriched +   3 purified): statsmodels fdr_bh\n",
      "     202 study items associated with significant GO IDs (enriched)\n",
      "       5 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1054 IDs ... 91%    959 of  1,054 study items found in association\n",
      "100%  1,054 of  1,054 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     603 GO terms are associated with    959 of  1,054 study items\n",
      "  METHOD fdr_bh:\n",
      "      42 GO terms found significant (< 0.05=alpha) ( 42 enriched +   0 purified): statsmodels fdr_bh\n",
      "     892 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1054 IDs ... 89%    938 of  1,054 study items found in association\n",
      "100%  1,054 of  1,054 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   1,009 GO terms are associated with    938 of  1,054 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  8 enriched +   1 purified): statsmodels fdr_bh\n",
      "     779 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[68] = 17 BP + 9 MF + 42 CC\n",
      "     68 items WROTE: GO_analysis_module_blue.xlsx\n",
      "     68 GOEA results for   974 study items. WROTE: GO_analysis_module_blue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.540418 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 57 IDs ... 95%     54 of     57 study items found in association\n",
      "100%     57 of     57 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     356 GO terms are associated with     54 of     57 study items\n",
      "  METHOD fdr_bh:\n",
      "      40 GO terms found significant (< 0.05=alpha) ( 40 enriched +   0 purified): statsmodels fdr_bh\n",
      "      45 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 57 IDs ... 96%     55 of     57 study items found in association\n",
      "100%     57 of     57 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      89 GO terms are associated with     55 of     57 study items\n",
      "  METHOD fdr_bh:\n",
      "      13 GO terms found significant (< 0.05=alpha) ( 13 enriched +   0 purified): statsmodels fdr_bh\n",
      "      51 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 57 IDs ... 98%     56 of     57 study items found in association\n",
      "100%     57 of     57 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     108 GO terms are associated with     56 of     57 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "      54 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[69] = 40 BP + 16 MF + 13 CC\n",
      "     69 items WROTE: GO_analysis_module_paleturquoise.xlsx\n",
      "     69 GOEA results for    56 study items. WROTE: GO_analysis_module_paleturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.521634 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 440 IDs ... 82%    363 of    440 study items found in association\n",
      "100%    440 of    440 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,415 GO terms are associated with    363 of    440 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 440 IDs ... 87%    381 of    440 study items found in association\n",
      "100%    440 of    440 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     381 GO terms are associated with    381 of    440 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "     284 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 440 IDs ... 87%    381 of    440 study items found in association\n",
      "100%    440 of    440 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     534 GO terms are associated with    381 of    440 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     293 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[11] = 0 BP + 2 MF + 9 CC\n",
      "     11 items WROTE: GO_analysis_module_pink.xlsx\n",
      "     11 GOEA results for   354 study items. WROTE: GO_analysis_module_pink.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.560398 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 416 IDs ... 94%    389 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,612 GO terms are associated with    389 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 416 IDs ... 95%    397 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     389 GO terms are associated with    397 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "      18 GO terms found significant (< 0.05=alpha) ( 18 enriched +   0 purified): statsmodels fdr_bh\n",
      "     349 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 416 IDs ... 94%    390 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     525 GO terms are associated with    390 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "     338 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[26] = 0 BP + 8 MF + 18 CC\n",
      "     26 items WROTE: GO_analysis_module_purple.xlsx\n",
      "     26 GOEA results for   387 study items. WROTE: GO_analysis_module_purple.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.455410 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 160 IDs ... 85%    136 of    160 study items found in association\n",
      "100%    160 of    160 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     717 GO terms are associated with    136 of    160 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 160 IDs ... 89%    142 of    160 study items found in association\n",
      "100%    160 of    160 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     211 GO terms are associated with    142 of    160 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      88 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 160 IDs ... 91%    145 of    160 study items found in association\n",
      "100%    160 of    160 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     322 GO terms are associated with    145 of    160 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "     110 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[6] = 0 BP + 1 MF + 5 CC\n",
      "      6 items WROTE: GO_analysis_module_darkgrey.xlsx\n",
      "      6 GOEA results for   127 study items. WROTE: GO_analysis_module_darkgrey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.501407 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1551 IDs ... 81%  1,258 of  1,551 study items found in association\n",
      "100%  1,551 of  1,551 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   4,278 GO terms are associated with  1,258 of  1,551 study items\n",
      "  METHOD fdr_bh:\n",
      "      29 GO terms found significant (< 0.05=alpha) ( 27 enriched +   2 purified): statsmodels fdr_bh\n",
      "     469 study items associated with significant GO IDs (enriched)\n",
      "       5 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1551 IDs ... 85%  1,318 of  1,551 study items found in association\n",
      "100%  1,551 of  1,551 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     714 GO terms are associated with  1,318 of  1,551 study items\n",
      "  METHOD fdr_bh:\n",
      "      59 GO terms found significant (< 0.05=alpha) ( 59 enriched +   0 purified): statsmodels fdr_bh\n",
      "   1,242 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1551 IDs ... 84%  1,305 of  1,551 study items found in association\n",
      "100%  1,551 of  1,551 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   1,304 GO terms are associated with  1,305 of  1,551 study items\n",
      "  METHOD fdr_bh:\n",
      "      24 GO terms found significant (< 0.05=alpha) ( 21 enriched +   3 purified): statsmodels fdr_bh\n",
      "   1,117 study items associated with significant GO IDs (enriched)\n",
      "      17 study items associated with significant GO IDs (purified)\n",
      "Significant results[112] = 29 BP + 24 MF + 59 CC\n",
      "    112 items WROTE: GO_analysis_module_turquoise.xlsx\n",
      "    112 GOEA results for  1338 study items. WROTE: GO_analysis_module_turquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.533948 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 225 IDs ... 90%    203 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,109 GO terms are associated with    203 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 225 IDs ... 95%    213 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     267 GO terms are associated with    213 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "     161 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 225 IDs ... 93%    210 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     365 GO terms are associated with    210 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "     190 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[8] = 0 BP + 4 MF + 4 CC\n",
      "      8 items WROTE: GO_analysis_module_royalblue.xlsx\n",
      "      8 GOEA results for   203 study items. WROTE: GO_analysis_module_royalblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.356764 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 180 IDs ... 96%    172 of    180 study items found in association\n",
      "100%    180 of    180 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,230 GO terms are associated with    172 of    180 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      24 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 180 IDs ... 96%    172 of    180 study items found in association\n",
      "100%    180 of    180 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     336 GO terms are associated with    172 of    180 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 30 enriched +   0 purified): statsmodels fdr_bh\n",
      "     158 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 180 IDs ... 94%    169 of    180 study items found in association\n",
      "100%    180 of    180 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     377 GO terms are associated with    169 of    180 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "     139 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[43] = 5 BP + 8 MF + 30 CC\n",
      "     43 items WROTE: GO_analysis_module_darkturquoise.xlsx\n",
      "     43 GOEA results for   171 study items. WROTE: GO_analysis_module_darkturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.562696 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 138 IDs ... 82%    113 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     864 GO terms are associated with    113 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 138 IDs ... 86%    118 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     183 GO terms are associated with    118 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      29 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 138 IDs ... 80%    111 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     283 GO terms are associated with    111 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 0 MF + 3 CC\n",
      "      3 items WROTE: GO_analysis_module_orange.xlsx\n",
      "      3 GOEA results for    29 study items. WROTE: GO_analysis_module_orange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.486028 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 327 IDs ... 79%    259 of    327 study items found in association\n",
      "100%    327 of    327 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,395 GO terms are associated with    259 of    327 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      53 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 327 IDs ... 85%    277 of    327 study items found in association\n",
      "100%    327 of    327 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     291 GO terms are associated with    277 of    327 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "     163 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 327 IDs ... 78%    256 of    327 study items found in association\n",
      "100%    327 of    327 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     423 GO terms are associated with    256 of    327 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[15] = 5 BP + 0 MF + 10 CC\n",
      "     15 items WROTE: GO_analysis_module_midnightblue.xlsx\n",
      "     15 GOEA results for   178 study items. WROTE: GO_analysis_module_midnightblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.475668 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 567 IDs ... 89%    506 of    567 study items found in association\n",
      "100%    567 of    567 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,901 GO terms are associated with    506 of    567 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      80 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 567 IDs ... 95%    537 of    567 study items found in association\n",
      "100%    567 of    567 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     478 GO terms are associated with    537 of    567 study items\n",
      "  METHOD fdr_bh:\n",
      "      31 GO terms found significant (< 0.05=alpha) ( 28 enriched +   3 purified): statsmodels fdr_bh\n",
      "     497 study items associated with significant GO IDs (enriched)\n",
      "      25 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 567 IDs ... 92%    523 of    567 study items found in association\n",
      "100%    567 of    567 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     558 GO terms are associated with    523 of    567 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) (  9 enriched +   1 purified): statsmodels fdr_bh\n",
      "     468 study items associated with significant GO IDs (enriched)\n",
      "       1 study items associated with significant GO IDs (purified)\n",
      "Significant results[46] = 5 BP + 10 MF + 31 CC\n",
      "     46 items WROTE: GO_analysis_module_red.xlsx\n",
      "     46 GOEA results for   533 study items. WROTE: GO_analysis_module_red.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.583495 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 433 IDs ... 87%    376 of    433 study items found in association\n",
      "100%    433 of    433 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,773 GO terms are associated with    376 of    433 study items\n",
      "  METHOD fdr_bh:\n",
      "      51 GO terms found significant (< 0.05=alpha) ( 51 enriched +   0 purified): statsmodels fdr_bh\n",
      "     181 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 433 IDs ... 92%    398 of    433 study items found in association\n",
      "100%    433 of    433 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     395 GO terms are associated with    398 of    433 study items\n",
      "  METHOD fdr_bh:\n",
      "      59 GO terms found significant (< 0.05=alpha) ( 59 enriched +   0 purified): statsmodels fdr_bh\n",
      "     339 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 433 IDs ... 88%    379 of    433 study items found in association\n",
      "100%    433 of    433 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     528 GO terms are associated with    379 of    433 study items\n",
      "  METHOD fdr_bh:\n",
      "      24 GO terms found significant (< 0.05=alpha) ( 24 enriched +   0 purified): statsmodels fdr_bh\n",
      "     293 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[134] = 51 BP + 24 MF + 59 CC\n",
      "    134 items WROTE: GO_analysis_module_magenta.xlsx\n",
      "    134 GOEA results for   384 study items. WROTE: GO_analysis_module_magenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.431449 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 377 IDs ... 85%    319 of    377 study items found in association\n",
      "100%    377 of    377 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,561 GO terms are associated with    319 of    377 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 30 enriched +   0 purified): statsmodels fdr_bh\n",
      "      87 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 377 IDs ... 92%    345 of    377 study items found in association\n",
      "100%    377 of    377 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     378 GO terms are associated with    345 of    377 study items\n",
      "  METHOD fdr_bh:\n",
      "      33 GO terms found significant (< 0.05=alpha) ( 32 enriched +   1 purified): statsmodels fdr_bh\n",
      "     319 study items associated with significant GO IDs (enriched)\n",
      "       4 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 377 IDs ... 92%    348 of    377 study items found in association\n",
      "100%    377 of    377 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     518 GO terms are associated with    348 of    377 study items\n",
      "  METHOD fdr_bh:\n",
      "      24 GO terms found significant (< 0.05=alpha) ( 24 enriched +   0 purified): statsmodels fdr_bh\n",
      "     305 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[87] = 30 BP + 24 MF + 33 CC\n",
      "     87 items WROTE: GO_analysis_module_greenyellow.xlsx\n",
      "     87 GOEA results for   344 study items. WROTE: GO_analysis_module_greenyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.290355 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 212 IDs ... 90%    190 of    212 study items found in association\n",
      "100%    212 of    212 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,214 GO terms are associated with    190 of    212 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       3 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 212 IDs ... 94%    200 of    212 study items found in association\n",
      "100%    212 of    212 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     338 GO terms are associated with    200 of    212 study items\n",
      "  METHOD fdr_bh:\n",
      "      19 GO terms found significant (< 0.05=alpha) ( 19 enriched +   0 purified): statsmodels fdr_bh\n",
      "     176 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 212 IDs ... 92%    196 of    212 study items found in association\n",
      "100%    212 of    212 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     401 GO terms are associated with    196 of    212 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "     175 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[29] = 1 BP + 9 MF + 19 CC\n",
      "     29 items WROTE: GO_analysis_module_darkred.xlsx\n",
      "     29 GOEA results for   192 study items. WROTE: GO_analysis_module_darkred.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.558226 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 497 IDs ... 88%    439 of    497 study items found in association\n",
      "100%    497 of    497 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,448 GO terms are associated with    439 of    497 study items\n",
      "  METHOD fdr_bh:\n",
      "     295 GO terms found significant (< 0.05=alpha) (295 enriched +   0 purified): statsmodels fdr_bh\n",
      "     356 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 497 IDs ... 94%    466 of    497 study items found in association\n",
      "100%    497 of    497 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     340 GO terms are associated with    466 of    497 study items\n",
      "  METHOD fdr_bh:\n",
      "      75 GO terms found significant (< 0.05=alpha) ( 75 enriched +   0 purified): statsmodels fdr_bh\n",
      "     432 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 497 IDs ... 91%    452 of    497 study items found in association\n",
      "100%    497 of    497 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     681 GO terms are associated with    452 of    497 study items\n",
      "  METHOD fdr_bh:\n",
      "      57 GO terms found significant (< 0.05=alpha) ( 57 enriched +   0 purified): statsmodels fdr_bh\n",
      "     398 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[427] = 295 BP + 57 MF + 75 CC\n",
      "    427 items WROTE: GO_analysis_module_black.xlsx\n",
      "    427 GOEA results for   456 study items. WROTE: GO_analysis_module_black.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.438400 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 518 IDs ... 83%    429 of    518 study items found in association\n",
      "100%    518 of    518 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,501 GO terms are associated with    429 of    518 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 22 enriched +   0 purified): statsmodels fdr_bh\n",
      "     171 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 518 IDs ... 87%    450 of    518 study items found in association\n",
      "100%    518 of    518 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     383 GO terms are associated with    450 of    518 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 26 enriched +   0 purified): statsmodels fdr_bh\n",
      "     393 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 518 IDs ... 85%    438 of    518 study items found in association\n",
      "100%    518 of    518 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     701 GO terms are associated with    438 of    518 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "     336 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[54] = 22 BP + 6 MF + 26 CC\n",
      "     54 items WROTE: GO_analysis_module_green.xlsx\n",
      "     54 GOEA results for   448 study items. WROTE: GO_analysis_module_green.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.612271 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 118 IDs ... 93%    110 of    118 study items found in association\n",
      "100%    118 of    118 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     729 GO terms are associated with    110 of    118 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "      42 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 118 IDs ... 96%    113 of    118 study items found in association\n",
      "100%    118 of    118 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     251 GO terms are associated with    113 of    118 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 21 enriched +   0 purified): statsmodels fdr_bh\n",
      "     106 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 118 IDs ... 92%    109 of    118 study items found in association\n",
      "100%    118 of    118 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     262 GO terms are associated with    109 of    118 study items\n",
      "  METHOD fdr_bh:\n",
      "      14 GO terms found significant (< 0.05=alpha) ( 14 enriched +   0 purified): statsmodels fdr_bh\n",
      "      96 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[46] = 11 BP + 14 MF + 21 CC\n",
      "     46 items WROTE: GO_analysis_module_white.xlsx\n",
      "     46 GOEA results for   110 study items. WROTE: GO_analysis_module_white.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.303362 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 250 IDs ... 94%    234 of    250 study items found in association\n",
      "100%    250 of    250 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,123 GO terms are associated with    234 of    250 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      32 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 250 IDs ... 97%    242 of    250 study items found in association\n",
      "100%    250 of    250 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     341 GO terms are associated with    242 of    250 study items\n",
      "  METHOD fdr_bh:\n",
      "      36 GO terms found significant (< 0.05=alpha) ( 36 enriched +   0 purified): statsmodels fdr_bh\n",
      "     231 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 250 IDs ... 94%    234 of    250 study items found in association\n",
      "100%    250 of    250 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     345 GO terms are associated with    234 of    250 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "     214 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[45] = 3 BP + 6 MF + 36 CC\n",
      "     45 items WROTE: GO_analysis_module_lightgreen.xlsx\n",
      "     45 GOEA results for   239 study items. WROTE: GO_analysis_module_lightgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.573554 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 849 IDs ... 87%    738 of    849 study items found in association\n",
      "100%    849 of    849 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,186 GO terms are associated with    738 of    849 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) ( 10 enriched +   2 purified): statsmodels fdr_bh\n",
      "     236 study items associated with significant GO IDs (enriched)\n",
      "       4 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 849 IDs ... 92%    777 of    849 study items found in association\n",
      "100%    849 of    849 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     559 GO terms are associated with    777 of    849 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 26 enriched +   4 purified): statsmodels fdr_bh\n",
      "     676 study items associated with significant GO IDs (enriched)\n",
      "     126 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 849 IDs ... 88%    751 of    849 study items found in association\n",
      "100%    849 of    849 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     813 GO terms are associated with    751 of    849 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 20 enriched +   2 purified): statsmodels fdr_bh\n",
      "     651 study items associated with significant GO IDs (enriched)\n",
      "       3 study items associated with significant GO IDs (purified)\n",
      "Significant results[64] = 12 BP + 22 MF + 30 CC\n",
      "     64 items WROTE: GO_analysis_module_brown.xlsx\n",
      "     64 GOEA results for   764 study items. WROTE: GO_analysis_module_brown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.497288 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 86 IDs ... 79%     68 of     86 study items found in association\n",
      "100%     86 of     86 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     534 GO terms are associated with     68 of     86 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 86 IDs ... 81%     70 of     86 study items found in association\n",
      "100%     86 of     86 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     118 GO terms are associated with     70 of     86 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 86 IDs ... 79%     68 of     86 study items found in association\n",
      "100%     86 of     86 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     174 GO terms are associated with     68 of     86 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_skyblue.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_skyblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.594846 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 112 IDs ... 81%     91 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     471 GO terms are associated with     91 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 112 IDs ... 86%     96 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     144 GO terms are associated with     96 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      60 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 112 IDs ... 86%     96 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     206 GO terms are associated with     96 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      84 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[6] = 0 BP + 3 MF + 3 CC\n",
      "      6 items WROTE: GO_analysis_module_darkorange.xlsx\n",
      "      6 GOEA results for    92 study items. WROTE: GO_analysis_module_darkorange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.529465 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 342 IDs ... 95%    325 of    342 study items found in association\n",
      "100%    342 of    342 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,264 GO terms are associated with    325 of    342 study items\n",
      "  METHOD fdr_bh:\n",
      "     169 GO terms found significant (< 0.05=alpha) (169 enriched +   0 purified): statsmodels fdr_bh\n",
      "     260 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 342 IDs ... 97%    332 of    342 study items found in association\n",
      "100%    342 of    342 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     359 GO terms are associated with    332 of    342 study items\n",
      "  METHOD fdr_bh:\n",
      "      63 GO terms found significant (< 0.05=alpha) ( 63 enriched +   0 purified): statsmodels fdr_bh\n",
      "     323 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 342 IDs ... 96%    330 of    342 study items found in association\n",
      "100%    342 of    342 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     580 GO terms are associated with    330 of    342 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 26 enriched +   0 purified): statsmodels fdr_bh\n",
      "     288 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[258] = 169 BP + 26 MF + 63 CC\n",
      "    258 items WROTE: GO_analysis_module_lightcyan.xlsx\n",
      "    258 GOEA results for   331 study items. WROTE: GO_analysis_module_lightcyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.557003 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 371 IDs ... 91%    339 of    371 study items found in association\n",
      "100%    371 of    371 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,779 GO terms are associated with    339 of    371 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 17 enriched +   0 purified): statsmodels fdr_bh\n",
      "     121 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 371 IDs ... 94%    349 of    371 study items found in association\n",
      "100%    371 of    371 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     424 GO terms are associated with    349 of    371 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 30 enriched +   0 purified): statsmodels fdr_bh\n",
      "     314 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 371 IDs ... 93%    345 of    371 study items found in association\n",
      "100%    371 of    371 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     462 GO terms are associated with    345 of    371 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 21 enriched +   0 purified): statsmodels fdr_bh\n",
      "     307 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[68] = 17 BP + 21 MF + 30 CC\n",
      "     68 items WROTE: GO_analysis_module_salmon.xlsx\n",
      "     68 GOEA results for   342 study items. WROTE: GO_analysis_module_salmon.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.329587 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 51 IDs ... 96%     49 of     51 study items found in association\n",
      "100%     51 of     51 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     319 GO terms are associated with     49 of     51 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 51 IDs ...100%     51 of     51 study items found in association\n",
      "100%     51 of     51 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     137 GO terms are associated with     51 of     51 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      36 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 51 IDs ... 94%     48 of     51 study items found in association\n",
      "100%     51 of     51 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     133 GO terms are associated with     48 of     51 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      43 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[11] = 0 BP + 8 MF + 3 CC\n",
      "     11 items WROTE: GO_analysis_module_violet.xlsx\n",
      "     11 GOEA results for    45 study items. WROTE: GO_analysis_module_violet.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.568103 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 195 IDs ... 85%    166 of    195 study items found in association\n",
      "100%    195 of    195 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     957 GO terms are associated with    166 of    195 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      48 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 195 IDs ... 91%    177 of    195 study items found in association\n",
      "100%    195 of    195 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     205 GO terms are associated with    177 of    195 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "     132 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 195 IDs ... 84%    164 of    195 study items found in association\n",
      "100%    195 of    195 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     322 GO terms are associated with    164 of    195 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "      42 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[24] = 7 BP + 6 MF + 11 CC\n",
      "     24 items WROTE: GO_analysis_module_darkgreen.xlsx\n",
      "     24 GOEA results for   135 study items. WROTE: GO_analysis_module_darkgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.516710 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 234 IDs ... 94%    221 of    234 study items found in association\n",
      "100%    234 of    234 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,512 GO terms are associated with    221 of    234 study items\n",
      "  METHOD fdr_bh:\n",
      "      80 GO terms found significant (< 0.05=alpha) ( 80 enriched +   0 purified): statsmodels fdr_bh\n",
      "     149 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 234 IDs ... 96%    225 of    234 study items found in association\n",
      "100%    234 of    234 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     245 GO terms are associated with    225 of    234 study items\n",
      "  METHOD fdr_bh:\n",
      "      38 GO terms found significant (< 0.05=alpha) ( 38 enriched +   0 purified): statsmodels fdr_bh\n",
      "     205 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 234 IDs ... 95%    223 of    234 study items found in association\n",
      "100%    234 of    234 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     399 GO terms are associated with    223 of    234 study items\n",
      "  METHOD fdr_bh:\n",
      "      20 GO terms found significant (< 0.05=alpha) ( 20 enriched +   0 purified): statsmodels fdr_bh\n",
      "     190 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[138] = 80 BP + 20 MF + 38 CC\n",
      "    138 items WROTE: GO_analysis_module_lightyellow.xlsx\n",
      "    138 GOEA results for   229 study items. WROTE: GO_analysis_module_lightyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.310915 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 304 IDs ... 86%    261 of    304 study items found in association\n",
      "100%    304 of    304 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,192 GO terms are associated with    261 of    304 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      76 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 304 IDs ... 92%    279 of    304 study items found in association\n",
      "100%    304 of    304 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     308 GO terms are associated with    279 of    304 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "     226 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 304 IDs ... 89%    272 of    304 study items found in association\n",
      "100%    304 of    304 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     404 GO terms are associated with    272 of    304 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "     237 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[17] = 5 BP + 3 MF + 9 CC\n",
      "     17 items WROTE: GO_analysis_module_grey60.xlsx\n",
      "     17 GOEA results for   267 study items. WROTE: GO_analysis_module_grey60.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.570661 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 387 IDs ... 88%    339 of    387 study items found in association\n",
      "100%    387 of    387 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,217 GO terms are associated with    339 of    387 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      27 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 387 IDs ... 94%    365 of    387 study items found in association\n",
      "100%    387 of    387 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     390 GO terms are associated with    365 of    387 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 21 enriched +   0 purified): statsmodels fdr_bh\n",
      "     318 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 387 IDs ... 93%    358 of    387 study items found in association\n",
      "100%    387 of    387 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     505 GO terms are associated with    358 of    387 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "     307 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[27] = 3 BP + 3 MF + 21 CC\n",
      "     27 items WROTE: GO_analysis_module_tan.xlsx\n",
      "     27 GOEA results for   360 study items. WROTE: GO_analysis_module_tan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.316526 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 59 IDs ... 92%     54 of     59 study items found in association\n",
      "100%     59 of     59 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     471 GO terms are associated with     54 of     59 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "      12 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 59 IDs ... 92%     54 of     59 study items found in association\n",
      "100%     59 of     59 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      79 GO terms are associated with     54 of     59 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      31 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 59 IDs ... 90%     53 of     59 study items found in association\n",
      "100%     59 of     59 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     129 GO terms are associated with     53 of     59 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      44 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[19] = 9 BP + 8 MF + 2 CC\n",
      "     19 items WROTE: GO_analysis_module_steelblue.xlsx\n",
      "     19 GOEA results for    47 study items. WROTE: GO_analysis_module_steelblue.txt\n"
     ]
    }
   ],
   "source": [
    "for mod in get_wgcna_modules().module.unique():\n",
    "    run_goea(mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
