{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential Expression with limma-voom pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressMessages({library(SummarizedExperiment)\n",
    "                  library(data.table)\n",
    "                  library(tidyverse)\n",
    "                  library(memoise)\n",
    "                  library(limma)\n",
    "                  library(edgeR)\n",
    "                  library(sva)})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function from jaffelab github\n",
    "merge_rse_metrics <- function(rse) {\n",
    "    stopifnot(is(rse, 'RangedSummarizedExperiment'))\n",
    "\n",
    "    rse$overallMapRate = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$overallMapRate, rse$numReads)\n",
    "    rse$mitoRate = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$mitoRate, rse$numMapped)\n",
    "    rse$rRNA_rate = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$rRNA_rate, rse$numMapped)\n",
    "    rse$totalAssignedGene = mapply(function(r, n) {\n",
    "        sum(r*n)/sum(n)\n",
    "    }, rse$totalAssignedGene, rse$numMapped)\n",
    "\n",
    "    rse$numMapped = sapply(rse$numMapped, sum)\n",
    "    rse$numReads = sapply(rse$numReads, sum)\n",
    "    rse$numUnmapped = sapply(rse$numUnmapped, sum)\n",
    "    rse$mitoMapped = sapply(rse$mitoMapped, sum)\n",
    "    rse$totalMapped = sapply(rse$totalMapped, sum)\n",
    "    return(rse)\n",
    "}\n",
    "\n",
    "\n",
    "save_volcanoPlot <- function(top, label, feature){\n",
    "    pdf(file=paste0(feature, \"/volcanoPlot_\", label, \".pdf\"), 8, 6)\n",
    "    with(top, plot(logFC, -log10(P.Value), pch=20, cex=0.6))\n",
    "    with(subset(top, adj.P.Val<=0.05), points(logFC, -log10(P.Value),\n",
    "                                              pch=20, col='red', cex=0.6))\n",
    "    with(subset(top, abs(logFC)>0.50), points(logFC, -log10(P.Value),\n",
    "                                              pch=20, col='orange', cex=0.6))\n",
    "    with(subset(top, adj.P.Val<=0.05 & abs(logFC)>0.50),\n",
    "         points(logFC, -log10(P.Value), pch=20, col='green', cex=0.6))\n",
    "    dev.off()\n",
    "}\n",
    "\n",
    "\n",
    "save_MAplot <- function(top, label, feature){\n",
    "    pdf(file=paste0(feature, \"/MAplot_\", label, \".pdf\"), 8, 6)\n",
    "    with(top, plot(AveExpr, logFC, pch=20, cex=0.5))\n",
    "    with(subset(top, adj.P.Val<0.05),\n",
    "         points(AveExpr, logFC, col=\"red\", pch=20, cex=0.5))\n",
    "    dev.off()\n",
    "}\n",
    "\n",
    "\n",
    "extract_de <- function(contrast, label, efit, feature){\n",
    "    top <- topTable(efit, coef=contrast, number=Inf, sort.by=\"P\")\n",
    "    top$SE <- sqrt(efit$s2.post) * efit$stdev.unscaled\n",
    "    top.fdr <- top %>% filter(adj.P.Val<=0.05)\n",
    "    print(paste(\"Comparison for:\", label))\n",
    "    print(paste('There are:', dim(top.fdr)[1], 'DE features!'))\n",
    "    fwrite(top, \n",
    "           file=paste0(feature, \"/diffExpr_\", label, \"_full.txt\"), \n",
    "           sep='\\t', row.names=TRUE)\n",
    "    fwrite(top.fdr, \n",
    "           file=paste0(feature, \"/diffExpr_\", label, \"_FDR05.txt\"), \n",
    "           sep='\\t', row.names=TRUE)\n",
    "    save_volcanoPlot(top, label, feature)\n",
    "    save_MAplot(top, label, feature)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cached functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "prep_data <- function(feature){\n",
    "    ancestry = \"../../../input/ancestry_structure/structure.out_ancestry_proportion_raceDemo_compare\"\n",
    "    counts_lt = list(\"genes\"=\"../../../input/counts/_m/caudate_brainseq_phase3_hg38_rseGene_merged_n464.rda\", \n",
    "                     \"transcripts\"=\"../../../input/counts/_m/caudate_brainseq_phase3_hg38_rseTx_merged_n464.rda\",\n",
    "                     \"exons\"=\"../../../input/counts/_m/caudate_brainseq_phase3_hg38_rseExon_merged_n464.rda\",\n",
    "                     \"junctions\"=\"../../../input/counts/_m/caudate_brainseq_phase3_hg38_rseJxn_merged_n464.rda\")\n",
    "    tx_file = \"../../../input/counts/_m/transcripts_counts/caudate_counts.txt\"\n",
    "    load(counts_lt[[feature]])\n",
    "    if(exists(\"rse_gene\")){\n",
    "        rse_df = rse_gene\n",
    "    } else if (exists(\"rse_tx\")){\n",
    "        rse_df = rse_tx\n",
    "        counts <- fread(tx_file) %>% column_to_rownames(\"gencodeTx\") %>% \n",
    "            select(-c(txLength, gencodeID, Symbol, gene_type)) %>% as.matrix\n",
    "        assays(rse_df)$counts = counts[, colnames(rse_df)]\n",
    "    } else if (exists(\"rse_exon\")){\n",
    "        rse_df = rse_exon\n",
    "    } else {\n",
    "        rse_df = rse_jxn\n",
    "    }\n",
    "    keepIndex = which((rse_df$Dx %in% c(\"Control\")) & \n",
    "                      rse_df$Age > 17 & \n",
    "                      rse_df$Race %in% c(\"AA\"))\n",
    "    rse_df = rse_df[, keepIndex]\n",
    "    rse_df$Dx = factor(rse_df$Dx, levels = c(\"Control\", \"Schizo\"))\n",
    "    rse_df$Sex <- factor(rse_df$Sex)\n",
    "    rse_df <- merge_rse_metrics(rse_df)\n",
    "    colData(rse_df)$RIN = sapply(colData(rse_df)$RIN,\"[\",1)\n",
    "    rownames(colData(rse_df)) <- sapply(strsplit(rownames(colData(rse_df)), \"_\"), \"[\", 1)\n",
    "    pheno = colData(rse_df) %>% as.data.frame %>% \n",
    "        inner_join(fread(ancestry), by=c(\"BrNum\"=\"id\", \"Race\"=\"group\"))\n",
    "    # Generate DGE list\n",
    "    x <- DGEList(counts=assays(rse_df)$counts[, pheno$RNum], \n",
    "                 genes=rowData(rse_df), \n",
    "                 samples=pheno)\n",
    "    # Filter by expression\n",
    "    keep.x <- filterByExpr(x)\n",
    "    x <- x[keep.x, , keep.lib.sizes=FALSE]\n",
    "    print(paste('There are:', sum(keep.x), 'features left!', sep=' '))\n",
    "    # Normalize library size\n",
    "    x <- calcNormFactors(x, method=\"TMM\")\n",
    "    return(x)\n",
    "}\n",
    "\n",
    "memo_prepData <- memoise(prep_data)\n",
    "\n",
    "qSV_model <- function(feature){\n",
    "    x <- memo_prepData(feature)\n",
    "    # Design matrix\n",
    "    mod = model.matrix(~Eur + Sex + Age + mitoRate + rRNA_rate + \n",
    "                       totalAssignedGene + overallMapRate, data = x$samples)\n",
    "    #colnames(mod) <- gsub(\"Dx\", \"\", colnames(mod))\n",
    "    colnames(mod) <- gsub(\"SexM\", \"Male\", colnames(mod))\n",
    "    colnames(mod) <- gsub(\"Eur\", \"EA\", colnames(mod))\n",
    "    colnames(mod) <- gsub(\"\\\\(Intercept\\\\)\", \"Intercept\", colnames(mod))\n",
    "    # Load qSV\n",
    "    qsv_file = \"../../../input/phenotypes/_m/qSV_caudate.csv\"\n",
    "    modQsva <- mod %>% as.data.frame %>% rownames_to_column() %>%\n",
    "        inner_join(fread(qsv_file), by=c(\"rowname\"=\"V1\")) %>% \n",
    "        rename_all(list(~str_replace_all(., 'PC', 'qPC'))) %>% \n",
    "        column_to_rownames(\"rowname\") %>% as.matrix\n",
    "    return(modQsva)\n",
    "}\n",
    "\n",
    "memo_qsvModel <- memoise(qSV_model)\n",
    "\n",
    "get_voom <- function(feature){\n",
    "    ### Preform voom\n",
    "    x <- memo_prepData(feature)\n",
    "    modQsva <- memo_qsvModel(feature)\n",
    "    v <- voom(x[, rownames(modQsva)], modQsva, plot=TRUE)\n",
    "    return(v)\n",
    "}\n",
    "\n",
    "memo_voom <- memoise(get_voom)\n",
    "\n",
    "cal_res <- function(feature){\n",
    "    ### Calculate residuals\n",
    "    v <- memo_voom(feature)\n",
    "    null_model <- v$design %>% as.data.frame %>% select(-c(\"EA\")) %>% as.matrix\n",
    "    fit_res <- lmFit(v, design=null_model)\n",
    "    res = v$E - ( fit_res$coefficients %*% t(null_model) )\n",
    "    res_sd = apply(res, 1, sd)\n",
    "    res_mean = apply(res, 1, mean)\n",
    "    res_norm = (res - res_mean) / res_sd\n",
    "    write.table(res_norm, file=paste0(feature, '/residualized_expression.tsv'), \n",
    "                sep=\"\\t\", quote=FALSE)\n",
    "}\n",
    "\n",
    "memo_res <- memoise(cal_res)\n",
    "\n",
    "fit_voom <- function(feature){\n",
    "    v <- memo_voom(feature)\n",
    "    modQsva <- memo_qsvModel(feature)\n",
    "    fit0 <- lmFit(v, modQsva)\n",
    "    contr.matrix <- makeContrasts(EAvsAA=EA, \n",
    "                                  levels=colnames(modQsva))\n",
    "    fit <- contrasts.fit(fit0, contrasts=contr.matrix)\n",
    "    esv <- eBayes(fit)\n",
    "    return(esv)\n",
    "}\n",
    "\n",
    "memo_efit <- memoise(fit_voom)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential Expression Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for(feature in c('genes', 'transcripts', 'junctions', 'exons')){\n",
    "    dir.create(feature)\n",
    "    # Preform voom\n",
    "    v <- memo_voom(feature)\n",
    "    save(v, file=paste0(feature,'/voomSVA.RData'))\n",
    "    # Fit model and apply eBayes\n",
    "    efit = memo_efit(feature)\n",
    "    # Save differential expression\n",
    "    extract_de(1, \"EAvsAA\", efit, feature)\n",
    "    # Calculate residuals\n",
    "    memo_res(feature)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reproducibility Information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Sys.time()\n",
    "proc.time()\n",
    "options(width = 120)\n",
    "sessioninfo::session_info()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
