{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GO module enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:26:06.948400Z",
     "iopub.status.busy": "2021-07-12T15:26:06.946987Z",
     "iopub.status.idle": "2021-07-12T15:26:07.567895Z",
     "shell.execute_reply": "2021-07-12T15:26:07.567299Z"
    }
   },
   "outputs": [],
   "source": [
    "import functools\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import collections as cx\n",
    "from pybiomart import Dataset\n",
    "from gtfparse import read_gtf\n",
    "from scipy.stats import fisher_exact\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "# GO analysis\n",
    "from goatools.base import download_go_basic_obo\n",
    "from goatools.base import download_ncbi_associations\n",
    "from goatools.obo_parser import GODag\n",
    "from goatools.anno.genetogo_reader import Gene2GoReader\n",
    "from goatools.goea.go_enrichment_ns import GOEnrichmentStudyNS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cached functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:26:07.575583Z",
     "iopub.status.busy": "2021-07-12T15:26:07.574319Z",
     "iopub.status.idle": "2021-07-12T15:26:07.576948Z",
     "shell.execute_reply": "2021-07-12T15:26:07.577970Z"
    }
   },
   "outputs": [],
   "source": [
    "@functools.lru_cache()\n",
    "def get_gtf_genes_df():\n",
    "    gtf_df = read_gtf(\"/ceph/genome/human/gencode25/gtf.CHR/_m/gencode.v25.annotation.gtf\")\n",
    "    return gtf_df[gtf_df[\"feature\"] == \"gene\"][['gene_id', 'gene_name']]\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_wgcna_modules():\n",
    "    return pd.read_csv(\"../../_m/modules.csv\", index_col=0)\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_database():\n",
    "    dataset = Dataset(name=\"hsapiens_gene_ensembl\", \n",
    "                      host=\"http://www.ensembl.org\",\n",
    "                      use_cache=True)\n",
    "    db = dataset.query(attributes=[\"ensembl_gene_id\", \n",
    "                                   \"external_gene_name\", \n",
    "                                   \"entrezgene_id\"], \n",
    "                       use_attr_names=True).dropna(subset=['entrezgene_id'])\n",
    "    return db"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:26:07.589717Z",
     "iopub.status.busy": "2021-07-12T15:26:07.588434Z",
     "iopub.status.idle": "2021-07-12T15:26:07.591014Z",
     "shell.execute_reply": "2021-07-12T15:26:07.592034Z"
    }
   },
   "outputs": [],
   "source": [
    "def convert2entrez(mod):\n",
    "    df = get_wgcna_modules()[(get_wgcna_modules().module) == mod].copy()\n",
    "    df[\"ensemblID\"] = df.index.str.replace(\"\\\\..*\", \"\", regex=True)\n",
    "    return df.merge(get_database(), left_on='ensemblID', \n",
    "                    right_on='ensembl_gene_id')\n",
    "\n",
    "\n",
    "def obo_annotation(alpha=0.05):\n",
    "    # database annotation\n",
    "    fn_obo = download_go_basic_obo()\n",
    "    fn_gene2go = download_ncbi_associations() # must be gunzip to work\n",
    "    obodag = GODag(fn_obo) # downloads most up-to-date\n",
    "    anno_hs = Gene2GoReader(fn_gene2go, taxids=[9606])\n",
    "    # get associations\n",
    "    ns2assoc = anno_hs.get_ns2assc()\n",
    "    for nspc, id2gos in ns2assoc.items():\n",
    "        print(\"{NS} {N:,} annotated human genes\".format(NS=nspc, N=len(id2gos)))\n",
    "    goeaobj = GOEnrichmentStudyNS(\n",
    "        get_database()['entrezgene_id'], # List of human genes with entrez IDs\n",
    "        ns2assoc, # geneid/GO associations\n",
    "        obodag, # Ontologies\n",
    "        propagate_counts = False,\n",
    "        alpha = alpha, # default significance cut-off\n",
    "        methods = ['fdr_bh'])\n",
    "    return goeaobj\n",
    "\n",
    "\n",
    "def run_goea(mod):\n",
    "    df = convert2entrez(mod)\n",
    "    geneids_study = {z[0]:z[1] for z in zip(df['entrezgene_id'], df['external_gene_name'])}\n",
    "    goeaobj = obo_annotation()\n",
    "    goea_results_all = goeaobj.run_study(geneids_study)\n",
    "    goea_results_sig = [r for r in goea_results_all if r.p_fdr_bh < 0.05]\n",
    "    ctr = cx.Counter([r.NS for r in goea_results_sig])\n",
    "    print('Significant results[{TOTAL}] = {BP} BP + {MF} MF + {CC} CC'.format(\n",
    "        TOTAL=len(goea_results_sig),\n",
    "        BP=ctr['BP'],  # biological_process\n",
    "        MF=ctr['MF'],  # molecular_function\n",
    "        CC=ctr['CC'])) # cellular_component\n",
    "    goeaobj.wr_xlsx(\"GO_analysis_module_%s.xlsx\" % mod, goea_results_sig)\n",
    "    goeaobj.wr_txt(\"GO_analysis_module_%s.txt\" % mod, goea_results_sig)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:26:07.597440Z",
     "iopub.status.busy": "2021-07-12T15:26:07.596163Z",
     "iopub.status.idle": "2021-07-12T15:27:23.121643Z",
     "shell.execute_reply": "2021-07-12T15:27:23.122707Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_status', 'gene_name', 'level', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_status', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'protein_id', 'ccdsid']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id gene_name\n",
       "0   ENSG00000223972.5   DDX11L1\n",
       "12  ENSG00000227232.5    WASH7P"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf = get_gtf_genes_df()\n",
    "gtf.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load WGCNA module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:27:23.129013Z",
     "iopub.status.busy": "2021-07-12T15:27:23.127762Z",
     "iopub.status.idle": "2021-07-12T15:27:23.194910Z",
     "shell.execute_reply": "2021-07-12T15:27:23.193835Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>grey</td>\n",
       "      <td>ENSG00000279457.3</td>\n",
       "      <td>FO538757.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    module            gene_id   gene_name\n",
       "12    grey  ENSG00000227232.5      WASH7P\n",
       "139   grey  ENSG00000279457.3  FO538757.1"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df = get_wgcna_modules().merge(gtf, left_index=True, right_on=\"gene_id\", how=\"left\")\n",
    "wgcna_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:27:23.217287Z",
     "iopub.status.busy": "2021-07-12T15:27:23.208415Z",
     "iopub.status.idle": "2021-07-12T15:27:23.225181Z",
     "shell.execute_reply": "2021-07-12T15:27:23.226367Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [module, gene_id, gene_name]\n",
       "Index: []"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_id.str.startswith(\"chr\"))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:27:23.239610Z",
     "iopub.status.busy": "2021-07-12T15:27:23.235226Z",
     "iopub.status.idle": "2021-07-12T15:27:23.242594Z",
     "shell.execute_reply": "2021-07-12T15:27:23.243650Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1445279</th>\n",
       "      <td>purple</td>\n",
       "      <td>ENSG00000149295.13</td>\n",
       "      <td>DRD2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         module             gene_id gene_name\n",
       "1445279  purple  ENSG00000149295.13      DRD2"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_name == 'DRD2')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:27:23.249612Z",
     "iopub.status.busy": "2021-07-12T15:27:23.248349Z",
     "iopub.status.idle": "2021-07-12T15:27:23.310463Z",
     "shell.execute_reply": "2021-07-12T15:27:23.311019Z"
    }
   },
   "outputs": [],
   "source": [
    "wgcna_df.to_csv(\"module_annotated.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GO enrichment for each cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:27:23.316148Z",
     "iopub.status.busy": "2021-07-12T15:27:23.315525Z",
     "iopub.status.idle": "2021-07-12T15:32:51.989064Z",
     "shell.execute_reply": "2021-07-12T15:32:51.988242Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "requests.get(http://purl.obolibrary.org/obo/go/go-basic.obo, stream=True)\n",
      "  WROTE: go-basic.obo\n",
      "\n",
      "FTP RETR ftp.ncbi.nlm.nih.gov gene/DATA gene2go.gz -> gene2go.gz\n",
      "  gunzip gene2go.gz\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.481962 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 8079 IDs ... 75%  6,032 of  8,079 study items found in association\n",
      "100%  8,079 of  8,079 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   8,465 GO terms are associated with  6,032 of  8,079 study items\n",
      "  METHOD fdr_bh:\n",
      "      34 GO terms found significant (< 0.05=alpha) ( 30 enriched +   4 purified): statsmodels fdr_bh\n",
      "   1,853 study items associated with significant GO IDs (enriched)\n",
      "     205 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 8079 IDs ... 79%  6,407 of  8,079 study items found in association\n",
      "100%  8,079 of  8,079 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "   1,300 GO terms are associated with  6,407 of  8,079 study items\n",
      "  METHOD fdr_bh:\n",
      "      46 GO terms found significant (< 0.05=alpha) ( 43 enriched +   3 purified): statsmodels fdr_bh\n",
      "   5,860 study items associated with significant GO IDs (enriched)\n",
      "      34 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 8079 IDs ... 77%  6,193 of  8,079 study items found in association\n",
      "100%  8,079 of  8,079 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   2,967 GO terms are associated with  6,193 of  8,079 study items\n",
      "  METHOD fdr_bh:\n",
      "      32 GO terms found significant (< 0.05=alpha) ( 28 enriched +   4 purified): statsmodels fdr_bh\n",
      "   5,327 study items associated with significant GO IDs (enriched)\n",
      "     119 study items associated with significant GO IDs (purified)\n",
      "Significant results[112] = 34 BP + 32 MF + 46 CC\n",
      "    112 items WROTE: GO_analysis_module_grey.xlsx\n",
      "    112 GOEA results for  6488 study items. WROTE: GO_analysis_module_grey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.118964 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1092 IDs ... 78%    850 of  1,092 study items found in association\n",
      "100%  1,092 of  1,092 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   3,124 GO terms are associated with    850 of  1,092 study items\n",
      "  METHOD fdr_bh:\n",
      "      64 GO terms found significant (< 0.05=alpha) ( 61 enriched +   3 purified): statsmodels fdr_bh\n",
      "     377 study items associated with significant GO IDs (enriched)\n",
      "       1 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1092 IDs ... 84%    912 of  1,092 study items found in association\n",
      "100%  1,092 of  1,092 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     550 GO terms are associated with    912 of  1,092 study items\n",
      "  METHOD fdr_bh:\n",
      "      67 GO terms found significant (< 0.05=alpha) ( 65 enriched +   2 purified): statsmodels fdr_bh\n",
      "     726 study items associated with significant GO IDs (enriched)\n",
      "      50 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1092 IDs ... 80%    873 of  1,092 study items found in association\n",
      "100%  1,092 of  1,092 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     856 GO terms are associated with    873 of  1,092 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 27 enriched +   3 purified): statsmodels fdr_bh\n",
      "     726 study items associated with significant GO IDs (enriched)\n",
      "      20 study items associated with significant GO IDs (purified)\n",
      "Significant results[161] = 64 BP + 30 MF + 67 CC\n",
      "    161 items WROTE: GO_analysis_module_blue.xlsx\n",
      "    161 GOEA results for   892 study items. WROTE: GO_analysis_module_blue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.659532 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 223 IDs ... 96%    213 of    223 study items found in association\n",
      "100%    223 of    223 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,042 GO terms are associated with    213 of    223 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      37 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 223 IDs ... 98%    219 of    223 study items found in association\n",
      "100%    223 of    223 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     347 GO terms are associated with    219 of    223 study items\n",
      "  METHOD fdr_bh:\n",
      "      47 GO terms found significant (< 0.05=alpha) ( 47 enriched +   0 purified): statsmodels fdr_bh\n",
      "     212 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 223 IDs ... 96%    215 of    223 study items found in association\n",
      "100%    223 of    223 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     407 GO terms are associated with    215 of    223 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "     191 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[60] = 5 BP + 8 MF + 47 CC\n",
      "     60 items WROTE: GO_analysis_module_cyan.xlsx\n",
      "     60 GOEA results for   221 study items. WROTE: GO_analysis_module_cyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.782061 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 56 IDs ... 91%     51 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     339 GO terms are associated with     51 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "      35 GO terms found significant (< 0.05=alpha) ( 35 enriched +   0 purified): statsmodels fdr_bh\n",
      "      39 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 56 IDs ... 95%     53 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      89 GO terms are associated with     53 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      42 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 56 IDs ... 96%     54 of     56 study items found in association\n",
      "100%     56 of     56 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     113 GO terms are associated with     54 of     56 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) ( 12 enriched +   0 purified): statsmodels fdr_bh\n",
      "      51 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[52] = 35 BP + 12 MF + 5 CC\n",
      "     52 items WROTE: GO_analysis_module_violet.xlsx\n",
      "     52 GOEA results for    52 study items. WROTE: GO_analysis_module_violet.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.629960 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 225 IDs ... 89%    201 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     900 GO terms are associated with    201 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       4 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 225 IDs ... 92%    208 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     283 GO terms are associated with    208 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "     169 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 225 IDs ... 95%    214 of    225 study items found in association\n",
      "100%    225 of    225 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     328 GO terms are associated with    214 of    225 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     187 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[9] = 1 BP + 2 MF + 6 CC\n",
      "      9 items WROTE: GO_analysis_module_tan.xlsx\n",
      "      9 GOEA results for   207 study items. WROTE: GO_analysis_module_tan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.232856 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 112 IDs ... 86%     96 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     489 GO terms are associated with     96 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 112 IDs ... 89%    100 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     168 GO terms are associated with    100 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      17 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 112 IDs ... 96%    107 of    112 study items found in association\n",
      "100%    112 of    112 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     193 GO terms are associated with    107 of    112 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      90 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 2 MF + 1 CC\n",
      "      3 items WROTE: GO_analysis_module_darkred.xlsx\n",
      "      3 GOEA results for    93 study items. WROTE: GO_analysis_module_darkred.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.274700 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 101 IDs ... 86%     87 of    101 study items found in association\n",
      "100%    101 of    101 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     508 GO terms are associated with     87 of    101 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      15 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 101 IDs ... 90%     91 of    101 study items found in association\n",
      "100%    101 of    101 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     186 GO terms are associated with     91 of    101 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      45 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 101 IDs ... 95%     96 of    101 study items found in association\n",
      "100%    101 of    101 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     198 GO terms are associated with     96 of    101 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      83 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[9] = 2 BP + 4 MF + 3 CC\n",
      "      9 items WROTE: GO_analysis_module_darkgreen.xlsx\n",
      "      9 GOEA results for    87 study items. WROTE: GO_analysis_module_darkgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.786410 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1681 IDs ... 88%  1,486 of  1,681 study items found in association\n",
      "100%  1,681 of  1,681 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   4,526 GO terms are associated with  1,486 of  1,681 study items\n",
      "  METHOD fdr_bh:\n",
      "      66 GO terms found significant (< 0.05=alpha) ( 63 enriched +   3 purified): statsmodels fdr_bh\n",
      "     747 study items associated with significant GO IDs (enriched)\n",
      "      14 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1681 IDs ... 93%  1,556 of  1,681 study items found in association\n",
      "100%  1,681 of  1,681 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     837 GO terms are associated with  1,556 of  1,681 study items\n",
      "  METHOD fdr_bh:\n",
      "      95 GO terms found significant (< 0.05=alpha) ( 95 enriched +   0 purified): statsmodels fdr_bh\n",
      "   1,501 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1681 IDs ... 91%  1,532 of  1,681 study items found in association\n",
      "100%  1,681 of  1,681 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   1,423 GO terms are associated with  1,532 of  1,681 study items\n",
      "  METHOD fdr_bh:\n",
      "      42 GO terms found significant (< 0.05=alpha) ( 39 enriched +   3 purified): statsmodels fdr_bh\n",
      "   1,384 study items associated with significant GO IDs (enriched)\n",
      "       9 study items associated with significant GO IDs (purified)\n",
      "Significant results[203] = 66 BP + 42 MF + 95 CC\n",
      "    203 items WROTE: GO_analysis_module_turquoise.xlsx\n",
      "    203 GOEA results for  1578 study items. WROTE: GO_analysis_module_turquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.451831 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 222 IDs ... 88%    195 of    222 study items found in association\n",
      "100%    222 of    222 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,128 GO terms are associated with    195 of    222 study items\n",
      "  METHOD fdr_bh:\n",
      "      15 GO terms found significant (< 0.05=alpha) ( 15 enriched +   0 purified): statsmodels fdr_bh\n",
      "      37 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 222 IDs ... 90%    200 of    222 study items found in association\n",
      "100%    222 of    222 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     309 GO terms are associated with    200 of    222 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 26 enriched +   0 purified): statsmodels fdr_bh\n",
      "     151 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 222 IDs ... 87%    194 of    222 study items found in association\n",
      "100%    222 of    222 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     365 GO terms are associated with    194 of    222 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "     152 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[50] = 15 BP + 9 MF + 26 CC\n",
      "     50 items WROTE: GO_analysis_module_purple.xlsx\n",
      "     50 GOEA results for   187 study items. WROTE: GO_analysis_module_purple.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.591606 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 352 IDs ... 83%    292 of    352 study items found in association\n",
      "100%    352 of    352 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,360 GO terms are associated with    292 of    352 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      38 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 352 IDs ... 89%    315 of    352 study items found in association\n",
      "100%    352 of    352 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     357 GO terms are associated with    315 of    352 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 25 enriched +   1 purified): statsmodels fdr_bh\n",
      "     252 study items associated with significant GO IDs (enriched)\n",
      "       6 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 352 IDs ... 85%    299 of    352 study items found in association\n",
      "100%    352 of    352 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     436 GO terms are associated with    299 of    352 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "     234 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[37] = 7 BP + 4 MF + 26 CC\n",
      "     37 items WROTE: GO_analysis_module_green.xlsx\n",
      "     37 GOEA results for   295 study items. WROTE: GO_analysis_module_green.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.719302 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 81 IDs ... 88%     71 of     81 study items found in association\n",
      "100%     81 of     81 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     499 GO terms are associated with     71 of     81 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 81 IDs ... 93%     75 of     81 study items found in association\n",
      "100%     81 of     81 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     119 GO terms are associated with     75 of     81 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      57 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 81 IDs ... 88%     71 of     81 study items found in association\n",
      "100%     81 of     81 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     178 GO terms are associated with     71 of     81 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 0 BP + 0 MF + 4 CC\n",
      "      4 items WROTE: GO_analysis_module_white.xlsx\n",
      "      4 GOEA results for    57 study items. WROTE: GO_analysis_module_white.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.388450 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 416 IDs ... 89%    370 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,968 GO terms are associated with    370 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      63 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 416 IDs ... 94%    390 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     347 GO terms are associated with    390 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "      32 GO terms found significant (< 0.05=alpha) ( 32 enriched +   0 purified): statsmodels fdr_bh\n",
      "     359 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 416 IDs ... 90%    376 of    416 study items found in association\n",
      "100%    416 of    416 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     666 GO terms are associated with    376 of    416 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "     307 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[49] = 7 BP + 10 MF + 32 CC\n",
      "     49 items WROTE: GO_analysis_module_yellow.xlsx\n",
      "     49 GOEA results for   383 study items. WROTE: GO_analysis_module_yellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.041970 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 330 IDs ... 91%    300 of    330 study items found in association\n",
      "100%    330 of    330 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,539 GO terms are associated with    300 of    330 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      28 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 330 IDs ... 93%    307 of    330 study items found in association\n",
      "100%    330 of    330 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     329 GO terms are associated with    307 of    330 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 26 enriched +   0 purified): statsmodels fdr_bh\n",
      "     208 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 330 IDs ... 91%    299 of    330 study items found in association\n",
      "100%    330 of    330 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     500 GO terms are associated with    299 of    330 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     214 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[30] = 2 BP + 2 MF + 26 CC\n",
      "     30 items WROTE: GO_analysis_module_red.xlsx\n",
      "     30 GOEA results for   289 study items. WROTE: GO_analysis_module_red.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.558729 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 73 IDs ... 97%     71 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     213 GO terms are associated with     71 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "      14 GO terms found significant (< 0.05=alpha) ( 14 enriched +   0 purified): statsmodels fdr_bh\n",
      "      54 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 73 IDs ... 97%     71 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      96 GO terms are associated with     71 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 22 enriched +   0 purified): statsmodels fdr_bh\n",
      "      68 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 73 IDs ... 96%     70 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      77 GO terms are associated with     70 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "      68 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[46] = 14 BP + 10 MF + 22 CC\n",
      "     46 items WROTE: GO_analysis_module_skyblue.xlsx\n",
      "     46 GOEA results for    71 study items. WROTE: GO_analysis_module_skyblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.925874 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 52 IDs ... 94%     49 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     306 GO terms are associated with     49 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "       8 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 52 IDs ... 98%     51 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     146 GO terms are associated with     51 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "      48 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 52 IDs ... 98%     51 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     125 GO terms are associated with     51 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      47 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[23] = 3 BP + 4 MF + 16 CC\n",
      "     23 items WROTE: GO_analysis_module_darkmagenta.xlsx\n",
      "     23 GOEA results for    51 study items. WROTE: GO_analysis_module_darkmagenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.505284 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 72 IDs ... 86%     62 of     72 study items found in association\n",
      "100%     72 of     72 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     459 GO terms are associated with     62 of     72 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 72 IDs ... 89%     64 of     72 study items found in association\n",
      "100%     72 of     72 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     120 GO terms are associated with     64 of     72 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      29 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 72 IDs ... 82%     59 of     72 study items found in association\n",
      "100%     72 of     72 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     178 GO terms are associated with     59 of     72 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 0 MF + 3 CC\n",
      "      3 items WROTE: GO_analysis_module_saddlebrown.xlsx\n",
      "      3 GOEA results for    29 study items. WROTE: GO_analysis_module_saddlebrown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.707804 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 100 IDs ... 94%     94 of    100 study items found in association\n",
      "100%    100 of    100 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     978 GO terms are associated with     94 of    100 study items\n",
      "  METHOD fdr_bh:\n",
      "      47 GO terms found significant (< 0.05=alpha) ( 47 enriched +   0 purified): statsmodels fdr_bh\n",
      "      68 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 100 IDs ... 95%     95 of    100 study items found in association\n",
      "100%    100 of    100 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     180 GO terms are associated with     95 of    100 study items\n",
      "  METHOD fdr_bh:\n",
      "      18 GO terms found significant (< 0.05=alpha) ( 18 enriched +   0 purified): statsmodels fdr_bh\n",
      "      77 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 100 IDs ... 95%     95 of    100 study items found in association\n",
      "100%    100 of    100 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     296 GO terms are associated with     95 of    100 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      78 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[70] = 47 BP + 5 MF + 18 CC\n",
      "     70 items WROTE: GO_analysis_module_darkturquoise.xlsx\n",
      "     70 GOEA results for    95 study items. WROTE: GO_analysis_module_darkturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.362310 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 164 IDs ... 92%    151 of    164 study items found in association\n",
      "100%    164 of    164 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,232 GO terms are associated with    151 of    164 study items\n",
      "  METHOD fdr_bh:\n",
      "     182 GO terms found significant (< 0.05=alpha) (182 enriched +   0 purified): statsmodels fdr_bh\n",
      "     136 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 164 IDs ... 98%    160 of    164 study items found in association\n",
      "100%    164 of    164 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     200 GO terms are associated with    160 of    164 study items\n",
      "  METHOD fdr_bh:\n",
      "      41 GO terms found significant (< 0.05=alpha) ( 41 enriched +   0 purified): statsmodels fdr_bh\n",
      "     151 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 164 IDs ... 93%    153 of    164 study items found in association\n",
      "100%    164 of    164 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     304 GO terms are associated with    153 of    164 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 21 enriched +   0 purified): statsmodels fdr_bh\n",
      "     137 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[244] = 182 BP + 21 MF + 41 CC\n",
      "    244 items WROTE: GO_analysis_module_grey60.xlsx\n",
      "    244 GOEA results for   160 study items. WROTE: GO_analysis_module_grey60.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.592357 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 898 IDs ... 83%    744 of    898 study items found in association\n",
      "100%    898 of    898 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,858 GO terms are associated with    744 of    898 study items\n",
      "  METHOD fdr_bh:\n",
      "      57 GO terms found significant (< 0.05=alpha) ( 55 enriched +   2 purified): statsmodels fdr_bh\n",
      "     302 study items associated with significant GO IDs (enriched)\n",
      "       3 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 898 IDs ... 88%    792 of    898 study items found in association\n",
      "100%    898 of    898 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     552 GO terms are associated with    792 of    898 study items\n",
      "  METHOD fdr_bh:\n",
      "      71 GO terms found significant (< 0.05=alpha) ( 71 enriched +   0 purified): statsmodels fdr_bh\n",
      "     730 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 898 IDs ... 85%    766 of    898 study items found in association\n",
      "100%    898 of    898 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     858 GO terms are associated with    766 of    898 study items\n",
      "  METHOD fdr_bh:\n",
      "      31 GO terms found significant (< 0.05=alpha) ( 30 enriched +   1 purified): statsmodels fdr_bh\n",
      "     602 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[159] = 57 BP + 31 MF + 71 CC\n",
      "    159 items WROTE: GO_analysis_module_brown.xlsx\n",
      "    159 GOEA results for   799 study items. WROTE: GO_analysis_module_brown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.085673 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 48 IDs ... 90%     43 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     371 GO terms are associated with     43 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 48 IDs ... 94%     45 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     136 GO terms are associated with     45 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) ( 12 enriched +   0 purified): statsmodels fdr_bh\n",
      "      29 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 48 IDs ... 94%     45 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     112 GO terms are associated with     45 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      36 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[13] = 0 BP + 1 MF + 12 CC\n",
      "     13 items WROTE: GO_analysis_module_sienna3.xlsx\n",
      "     13 GOEA results for    43 study items. WROTE: GO_analysis_module_sienna3.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.711722 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 196 IDs ... 92%    180 of    196 study items found in association\n",
      "100%    196 of    196 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,462 GO terms are associated with    180 of    196 study items\n",
      "  METHOD fdr_bh:\n",
      "      38 GO terms found significant (< 0.05=alpha) ( 38 enriched +   0 purified): statsmodels fdr_bh\n",
      "      81 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 196 IDs ... 93%    182 of    196 study items found in association\n",
      "100%    196 of    196 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     242 GO terms are associated with    182 of    196 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) ( 12 enriched +   0 purified): statsmodels fdr_bh\n",
      "     147 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 196 IDs ... 91%    179 of    196 study items found in association\n",
      "100%    196 of    196 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     359 GO terms are associated with    179 of    196 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 11 enriched +   0 purified): statsmodels fdr_bh\n",
      "     143 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[61] = 38 BP + 11 MF + 12 CC\n",
      "     61 items WROTE: GO_analysis_module_lightcyan.xlsx\n",
      "     61 GOEA results for   170 study items. WROTE: GO_analysis_module_lightcyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.085548 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 154 IDs ... 91%    140 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     924 GO terms are associated with    140 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 30 enriched +   0 purified): statsmodels fdr_bh\n",
      "      57 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 154 IDs ... 95%    146 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     246 GO terms are associated with    146 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      27 GO terms found significant (< 0.05=alpha) ( 27 enriched +   0 purified): statsmodels fdr_bh\n",
      "     137 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 154 IDs ... 96%    148 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     309 GO terms are associated with    148 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 21 enriched +   0 purified): statsmodels fdr_bh\n",
      "     136 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[78] = 30 BP + 21 MF + 27 CC\n",
      "     78 items WROTE: GO_analysis_module_lightgreen.xlsx\n",
      "     78 GOEA results for   149 study items. WROTE: GO_analysis_module_lightgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.010068 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 138 IDs ... 94%    130 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     855 GO terms are associated with    130 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      33 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 138 IDs ... 96%    133 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     263 GO terms are associated with    133 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "      19 GO terms found significant (< 0.05=alpha) ( 19 enriched +   0 purified): statsmodels fdr_bh\n",
      "     124 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 138 IDs ... 94%    130 of    138 study items found in association\n",
      "100%    138 of    138 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     290 GO terms are associated with    130 of    138 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     122 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[31] = 7 BP + 5 MF + 19 CC\n",
      "     31 items WROTE: GO_analysis_module_royalblue.xlsx\n",
      "     31 GOEA results for   133 study items. WROTE: GO_analysis_module_royalblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.314044 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 219 IDs ... 89%    196 of    219 study items found in association\n",
      "100%    219 of    219 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     948 GO terms are associated with    196 of    219 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      35 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 219 IDs ... 94%    206 of    219 study items found in association\n",
      "100%    219 of    219 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     319 GO terms are associated with    206 of    219 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 20 enriched +   1 purified): statsmodels fdr_bh\n",
      "     182 study items associated with significant GO IDs (enriched)\n",
      "       2 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 219 IDs ... 91%    199 of    219 study items found in association\n",
      "100%    219 of    219 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     351 GO terms are associated with    199 of    219 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     178 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[30] = 4 BP + 5 MF + 21 CC\n",
      "     30 items WROTE: GO_analysis_module_salmon.xlsx\n",
      "     30 GOEA results for   201 study items. WROTE: GO_analysis_module_salmon.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:06.446500 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 306 IDs ... 93%    286 of    306 study items found in association\n",
      "100%    306 of    306 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,249 GO terms are associated with    286 of    306 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "     104 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 306 IDs ... 97%    298 of    306 study items found in association\n",
      "100%    306 of    306 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     352 GO terms are associated with    298 of    306 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 15 enriched +   1 purified): statsmodels fdr_bh\n",
      "     256 study items associated with significant GO IDs (enriched)\n",
      "       1 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 306 IDs ... 93%    285 of    306 study items found in association\n",
      "100%    306 of    306 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     365 GO terms are associated with    285 of    306 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 17 enriched +   0 purified): statsmodels fdr_bh\n",
      "     262 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[40] = 7 BP + 17 MF + 16 CC\n",
      "     40 items WROTE: GO_analysis_module_pink.xlsx\n",
      "     40 GOEA results for   285 study items. WROTE: GO_analysis_module_pink.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:07.310902 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 154 IDs ... 93%    143 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,383 GO terms are associated with    143 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      63 GO terms found significant (< 0.05=alpha) ( 63 enriched +   0 purified): statsmodels fdr_bh\n",
      "     103 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 154 IDs ... 95%    146 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     209 GO terms are associated with    146 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "     135 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 154 IDs ... 93%    143 of    154 study items found in association\n",
      "100%    154 of    154 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     288 GO terms are associated with    143 of    154 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     118 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[84] = 63 BP + 5 MF + 16 CC\n",
      "     84 items WROTE: GO_analysis_module_lightyellow.xlsx\n",
      "     84 GOEA results for   141 study items. WROTE: GO_analysis_module_lightyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:06.333858 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 261 IDs ... 89%    232 of    261 study items found in association\n",
      "100%    261 of    261 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,402 GO terms are associated with    232 of    261 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      23 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 261 IDs ... 95%    248 of    261 study items found in association\n",
      "100%    261 of    261 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     382 GO terms are associated with    248 of    261 study items\n",
      "  METHOD fdr_bh:\n",
      "      24 GO terms found significant (< 0.05=alpha) ( 24 enriched +   0 purified): statsmodels fdr_bh\n",
      "     225 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 261 IDs ... 95%    249 of    261 study items found in association\n",
      "100%    261 of    261 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     419 GO terms are associated with    249 of    261 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     219 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[32] = 3 BP + 5 MF + 24 CC\n",
      "     32 items WROTE: GO_analysis_module_magenta.xlsx\n",
      "     32 GOEA results for   243 study items. WROTE: GO_analysis_module_magenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:06.677788 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 93 IDs ... 97%     90 of     93 study items found in association\n",
      "100%     93 of     93 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     471 GO terms are associated with     90 of     93 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       3 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 93 IDs ... 95%     88 of     93 study items found in association\n",
      "100%     93 of     93 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     176 GO terms are associated with     88 of     93 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      47 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 93 IDs ... 90%     84 of     93 study items found in association\n",
      "100%     93 of     93 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     191 GO terms are associated with     84 of     93 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      65 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 1 BP + 1 MF + 2 CC\n",
      "      4 items WROTE: GO_analysis_module_orange.xlsx\n",
      "      4 GOEA results for    74 study items. WROTE: GO_analysis_module_orange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.500018 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 310 IDs ... 94%    292 of    310 study items found in association\n",
      "100%    310 of    310 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,177 GO terms are associated with    292 of    310 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 17 enriched +   0 purified): statsmodels fdr_bh\n",
      "     152 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 310 IDs ... 96%    298 of    310 study items found in association\n",
      "100%    310 of    310 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     362 GO terms are associated with    298 of    310 study items\n",
      "  METHOD fdr_bh:\n",
      "      33 GO terms found significant (< 0.05=alpha) ( 31 enriched +   2 purified): statsmodels fdr_bh\n",
      "     279 study items associated with significant GO IDs (enriched)\n",
      "       8 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 310 IDs ... 96%    297 of    310 study items found in association\n",
      "100%    310 of    310 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     370 GO terms are associated with    297 of    310 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 17 enriched +   0 purified): statsmodels fdr_bh\n",
      "     267 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[67] = 17 BP + 17 MF + 33 CC\n",
      "     67 items WROTE: GO_analysis_module_black.xlsx\n",
      "     67 GOEA results for   299 study items. WROTE: GO_analysis_module_black.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.705588 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 61 IDs ... 92%     56 of     61 study items found in association\n",
      "100%     61 of     61 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     405 GO terms are associated with     56 of     61 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       5 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 61 IDs ... 92%     56 of     61 study items found in association\n",
      "100%     61 of     61 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      75 GO terms are associated with     56 of     61 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 61 IDs ... 97%     59 of     61 study items found in association\n",
      "100%     61 of     61 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     149 GO terms are associated with     59 of     61 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      44 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 1 BP + 2 MF + 0 CC\n",
      "      3 items WROTE: GO_analysis_module_paleturquoise.xlsx\n",
      "      3 GOEA results for    44 study items. WROTE: GO_analysis_module_paleturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.302565 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 52 IDs ... 94%     49 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     511 GO terms are associated with     49 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) ( 12 enriched +   0 purified): statsmodels fdr_bh\n",
      "      25 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 52 IDs ... 96%     50 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      86 GO terms are associated with     50 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      27 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 52 IDs ... 98%     51 of     52 study items found in association\n",
      "100%     52 of     52 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     121 GO terms are associated with     51 of     52 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "      47 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[24] = 12 BP + 9 MF + 3 CC\n",
      "     24 items WROTE: GO_analysis_module_darkolivegreen.xlsx\n",
      "     24 GOEA results for    48 study items. WROTE: GO_analysis_module_darkolivegreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.173565 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 230 IDs ... 87%    200 of    230 study items found in association\n",
      "100%    230 of    230 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     964 GO terms are associated with    200 of    230 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      16 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 230 IDs ... 93%    214 of    230 study items found in association\n",
      "100%    230 of    230 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     241 GO terms are associated with    214 of    230 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     188 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 230 IDs ... 91%    209 of    230 study items found in association\n",
      "100%    230 of    230 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     359 GO terms are associated with    209 of    230 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "     185 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[15] = 1 BP + 9 MF + 5 CC\n",
      "     15 items WROTE: GO_analysis_module_greenyellow.xlsx\n",
      "     15 GOEA results for   212 study items. WROTE: GO_analysis_module_greenyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.732825 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 87 IDs ... 94%     82 of     87 study items found in association\n",
      "100%     87 of     87 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     579 GO terms are associated with     82 of     87 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      17 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 87 IDs ... 95%     83 of     87 study items found in association\n",
      "100%     87 of     87 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     141 GO terms are associated with     83 of     87 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "      61 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 87 IDs ... 92%     80 of     87 study items found in association\n",
      "100%     87 of     87 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     219 GO terms are associated with     80 of     87 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      17 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[26] = 5 BP + 5 MF + 16 CC\n",
      "     26 items WROTE: GO_analysis_module_darkorange.xlsx\n",
      "     26 GOEA results for    64 study items. WROTE: GO_analysis_module_darkorange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.809330 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 209 IDs ... 89%    185 of    209 study items found in association\n",
      "100%    209 of    209 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     987 GO terms are associated with    185 of    209 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      36 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 209 IDs ... 95%    198 of    209 study items found in association\n",
      "100%    209 of    209 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     261 GO terms are associated with    198 of    209 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 17 enriched +   0 purified): statsmodels fdr_bh\n",
      "     178 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 209 IDs ... 93%    194 of    209 study items found in association\n",
      "100%    209 of    209 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     327 GO terms are associated with    194 of    209 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "     160 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[32] = 7 BP + 8 MF + 17 CC\n",
      "     32 items WROTE: GO_analysis_module_midnightblue.xlsx\n",
      "     32 GOEA results for   196 study items. WROTE: GO_analysis_module_midnightblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:06.279096 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 66 IDs ... 88%     58 of     66 study items found in association\n",
      "100%     66 of     66 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     341 GO terms are associated with     58 of     66 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 66 IDs ... 91%     60 of     66 study items found in association\n",
      "100%     66 of     66 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      96 GO terms are associated with     60 of     66 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 66 IDs ... 89%     59 of     66 study items found in association\n",
      "100%     66 of     66 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     129 GO terms are associated with     59 of     66 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_steelblue.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_steelblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.367815 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 84 IDs ... 81%     68 of     84 study items found in association\n",
      "100%     84 of     84 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     487 GO terms are associated with     68 of     84 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 84 IDs ... 89%     75 of     84 study items found in association\n",
      "100%     84 of     84 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     129 GO terms are associated with     75 of     84 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 84 IDs ... 82%     69 of     84 study items found in association\n",
      "100%     84 of     84 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     163 GO terms are associated with     69 of     84 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_darkgrey.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_darkgrey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.323938 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 26 IDs ...100%     26 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     198 GO terms are associated with     26 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "       7 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 26 IDs ...100%     26 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      77 GO terms are associated with     26 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      17 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 26 IDs ...100%     26 of     26 study items found in association\n",
      "100%     26 of     26 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      79 GO terms are associated with     26 of     26 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[7] = 3 BP + 0 MF + 4 CC\n",
      "      7 items WROTE: GO_analysis_module_yellowgreen.xlsx\n",
      "      7 GOEA results for    17 study items. WROTE: GO_analysis_module_yellowgreen.txt\n"
     ]
    }
   ],
   "source": [
    "for mod in get_wgcna_modules().module.unique():\n",
    "    run_goea(mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
