{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GO module enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:13:18.581813Z",
     "iopub.status.busy": "2021-07-12T15:13:18.581117Z",
     "iopub.status.idle": "2021-07-12T15:13:19.276552Z",
     "shell.execute_reply": "2021-07-12T15:13:19.274792Z"
    }
   },
   "outputs": [],
   "source": [
    "import functools\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import collections as cx\n",
    "from pybiomart import Dataset\n",
    "from gtfparse import read_gtf\n",
    "from scipy.stats import fisher_exact\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "# GO analysis\n",
    "from goatools.base import download_go_basic_obo\n",
    "from goatools.base import download_ncbi_associations\n",
    "from goatools.obo_parser import GODag\n",
    "from goatools.anno.genetogo_reader import Gene2GoReader\n",
    "from goatools.goea.go_enrichment_ns import GOEnrichmentStudyNS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cached functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:13:19.286752Z",
     "iopub.status.busy": "2021-07-12T15:13:19.285382Z",
     "iopub.status.idle": "2021-07-12T15:13:19.288090Z",
     "shell.execute_reply": "2021-07-12T15:13:19.289109Z"
    }
   },
   "outputs": [],
   "source": [
    "@functools.lru_cache()\n",
    "def get_gtf_genes_df():\n",
    "    gtf_df = read_gtf(\"/ceph/genome/human/gencode25/gtf.CHR/_m/gencode.v25.annotation.gtf\")\n",
    "    return gtf_df[gtf_df[\"feature\"] == \"gene\"][['gene_id', 'gene_name']]\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_wgcna_modules():\n",
    "    return pd.read_csv(\"../../_m/modules.csv\", index_col=0)\n",
    "\n",
    "\n",
    "@functools.lru_cache()\n",
    "def get_database():\n",
    "    dataset = Dataset(name=\"hsapiens_gene_ensembl\", \n",
    "                      host=\"http://www.ensembl.org\",\n",
    "                      use_cache=True)\n",
    "    db = dataset.query(attributes=[\"ensembl_gene_id\", \n",
    "                                   \"external_gene_name\", \n",
    "                                   \"entrezgene_id\"], \n",
    "                       use_attr_names=True).dropna(subset=['entrezgene_id'])\n",
    "    return db"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:13:19.302731Z",
     "iopub.status.busy": "2021-07-12T15:13:19.300878Z",
     "iopub.status.idle": "2021-07-12T15:13:19.304618Z",
     "shell.execute_reply": "2021-07-12T15:13:19.306081Z"
    }
   },
   "outputs": [],
   "source": [
    "def convert2entrez(mod):\n",
    "    df = get_wgcna_modules()[(get_wgcna_modules().module) == mod].copy()\n",
    "    df[\"ensemblID\"] = df.index.str.replace(\"\\\\..*\", \"\", regex=True)\n",
    "    return df.merge(get_database(), left_on='ensemblID', \n",
    "                    right_on='ensembl_gene_id')\n",
    "\n",
    "\n",
    "def obo_annotation(alpha=0.05):\n",
    "    # database annotation\n",
    "    fn_obo = download_go_basic_obo()\n",
    "    fn_gene2go = download_ncbi_associations() # must be gunzip to work\n",
    "    obodag = GODag(fn_obo) # downloads most up-to-date\n",
    "    anno_hs = Gene2GoReader(fn_gene2go, taxids=[9606])\n",
    "    # get associations\n",
    "    ns2assoc = anno_hs.get_ns2assc()\n",
    "    for nspc, id2gos in ns2assoc.items():\n",
    "        print(\"{NS} {N:,} annotated human genes\".format(NS=nspc, N=len(id2gos)))\n",
    "    goeaobj = GOEnrichmentStudyNS(\n",
    "        get_database()['entrezgene_id'], # List of human genes with entrez IDs\n",
    "        ns2assoc, # geneid/GO associations\n",
    "        obodag, # Ontologies\n",
    "        propagate_counts = False,\n",
    "        alpha = alpha, # default significance cut-off\n",
    "        methods = ['fdr_bh'])\n",
    "    return goeaobj\n",
    "\n",
    "\n",
    "def run_goea(mod):\n",
    "    df = convert2entrez(mod)\n",
    "    geneids_study = {z[0]:z[1] for z in zip(df['entrezgene_id'], df['external_gene_name'])}\n",
    "    goeaobj = obo_annotation()\n",
    "    goea_results_all = goeaobj.run_study(geneids_study)\n",
    "    goea_results_sig = [r for r in goea_results_all if r.p_fdr_bh < 0.05]\n",
    "    ctr = cx.Counter([r.NS for r in goea_results_sig])\n",
    "    print('Significant results[{TOTAL}] = {BP} BP + {MF} MF + {CC} CC'.format(\n",
    "        TOTAL=len(goea_results_sig),\n",
    "        BP=ctr['BP'],  # biological_process\n",
    "        MF=ctr['MF'],  # molecular_function\n",
    "        CC=ctr['CC'])) # cellular_component\n",
    "    goeaobj.wr_xlsx(\"GO_analysis_module_%s.xlsx\" % mod, goea_results_sig)\n",
    "    goeaobj.wr_txt(\"GO_analysis_module_%s.txt\" % mod, goea_results_sig)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:13:19.315888Z",
     "iopub.status.busy": "2021-07-12T15:13:19.313940Z",
     "iopub.status.idle": "2021-07-12T15:14:43.381366Z",
     "shell.execute_reply": "2021-07-12T15:14:43.382444Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_status', 'gene_name', 'level', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_status', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'protein_id', 'ccdsid']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id gene_name\n",
       "0   ENSG00000223972.5   DDX11L1\n",
       "12  ENSG00000227232.5    WASH7P"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf = get_gtf_genes_df()\n",
    "gtf.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load WGCNA module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:14:43.391350Z",
     "iopub.status.busy": "2021-07-12T15:14:43.390063Z",
     "iopub.status.idle": "2021-07-12T15:14:43.452715Z",
     "shell.execute_reply": "2021-07-12T15:14:43.453876Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>greenyellow</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>brown</td>\n",
       "      <td>ENSG00000278267.1</td>\n",
       "      <td>MIR6859-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         module            gene_id  gene_name\n",
       "12  greenyellow  ENSG00000227232.5     WASH7P\n",
       "25        brown  ENSG00000278267.1  MIR6859-1"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df = get_wgcna_modules().merge(gtf, left_index=True, right_on=\"gene_id\", how=\"left\")\n",
    "wgcna_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:14:43.468344Z",
     "iopub.status.busy": "2021-07-12T15:14:43.466938Z",
     "iopub.status.idle": "2021-07-12T15:14:43.484564Z",
     "shell.execute_reply": "2021-07-12T15:14:43.485592Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [module, gene_id, gene_name]\n",
       "Index: []"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_id.str.startswith(\"chr\"))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:14:43.505343Z",
     "iopub.status.busy": "2021-07-12T15:14:43.500313Z",
     "iopub.status.idle": "2021-07-12T15:14:43.514312Z",
     "shell.execute_reply": "2021-07-12T15:14:43.513603Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1445279</th>\n",
       "      <td>brown</td>\n",
       "      <td>ENSG00000149295.13</td>\n",
       "      <td>DRD2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        module             gene_id gene_name\n",
       "1445279  brown  ENSG00000149295.13      DRD2"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgcna_df[(wgcna_df.gene_name == 'DRD2')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:14:43.520203Z",
     "iopub.status.busy": "2021-07-12T15:14:43.519565Z",
     "iopub.status.idle": "2021-07-12T15:14:43.573748Z",
     "shell.execute_reply": "2021-07-12T15:14:43.573056Z"
    }
   },
   "outputs": [],
   "source": [
    "wgcna_df.to_csv(\"module_annotated.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GO enrichment for each cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-07-12T15:14:43.581100Z",
     "iopub.status.busy": "2021-07-12T15:14:43.579789Z",
     "iopub.status.idle": "2021-07-12T15:21:02.095392Z",
     "shell.execute_reply": "2021-07-12T15:21:02.097491Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "requests.get(http://purl.obolibrary.org/obo/go/go-basic.obo, stream=True)\n",
      "  WROTE: go-basic.obo\n",
      "\n",
      "FTP RETR ftp.ncbi.nlm.nih.gov gene/DATA gene2go.gz -> gene2go.gz\n",
      "  gunzip gene2go.gz\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.345673 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 406 IDs ... 90%    366 of    406 study items found in association\n",
      "100%    406 of    406 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,573 GO terms are associated with    366 of    406 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      29 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 406 IDs ... 93%    379 of    406 study items found in association\n",
      "100%    406 of    406 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     449 GO terms are associated with    379 of    406 study items\n",
      "  METHOD fdr_bh:\n",
      "      32 GO terms found significant (< 0.05=alpha) ( 32 enriched +   0 purified): statsmodels fdr_bh\n",
      "     357 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 406 IDs ... 94%    381 of    406 study items found in association\n",
      "100%    406 of    406 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     557 GO terms are associated with    381 of    406 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "     330 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[38] = 2 BP + 4 MF + 32 CC\n",
      "     38 items WROTE: GO_analysis_module_greenyellow.xlsx\n",
      "     38 GOEA results for   385 study items. WROTE: GO_analysis_module_greenyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:06.186145 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1194 IDs ... 81%    962 of  1,194 study items found in association\n",
      "100%  1,194 of  1,194 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   3,203 GO terms are associated with    962 of  1,194 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  8 enriched +   1 purified): statsmodels fdr_bh\n",
      "     166 study items associated with significant GO IDs (enriched)\n",
      "       1 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1194 IDs ... 85%  1,015 of  1,194 study items found in association\n",
      "100%  1,194 of  1,194 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     641 GO terms are associated with  1,015 of  1,194 study items\n",
      "  METHOD fdr_bh:\n",
      "      21 GO terms found significant (< 0.05=alpha) ( 21 enriched +   0 purified): statsmodels fdr_bh\n",
      "     838 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1194 IDs ... 84%  1,003 of  1,194 study items found in association\n",
      "100%  1,194 of  1,194 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   1,097 GO terms are associated with  1,003 of  1,194 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 15 enriched +   2 purified): statsmodels fdr_bh\n",
      "     864 study items associated with significant GO IDs (enriched)\n",
      "      10 study items associated with significant GO IDs (purified)\n",
      "Significant results[47] = 9 BP + 17 MF + 21 CC\n",
      "     47 items WROTE: GO_analysis_module_brown.xlsx\n",
      "     47 GOEA results for  1007 study items. WROTE: GO_analysis_module_brown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.742197 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 3509 IDs ... 74%  2,609 of  3,509 study items found in association\n",
      "100%  3,509 of  3,509 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   5,348 GO terms are associated with  2,609 of  3,509 study items\n",
      "  METHOD fdr_bh:\n",
      "      12 GO terms found significant (< 0.05=alpha) (  9 enriched +   3 purified): statsmodels fdr_bh\n",
      "     589 study items associated with significant GO IDs (enriched)\n",
      "     103 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 3509 IDs ... 79%  2,773 of  3,509 study items found in association\n",
      "100%  3,509 of  3,509 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     932 GO terms are associated with  2,773 of  3,509 study items\n",
      "  METHOD fdr_bh:\n",
      "      28 GO terms found significant (< 0.05=alpha) ( 26 enriched +   2 purified): statsmodels fdr_bh\n",
      "   1,952 study items associated with significant GO IDs (enriched)\n",
      "      20 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 3509 IDs ... 74%  2,612 of  3,509 study items found in association\n",
      "100%  3,509 of  3,509 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   1,876 GO terms are associated with  2,612 of  3,509 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  5 enriched +   2 purified): statsmodels fdr_bh\n",
      "   2,049 study items associated with significant GO IDs (enriched)\n",
      "      24 study items associated with significant GO IDs (purified)\n",
      "Significant results[47] = 12 BP + 7 MF + 28 CC\n",
      "     47 items WROTE: GO_analysis_module_grey.xlsx\n",
      "     47 GOEA results for  2546 study items. WROTE: GO_analysis_module_grey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:06.288791 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1507 IDs ... 82%  1,236 of  1,507 study items found in association\n",
      "100%  1,507 of  1,507 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   4,323 GO terms are associated with  1,236 of  1,507 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 15 enriched +   2 purified): statsmodels fdr_bh\n",
      "     281 study items associated with significant GO IDs (enriched)\n",
      "       8 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1507 IDs ... 85%  1,277 of  1,507 study items found in association\n",
      "100%  1,507 of  1,507 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     696 GO terms are associated with  1,277 of  1,507 study items\n",
      "  METHOD fdr_bh:\n",
      "      53 GO terms found significant (< 0.05=alpha) ( 53 enriched +   0 purified): statsmodels fdr_bh\n",
      "   1,223 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1507 IDs ... 82%  1,236 of  1,507 study items found in association\n",
      "100%  1,507 of  1,507 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   1,290 GO terms are associated with  1,236 of  1,507 study items\n",
      "  METHOD fdr_bh:\n",
      "      18 GO terms found significant (< 0.05=alpha) ( 16 enriched +   2 purified): statsmodels fdr_bh\n",
      "     979 study items associated with significant GO IDs (enriched)\n",
      "       2 study items associated with significant GO IDs (purified)\n",
      "Significant results[88] = 17 BP + 18 MF + 53 CC\n",
      "     88 items WROTE: GO_analysis_module_blue.xlsx\n",
      "     88 GOEA results for  1297 study items. WROTE: GO_analysis_module_blue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:07.435127 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 99 IDs ... 86%     85 of     99 study items found in association\n",
      "100%     99 of     99 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     402 GO terms are associated with     85 of     99 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "       9 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 99 IDs ... 87%     86 of     99 study items found in association\n",
      "100%     99 of     99 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     160 GO terms are associated with     86 of     99 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      53 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 99 IDs ... 81%     80 of     99 study items found in association\n",
      "100%     99 of     99 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     164 GO terms are associated with     80 of     99 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "      67 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[16] = 3 BP + 5 MF + 8 CC\n",
      "     16 items WROTE: GO_analysis_module_darkgreen.xlsx\n",
      "     16 GOEA results for    80 study items. WROTE: GO_analysis_module_darkgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:06.436543 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 214 IDs ... 89%    191 of    214 study items found in association\n",
      "100%    214 of    214 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,571 GO terms are associated with    191 of    214 study items\n",
      "  METHOD fdr_bh:\n",
      "     177 GO terms found significant (< 0.05=alpha) (177 enriched +   0 purified): statsmodels fdr_bh\n",
      "     154 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 214 IDs ... 93%    199 of    214 study items found in association\n",
      "100%    214 of    214 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     229 GO terms are associated with    199 of    214 study items\n",
      "  METHOD fdr_bh:\n",
      "      46 GO terms found significant (< 0.05=alpha) ( 46 enriched +   0 purified): statsmodels fdr_bh\n",
      "     187 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 214 IDs ... 90%    193 of    214 study items found in association\n",
      "100%    214 of    214 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     380 GO terms are associated with    193 of    214 study items\n",
      "  METHOD fdr_bh:\n",
      "      33 GO terms found significant (< 0.05=alpha) ( 33 enriched +   0 purified): statsmodels fdr_bh\n",
      "     175 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[256] = 177 BP + 33 MF + 46 CC\n",
      "    256 items WROTE: GO_analysis_module_lightcyan.xlsx\n",
      "    256 GOEA results for   198 study items. WROTE: GO_analysis_module_lightcyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:08.690241 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 1933 IDs ... 84%  1,626 of  1,933 study items found in association\n",
      "100%  1,933 of  1,933 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   4,585 GO terms are associated with  1,626 of  1,933 study items\n",
      "  METHOD fdr_bh:\n",
      "      36 GO terms found significant (< 0.05=alpha) ( 33 enriched +   3 purified): statsmodels fdr_bh\n",
      "     733 study items associated with significant GO IDs (enriched)\n",
      "      16 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 1933 IDs ... 89%  1,718 of  1,933 study items found in association\n",
      "100%  1,933 of  1,933 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     816 GO terms are associated with  1,718 of  1,933 study items\n",
      "  METHOD fdr_bh:\n",
      "      57 GO terms found significant (< 0.05=alpha) ( 57 enriched +   0 purified): statsmodels fdr_bh\n",
      "   1,631 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 1933 IDs ... 86%  1,666 of  1,933 study items found in association\n",
      "100%  1,933 of  1,933 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "   1,391 GO terms are associated with  1,666 of  1,933 study items\n",
      "  METHOD fdr_bh:\n",
      "      30 GO terms found significant (< 0.05=alpha) ( 28 enriched +   2 purified): statsmodels fdr_bh\n",
      "   1,446 study items associated with significant GO IDs (enriched)\n",
      "       4 study items associated with significant GO IDs (purified)\n",
      "Significant results[123] = 36 BP + 30 MF + 57 CC\n",
      "    123 items WROTE: GO_analysis_module_turquoise.xlsx\n",
      "    123 GOEA results for  1755 study items. WROTE: GO_analysis_module_turquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.387755 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 423 IDs ... 85%    361 of    423 study items found in association\n",
      "100%    423 of    423 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,435 GO terms are associated with    361 of    423 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      23 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 423 IDs ... 88%    374 of    423 study items found in association\n",
      "100%    423 of    423 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     377 GO terms are associated with    374 of    423 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "     233 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 423 IDs ... 89%    375 of    423 study items found in association\n",
      "100%    423 of    423 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     567 GO terms are associated with    375 of    423 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     277 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[11] = 2 BP + 2 MF + 7 CC\n",
      "     11 items WROTE: GO_analysis_module_pink.xlsx\n",
      "     11 GOEA results for   347 study items. WROTE: GO_analysis_module_pink.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.575022 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 61 IDs ... 87%     53 of     61 study items found in association\n",
      "100%     61 of     61 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     292 GO terms are associated with     53 of     61 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       3 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 61 IDs ... 97%     59 of     61 study items found in association\n",
      "100%     61 of     61 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     109 GO terms are associated with     59 of     61 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 61 IDs ... 84%     51 of     61 study items found in association\n",
      "100%     61 of     61 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     111 GO terms are associated with     51 of     61 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[1] = 1 BP + 0 MF + 0 CC\n",
      "      1 items WROTE: GO_analysis_module_violet.xlsx\n",
      "      1 GOEA results for     3 study items. WROTE: GO_analysis_module_violet.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.735098 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 453 IDs ... 91%    414 of    453 study items found in association\n",
      "100%    453 of    453 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,731 GO terms are associated with    414 of    453 study items\n",
      "  METHOD fdr_bh:\n",
      "     104 GO terms found significant (< 0.05=alpha) (103 enriched +   1 purified): statsmodels fdr_bh\n",
      "     251 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 453 IDs ... 94%    427 of    453 study items found in association\n",
      "100%    453 of    453 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     378 GO terms are associated with    427 of    453 study items\n",
      "  METHOD fdr_bh:\n",
      "      55 GO terms found significant (< 0.05=alpha) ( 55 enriched +   0 purified): statsmodels fdr_bh\n",
      "     395 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 453 IDs ... 92%    416 of    453 study items found in association\n",
      "100%    453 of    453 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     726 GO terms are associated with    416 of    453 study items\n",
      "  METHOD fdr_bh:\n",
      "      17 GO terms found significant (< 0.05=alpha) ( 17 enriched +   0 purified): statsmodels fdr_bh\n",
      "     350 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[176] = 104 BP + 17 MF + 55 CC\n",
      "    176 items WROTE: GO_analysis_module_magenta.xlsx\n",
      "    176 GOEA results for   421 study items. WROTE: GO_analysis_module_magenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.832440 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 127 IDs ... 72%     92 of    127 study items found in association\n",
      "100%    127 of    127 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     631 GO terms are associated with     92 of    127 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 127 IDs ... 72%     92 of    127 study items found in association\n",
      "100%    127 of    127 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     195 GO terms are associated with     92 of    127 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 127 IDs ... 74%     94 of    127 study items found in association\n",
      "100%    127 of    127 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     168 GO terms are associated with     94 of    127 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_grey60.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_grey60.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.922303 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 267 IDs ... 78%    209 of    267 study items found in association\n",
      "100%    267 of    267 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     997 GO terms are associated with    209 of    267 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 267 IDs ... 82%    220 of    267 study items found in association\n",
      "100%    267 of    267 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     299 GO terms are associated with    220 of    267 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "     122 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 267 IDs ... 81%    216 of    267 study items found in association\n",
      "100%    267 of    267 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     388 GO terms are associated with    216 of    267 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "     164 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[5] = 0 BP + 1 MF + 4 CC\n",
      "      5 items WROTE: GO_analysis_module_salmon.xlsx\n",
      "      5 GOEA results for   187 study items. WROTE: GO_analysis_module_salmon.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.328531 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 143 IDs ... 95%    136 of    143 study items found in association\n",
      "100%    143 of    143 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     991 GO terms are associated with    136 of    143 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "       5 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 143 IDs ... 97%    138 of    143 study items found in association\n",
      "100%    143 of    143 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     272 GO terms are associated with    138 of    143 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "     126 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 143 IDs ... 97%    139 of    143 study items found in association\n",
      "100%    143 of    143 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     282 GO terms are associated with    139 of    143 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "     128 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[28] = 2 BP + 10 MF + 16 CC\n",
      "     28 items WROTE: GO_analysis_module_royalblue.xlsx\n",
      "     28 GOEA results for   139 study items. WROTE: GO_analysis_module_royalblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.401638 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 85 IDs ... 92%     78 of     85 study items found in association\n",
      "100%     85 of     85 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     551 GO terms are associated with     78 of     85 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 85 IDs ... 92%     78 of     85 study items found in association\n",
      "100%     85 of     85 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     195 GO terms are associated with     78 of     85 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      50 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 85 IDs ... 89%     76 of     85 study items found in association\n",
      "100%     85 of     85 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     186 GO terms are associated with     76 of     85 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      65 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 0 BP + 1 MF + 3 CC\n",
      "      4 items WROTE: GO_analysis_module_darkorange.xlsx\n",
      "      4 GOEA results for    72 study items. WROTE: GO_analysis_module_darkorange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.536354 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 73 IDs ... 86%     63 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     518 GO terms are associated with     63 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "      28 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 73 IDs ... 95%     69 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     112 GO terms are associated with     69 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      19 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 73 IDs ... 90%     66 of     73 study items found in association\n",
      "100%     73 of     73 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     199 GO terms are associated with     66 of     73 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[19] = 16 BP + 0 MF + 3 CC\n",
      "     19 items WROTE: GO_analysis_module_saddlebrown.xlsx\n",
      "     19 GOEA results for    33 study items. WROTE: GO_analysis_module_saddlebrown.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.451756 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 107 IDs ... 85%     91 of    107 study items found in association\n",
      "100%    107 of    107 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     455 GO terms are associated with     91 of    107 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       3 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 107 IDs ... 90%     96 of    107 study items found in association\n",
      "100%    107 of    107 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     169 GO terms are associated with     96 of    107 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 107 IDs ... 88%     94 of    107 study items found in association\n",
      "100%    107 of    107 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     187 GO terms are associated with     94 of    107 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[1] = 1 BP + 0 MF + 0 CC\n",
      "      1 items WROTE: GO_analysis_module_darkturquoise.xlsx\n",
      "      1 GOEA results for     3 study items. WROTE: GO_analysis_module_darkturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.873166 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 737 IDs ... 68%    498 of    737 study items found in association\n",
      "100%    737 of    737 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,739 GO terms are associated with    498 of    737 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 737 IDs ... 74%    544 of    737 study items found in association\n",
      "100%    737 of    737 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     415 GO terms are associated with    544 of    737 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  1 enriched +   1 purified): statsmodels fdr_bh\n",
      "     189 study items associated with significant GO IDs (enriched)\n",
      "      71 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 737 IDs ... 72%    529 of    737 study items found in association\n",
      "100%    737 of    737 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     608 GO terms are associated with    529 of    737 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     409 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 0 BP + 2 MF + 2 CC\n",
      "      4 items WROTE: GO_analysis_module_yellow.xlsx\n",
      "      4 GOEA results for   450 study items. WROTE: GO_analysis_module_yellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.974234 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 98 IDs ... 86%     84 of     98 study items found in association\n",
      "100%     98 of     98 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     536 GO terms are associated with     84 of     98 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 98 IDs ... 87%     85 of     98 study items found in association\n",
      "100%     98 of     98 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     161 GO terms are associated with     85 of     98 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 98 IDs ... 89%     87 of     98 study items found in association\n",
      "100%     98 of     98 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     180 GO terms are associated with     87 of     98 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      67 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[1] = 0 BP + 1 MF + 0 CC\n",
      "      1 items WROTE: GO_analysis_module_darkgrey.xlsx\n",
      "      1 GOEA results for    67 study items. WROTE: GO_analysis_module_darkgrey.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.290376 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 341 IDs ... 74%    252 of    341 study items found in association\n",
      "100%    341 of    341 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,227 GO terms are associated with    252 of    341 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 341 IDs ... 82%    279 of    341 study items found in association\n",
      "100%    341 of    341 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     337 GO terms are associated with    279 of    341 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "     141 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 341 IDs ... 77%    261 of    341 study items found in association\n",
      "100%    341 of    341 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     436 GO terms are associated with    261 of    341 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "     192 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[4] = 0 BP + 1 MF + 3 CC\n",
      "      4 items WROTE: GO_analysis_module_black.xlsx\n",
      "      4 GOEA results for   229 study items. WROTE: GO_analysis_module_black.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.075495 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 911 IDs ... 92%    834 of    911 study items found in association\n",
      "100%    911 of    911 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,843 GO terms are associated with    834 of    911 study items\n",
      "  METHOD fdr_bh:\n",
      "      45 GO terms found significant (< 0.05=alpha) ( 42 enriched +   3 purified): statsmodels fdr_bh\n",
      "     409 study items associated with significant GO IDs (enriched)\n",
      "       8 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 911 IDs ... 95%    861 of    911 study items found in association\n",
      "100%    911 of    911 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     668 GO terms are associated with    861 of    911 study items\n",
      "  METHOD fdr_bh:\n",
      "      49 GO terms found significant (< 0.05=alpha) ( 48 enriched +   1 purified): statsmodels fdr_bh\n",
      "     804 study items associated with significant GO IDs (enriched)\n",
      "      18 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 911 IDs ... 93%    843 of    911 study items found in association\n",
      "100%    911 of    911 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     901 GO terms are associated with    843 of    911 study items\n",
      "  METHOD fdr_bh:\n",
      "      55 GO terms found significant (< 0.05=alpha) ( 53 enriched +   2 purified): statsmodels fdr_bh\n",
      "     786 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[149] = 45 BP + 55 MF + 49 CC\n",
      "    149 items WROTE: GO_analysis_module_green.xlsx\n",
      "    149 GOEA results for   862 study items. WROTE: GO_analysis_module_green.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.103439 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 44 IDs ... 89%     39 of     44 study items found in association\n",
      "100%     44 of     44 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     331 GO terms are associated with     39 of     44 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       3 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 44 IDs ... 93%     41 of     44 study items found in association\n",
      "100%     44 of     44 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     142 GO terms are associated with     41 of     44 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      26 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 44 IDs ... 91%     40 of     44 study items found in association\n",
      "100%     44 of     44 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     124 GO terms are associated with     40 of     44 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 1 BP + 0 MF + 2 CC\n",
      "      3 items WROTE: GO_analysis_module_plum1.xlsx\n",
      "      3 GOEA results for    27 study items. WROTE: GO_analysis_module_plum1.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.497804 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 355 IDs ... 91%    324 of    355 study items found in association\n",
      "100%    355 of    355 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,606 GO terms are associated with    324 of    355 study items\n",
      "  METHOD fdr_bh:\n",
      "       9 GO terms found significant (< 0.05=alpha) (  9 enriched +   0 purified): statsmodels fdr_bh\n",
      "      79 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 355 IDs ... 95%    337 of    355 study items found in association\n",
      "100%    355 of    355 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     469 GO terms are associated with    337 of    355 study items\n",
      "  METHOD fdr_bh:\n",
      "      40 GO terms found significant (< 0.05=alpha) ( 40 enriched +   0 purified): statsmodels fdr_bh\n",
      "     319 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 355 IDs ... 90%    321 of    355 study items found in association\n",
      "100%    355 of    355 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     529 GO terms are associated with    321 of    355 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "     288 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[59] = 9 BP + 10 MF + 40 CC\n",
      "     59 items WROTE: GO_analysis_module_tan.xlsx\n",
      "     59 GOEA results for   333 study items. WROTE: GO_analysis_module_tan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.366670 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 89 IDs ... 96%     85 of     89 study items found in association\n",
      "100%     89 of     89 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     585 GO terms are associated with     85 of     89 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "       5 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 89 IDs ... 97%     86 of     89 study items found in association\n",
      "100%     89 of     89 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     169 GO terms are associated with     86 of     89 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      67 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 89 IDs ... 98%     87 of     89 study items found in association\n",
      "100%     89 of     89 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     188 GO terms are associated with     87 of     89 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      74 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[10] = 1 BP + 1 MF + 8 CC\n",
      "     10 items WROTE: GO_analysis_module_white.xlsx\n",
      "     10 GOEA results for    81 study items. WROTE: GO_analysis_module_white.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.643448 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 782 IDs ... 88%    692 of    782 study items found in association\n",
      "100%    782 of    782 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   2,882 GO terms are associated with    692 of    782 study items\n",
      "  METHOD fdr_bh:\n",
      "      26 GO terms found significant (< 0.05=alpha) ( 25 enriched +   1 purified): statsmodels fdr_bh\n",
      "     249 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 782 IDs ... 93%    727 of    782 study items found in association\n",
      "100%    782 of    782 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     577 GO terms are associated with    727 of    782 study items\n",
      "  METHOD fdr_bh:\n",
      "      44 GO terms found significant (< 0.05=alpha) ( 44 enriched +   0 purified): statsmodels fdr_bh\n",
      "     687 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 782 IDs ... 91%    713 of    782 study items found in association\n",
      "100%    782 of    782 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     897 GO terms are associated with    713 of    782 study items\n",
      "  METHOD fdr_bh:\n",
      "      11 GO terms found significant (< 0.05=alpha) ( 10 enriched +   1 purified): statsmodels fdr_bh\n",
      "     598 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[81] = 26 BP + 11 MF + 44 CC\n",
      "     81 items WROTE: GO_analysis_module_red.xlsx\n",
      "     81 GOEA results for   734 study items. WROTE: GO_analysis_module_red.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.669227 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 365 IDs ... 78%    286 of    365 study items found in association\n",
      "100%    365 of    365 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     938 GO terms are associated with    286 of    365 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      56 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 365 IDs ... 83%    304 of    365 study items found in association\n",
      "100%    365 of    365 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     291 GO terms are associated with    304 of    365 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  5 enriched +   1 purified): statsmodels fdr_bh\n",
      "     187 study items associated with significant GO IDs (enriched)\n",
      "      28 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 365 IDs ... 86%    315 of    365 study items found in association\n",
      "100%    365 of    365 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     366 GO terms are associated with    315 of    365 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "     264 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[13] = 1 BP + 6 MF + 6 CC\n",
      "     13 items WROTE: GO_analysis_module_purple.xlsx\n",
      "     13 GOEA results for   287 study items. WROTE: GO_analysis_module_purple.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.247703 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 79 IDs ... 92%     73 of     79 study items found in association\n",
      "100%     79 of     79 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     518 GO terms are associated with     73 of     79 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      14 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 79 IDs ... 96%     76 of     79 study items found in association\n",
      "100%     79 of     79 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     133 GO terms are associated with     76 of     79 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "      33 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 79 IDs ... 94%     74 of     79 study items found in association\n",
      "100%     79 of     79 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     177 GO terms are associated with     74 of     79 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[2] = 1 BP + 0 MF + 1 CC\n",
      "      2 items WROTE: GO_analysis_module_skyblue.xlsx\n",
      "      2 GOEA results for    44 study items. WROTE: GO_analysis_module_skyblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.228906 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 140 IDs ... 86%    121 of    140 study items found in association\n",
      "100%    140 of    140 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     702 GO terms are associated with    121 of    140 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 140 IDs ... 91%    127 of    140 study items found in association\n",
      "100%    140 of    140 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     215 GO terms are associated with    127 of    140 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      71 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 140 IDs ... 89%    125 of    140 study items found in association\n",
      "100%    140 of    140 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     239 GO terms are associated with    125 of    140 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "     104 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 1 MF + 2 CC\n",
      "      3 items WROTE: GO_analysis_module_lightyellow.xlsx\n",
      "      3 GOEA results for   112 study items. WROTE: GO_analysis_module_lightyellow.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.777264 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 249 IDs ... 92%    228 of    249 study items found in association\n",
      "100%    249 of    249 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "   1,466 GO terms are associated with    228 of    249 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      23 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 249 IDs ... 94%    233 of    249 study items found in association\n",
      "100%    249 of    249 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     298 GO terms are associated with    233 of    249 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "     181 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 249 IDs ... 93%    232 of    249 study items found in association\n",
      "100%    249 of    249 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     436 GO terms are associated with    232 of    249 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "     187 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[12] = 3 BP + 2 MF + 7 CC\n",
      "     12 items WROTE: GO_analysis_module_midnightblue.xlsx\n",
      "     12 GOEA results for   216 study items. WROTE: GO_analysis_module_midnightblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.045340 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 74 IDs ... 96%     71 of     74 study items found in association\n",
      "100%     74 of     74 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     423 GO terms are associated with     71 of     74 study items\n",
      "  METHOD fdr_bh:\n",
      "      10 GO terms found significant (< 0.05=alpha) ( 10 enriched +   0 purified): statsmodels fdr_bh\n",
      "      32 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 74 IDs ... 97%     72 of     74 study items found in association\n",
      "100%     74 of     74 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     165 GO terms are associated with     72 of     74 study items\n",
      "  METHOD fdr_bh:\n",
      "      28 GO terms found significant (< 0.05=alpha) ( 28 enriched +   0 purified): statsmodels fdr_bh\n",
      "      69 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 74 IDs ... 97%     72 of     74 study items found in association\n",
      "100%     74 of     74 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     131 GO terms are associated with     72 of     74 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      70 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[45] = 10 BP + 7 MF + 28 CC\n",
      "     45 items WROTE: GO_analysis_module_paleturquoise.xlsx\n",
      "     45 GOEA results for    73 study items. WROTE: GO_analysis_module_paleturquoise.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.419518 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 43 IDs ... 81%     35 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     341 GO terms are associated with     35 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 43 IDs ... 81%     35 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      96 GO terms are associated with     35 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 43 IDs ... 84%     36 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     113 GO terms are associated with     36 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_darkolivegreen.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_darkolivegreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.553703 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 275 IDs ... 87%    240 of    275 study items found in association\n",
      "100%    275 of    275 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     875 GO terms are associated with    240 of    275 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      59 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 275 IDs ... 92%    254 of    275 study items found in association\n",
      "100%    275 of    275 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     273 GO terms are associated with    254 of    275 study items\n",
      "  METHOD fdr_bh:\n",
      "       5 GO terms found significant (< 0.05=alpha) (  5 enriched +   0 purified): statsmodels fdr_bh\n",
      "     187 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 275 IDs ... 89%    244 of    275 study items found in association\n",
      "100%    275 of    275 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     319 GO terms are associated with    244 of    275 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "     221 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[13] = 2 BP + 6 MF + 5 CC\n",
      "     13 items WROTE: GO_analysis_module_cyan.xlsx\n",
      "     13 GOEA results for   237 study items. WROTE: GO_analysis_module_cyan.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.340845 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 79 IDs ... 89%     70 of     79 study items found in association\n",
      "100%     79 of     79 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     601 GO terms are associated with     70 of     79 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 79 IDs ... 91%     72 of     79 study items found in association\n",
      "100%     79 of     79 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     155 GO terms are associated with     72 of     79 study items\n",
      "  METHOD fdr_bh:\n",
      "       3 GO terms found significant (< 0.05=alpha) (  3 enriched +   0 purified): statsmodels fdr_bh\n",
      "      47 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 79 IDs ... 85%     67 of     79 study items found in association\n",
      "100%     79 of     79 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     181 GO terms are associated with     67 of     79 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[3] = 0 BP + 0 MF + 3 CC\n",
      "      3 items WROTE: GO_analysis_module_orange.xlsx\n",
      "      3 GOEA results for    47 study items. WROTE: GO_analysis_module_orange.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.016457 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 78 IDs ... 72%     56 of     78 study items found in association\n",
      "100%     78 of     78 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     255 GO terms are associated with     56 of     78 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 78 IDs ... 85%     66 of     78 study items found in association\n",
      "100%     78 of     78 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      92 GO terms are associated with     66 of     78 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 78 IDs ... 85%     66 of     78 study items found in association\n",
      "100%     78 of     78 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     117 GO terms are associated with     66 of     78 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_steelblue.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_steelblue.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.794144 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 165 IDs ... 91%    150 of    165 study items found in association\n",
      "100%    165 of    165 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     893 GO terms are associated with    150 of    165 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "      16 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 165 IDs ... 94%    155 of    165 study items found in association\n",
      "100%    165 of    165 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     277 GO terms are associated with    155 of    165 study items\n",
      "  METHOD fdr_bh:\n",
      "      16 GO terms found significant (< 0.05=alpha) ( 16 enriched +   0 purified): statsmodels fdr_bh\n",
      "     137 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 165 IDs ... 93%    154 of    165 study items found in association\n",
      "100%    165 of    165 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     304 GO terms are associated with    154 of    165 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "     129 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[22] = 2 BP + 4 MF + 16 CC\n",
      "     22 items WROTE: GO_analysis_module_lightgreen.xlsx\n",
      "     22 GOEA results for   150 study items. WROTE: GO_analysis_module_lightgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.215976 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 139 IDs ... 90%    125 of    139 study items found in association\n",
      "100%    139 of    139 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     666 GO terms are associated with    125 of    139 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 139 IDs ... 93%    129 of    139 study items found in association\n",
      "100%    139 of    139 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     226 GO terms are associated with    129 of    139 study items\n",
      "  METHOD fdr_bh:\n",
      "       4 GO terms found significant (< 0.05=alpha) (  4 enriched +   0 purified): statsmodels fdr_bh\n",
      "      91 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 139 IDs ... 89%    124 of    139 study items found in association\n",
      "100%    139 of    139 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     221 GO terms are associated with    124 of    139 study items\n",
      "  METHOD fdr_bh:\n",
      "       1 GO terms found significant (< 0.05=alpha) (  1 enriched +   0 purified): statsmodels fdr_bh\n",
      "     107 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[5] = 0 BP + 1 MF + 4 CC\n",
      "      5 items WROTE: GO_analysis_module_darkred.xlsx\n",
      "      5 GOEA results for   121 study items. WROTE: GO_analysis_module_darkred.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.205761 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 48 IDs ... 85%     41 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     482 GO terms are associated with     41 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "      23 GO terms found significant (< 0.05=alpha) ( 23 enriched +   0 purified): statsmodels fdr_bh\n",
      "      19 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 48 IDs ... 90%     43 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     124 GO terms are associated with     43 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "      15 GO terms found significant (< 0.05=alpha) ( 15 enriched +   0 purified): statsmodels fdr_bh\n",
      "      38 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 48 IDs ... 94%     45 of     48 study items found in association\n",
      "100%     48 of     48 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     159 GO terms are associated with     45 of     48 study items\n",
      "  METHOD fdr_bh:\n",
      "      22 GO terms found significant (< 0.05=alpha) ( 22 enriched +   0 purified): statsmodels fdr_bh\n",
      "      41 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[60] = 23 BP + 22 MF + 15 CC\n",
      "     60 items WROTE: GO_analysis_module_darkmagenta.xlsx\n",
      "     60 GOEA results for    43 study items. WROTE: GO_analysis_module_darkmagenta.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.537411 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 45 IDs ... 96%     43 of     45 study items found in association\n",
      "100%     45 of     45 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     380 GO terms are associated with     43 of     45 study items\n",
      "  METHOD fdr_bh:\n",
      "       2 GO terms found significant (< 0.05=alpha) (  2 enriched +   0 purified): statsmodels fdr_bh\n",
      "       5 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 45 IDs ... 96%     43 of     45 study items found in association\n",
      "100%     45 of     45 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "     115 GO terms are associated with     43 of     45 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      35 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 45 IDs ... 91%     41 of     45 study items found in association\n",
      "100%     45 of     45 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     100 GO terms are associated with     41 of     45 study items\n",
      "  METHOD fdr_bh:\n",
      "       7 GO terms found significant (< 0.05=alpha) (  7 enriched +   0 purified): statsmodels fdr_bh\n",
      "      36 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[17] = 2 BP + 7 MF + 8 CC\n",
      "     17 items WROTE: GO_analysis_module_yellowgreen.xlsx\n",
      "     17 GOEA results for    40 study items. WROTE: GO_analysis_module_yellowgreen.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:04.739587 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 46 IDs ... 85%     39 of     46 study items found in association\n",
      "100%     46 of     46 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     393 GO terms are associated with     39 of     46 study items\n",
      "  METHOD fdr_bh:\n",
      "       8 GO terms found significant (< 0.05=alpha) (  8 enriched +   0 purified): statsmodels fdr_bh\n",
      "      15 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 46 IDs ... 98%     45 of     46 study items found in association\n",
      "100%     46 of     46 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      91 GO terms are associated with     45 of     46 study items\n",
      "  METHOD fdr_bh:\n",
      "       6 GO terms found significant (< 0.05=alpha) (  6 enriched +   0 purified): statsmodels fdr_bh\n",
      "      24 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 46 IDs ... 93%     43 of     46 study items found in association\n",
      "100%     46 of     46 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "     108 GO terms are associated with     43 of     46 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[14] = 8 BP + 0 MF + 6 CC\n",
      "     14 items WROTE: GO_analysis_module_sienna3.xlsx\n",
      "     14 GOEA results for    28 study items. WROTE: GO_analysis_module_sienna3.txt\n",
      "  EXISTS: go-basic.obo\n",
      "  EXISTS: gene2go\n",
      "go-basic.obo: fmt(1.2) rel(2021-07-02) 47,229 GO Terms\n",
      "HMS:0:00:05.058004 330,320 annotations, 20,687 genes, 18,684 GOs, 1 taxids READ: gene2go \n",
      "MF 18,191 annotated human genes\n",
      "CC 19,424 annotated human genes\n",
      "BP 18,506 annotated human genes\n",
      "\n",
      "Load BP Gene Ontology Analysis ...\n",
      " 70% 20,237 of 29,107 population items found in association\n",
      "\n",
      "Load CC Gene Ontology Analysis ...\n",
      " 74% 21,430 of 29,107 population items found in association\n",
      "\n",
      "Load MF Gene Ontology Analysis ...\n",
      " 70% 20,355 of 29,107 population items found in association\n",
      "\n",
      "Run BP Gene Ontology Analysis: current study set of 43 IDs ... 88%     38 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 12,438 uncorrected p-values using fisher\n",
      "  12,438 GO terms are associated with 17,849 of 29,107 population items\n",
      "     245 GO terms are associated with     38 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run CC Gene Ontology Analysis: current study set of 43 IDs ... 91%     39 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 1,756 uncorrected p-values using fisher\n",
      "   1,756 GO terms are associated with 18,713 of 29,107 population items\n",
      "      94 GO terms are associated with     39 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "\n",
      "Run MF Gene Ontology Analysis: current study set of 43 IDs ... 86%     37 of     43 study items found in association\n",
      "100%     43 of     43 study items found in population(29107)\n",
      "Calculating 4,434 uncorrected p-values using fisher\n",
      "   4,434 GO terms are associated with 17,839 of 29,107 population items\n",
      "      97 GO terms are associated with     37 of     43 study items\n",
      "  METHOD fdr_bh:\n",
      "       0 GO terms found significant (< 0.05=alpha) (  0 enriched +   0 purified): statsmodels fdr_bh\n",
      "       0 study items associated with significant GO IDs (enriched)\n",
      "       0 study items associated with significant GO IDs (purified)\n",
      "Significant results[0] = 0 BP + 0 MF + 0 CC\n",
      "      0 items. NOT WRITING GO_analysis_module_skyblue3.xlsx\n",
      "      0 GOEA results. NOT WRITING GO_analysis_module_skyblue3.txt\n"
     ]
    }
   ],
   "source": [
    "for mod in get_wgcna_modules().module.unique():\n",
    "    run_goea(mod)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
